---
title: "Tomato_seed_microbiome_review"
author: "Expedito Olimi"
format: html
server: shiny
---

#Set the packages
```{r}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")
library(phyloseq);packageVersion("phyloseq")#‘1.46.0’
library(vegan);packageVersion("vegan")# ‘2.6.4’
library(genefilter);packageVersion("genefilter")#‘1.84.0’
library(psych);packageVersion("psych")#‘2.4.1’
library(DESeq2);packageVersion("DESeq2")#‘1.42.1’
library(ggplot2);packageVersion("ggplot2")#‘3.5.0’
library(reshape2);packageVersion("reshape2")#‘1.4.4’
library(microbiome);packageVersion("microbiome")#‘1.24.0’
library(ranacapa);packageVersion("ranacapa")#‘0.1.0’
library(S4Vectors);packageVersion("S4Vectors")#‘0.40.2’
library(Biobase);packageVersion("Biobase")#‘2.62.0’
library(BiocGenerics);packageVersion("BiocGenerics")#‘0.48.1’
library(parallel);packageVersion("parallel")#‘4.3.2’
library(ggpubr);packageVersion("ggpubr")#‘0.6.0’
#library(metagMisc)
library(decontam);packageVersion("decontam")#‘1.22.0’
library(RColorBrewer);packageVersion("RColorBrewer")#‘1.1.3’
library(multcompView);packageVersion("multcompView")#‘0.1.10’
library(metagenomeSeq);packageVersion("metagenomeSeq")#‘1.43.0’
#library(devtools)
library(lawstat);packageVersion("lawstat")#‘3.6’
library(DirichletMultinomial);packageVersion("DirichletMultinomial")#‘1.44.0’
library(lattice);packageVersion("lattice")# ‘0.22.5’
library(xtable);packageVersion("xtable")#‘1.8.4’
library(permute);packageVersion("permute")#‘0.9.7’
library(grid);packageVersion("grid")#‘4.3.2’
library("doParallel");packageVersion("doParallel")#‘1.0.17’
#library(igraph)
library(ade4);packageVersion("ade4")#‘1.7.22’
#library(rgl)
library(car);packageVersion("car")#‘3.1.2’
library(gridExtra);packageVersion("gridExtra")# ‘2.3’
library(tidyverse);packageVersion("tidyverse")#‘2.0.0’
library(phia);packageVersion("phia")#‘0.3.1’
library(lme4);packageVersion("lme4")#‘1.1.35.1’
library(broom);packageVersion("broom")#‘1.0.5’
library(AICcmodavg);packageVersion("AICcmodavg")#‘2.3.3’
library(multcomp);packageVersion("multcomp")#‘1.4.25’
library(rcompanion);packageVersion("rcompanion")#‘2.4.35’
library(hrbrthemes);packageVersion("hrbrthemes")#‘0.8.7’
library(viridis);packageVersion("viridis")#‘0.6.5’
library(ape);packageVersion("ape")#‘5.7.1’#loading the tree
library(microbiomeutilities);packageVersion("microbiomeutilities")#‘1.0.17’#process taxonomy table
library(microViz);packageVersion("microViz")#‘0.12.1’#clean taxonomy table to be cited
library(GUniFrac);packageVersion("GUniFrac")#‘1.8’
library(ggtext) ;packageVersion("ggtext")# ‘0.1.2’for rotated labels on ord_plot() 
library(ggraph);packageVersion("ggraph")# ‘2.2.1’for taxatree_plots()
library(DT);packageVersion("DT")# ‘0.32’for tax_fix_interactive()
library(corncob);packageVersion("corncob")#‘0.4.1’ for beta binomial models in tax_model()
library(VennDiagram);packageVersion("VennDiagram")#‘1.7.3’
library(UpSetR);packageVersion("UpSetR")#‘1.4.0’
library(rstatix);packageVersion("rstatix")#‘0.7.2’
library(ALDEx2);packageVersion("ALDEx2")#[1] ‘1.34.0’
library(cowplot);packageVersion("cowplot")#‘1.1.3’
library(HMP);packageVersion("HMP")#HMP: Hypothesis Testing & Power Calculations for Comparing Metagenomic Samples from HMP #‘2.0.1’
library(dendextend);packageVersion("dendextend")#‘1.17.1’
library(dplyr)
library(gcookbook)
library(tidyverse)
library(patchwork)
library(MicrobiotaProcess)
library(rmarkdown)#install.packages("rmarkdown")
library(dada2)
library(Biostrings)
library(ggalluvial)
library(ggh4x)
library(tidyverse)
library(ggrepel)
library(qiime2R)#https://github.com/jbisanz/qiime2R
library(dplyr)
library(pals)
library(ggforce)
library(gridExtra)
library(vegan)
library(data.table)
library(picante)
library(ggtreeExtra)
library(ggpmisc)
library(DECIPHER) #BiocManager::install("DECIPHER")
library(phangorn) #remotes::install_github("KlausVigo/phangorn")
library(ggtree) #BiocManager::install("ggtree")
library(plotly)
library(speedyseq) #remotes::install_github("mikemc/speedyseq")
library(ComplexHeatmap)
library(agricolae)
library(scales)
library(ggpmisc)
library(pairwiseAdonis)#library(devtools) install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
#remotes::install_github("mahendra-mariadassou/phyloseq-extended", ref = "dev")
library(phyloseq.extended)#remotes::install_github(repo = "mahendra-mariadassou/phyloseq-extended@v0.1.0.9000-beta", ref = "dev")
#The following need to be installed but NOT loaded (same naming of different functions):
#library(metagMisc) #devtools::install_github("vmikk/metagMisc")
#library(metagenomeSeq) #BiocManager::install("metagenomeSeq")
source("Rscript_parwise.adonis.txt")
source("aggregate_top_taxa.txt")
source("PD.txt")
source("aggregate_rare.txt")
source("plot_composition.txt")
source("RSCRIPT_parwise.adonis.txt")
source("plot_hierarchy.R")
library(phyloseq)
library(ggplot2)
library(microbiome)   # for rarefy_even_depth
library(tidyr)        # for pivoting data
library(dplyr)
```


#Load colors 
```{r}
mycolors=c("grey35","#00AFBB","goldenrod","#0dba2f","#392682","coral3","#8E9CA3",
           "chartreuse4","grey85","#FF2700","bisque4","#0B775E","darkred","#F39B7FB2","#FA41CA",
           "steelblue","blue","#D16103", "#C3D7A4", "#52854C","deepskyblue","#293352","#FFDB6D",
           "Yellow","#CC79A7","#392682", "#CAB2D6","maroon","orchid1","khaki2","#86486F","#7E6148B2");mycolors

mycola=c("grey35","chartreuse4","#FE7F9D","darkturquoise", "#F0E442","#009E73","#FF0000",
         "Orange","blue","#850dba","green1","#FF6AD5","#6A3D9A","Yellow","Blue",
         "deeppink","#1C0221","#00AFBB","steelblue","#D16103", "#C3D7A4","#52854C",
         "#4E84C4","#293352","#FFDB6D","#CC79A7","#C4961A", "#F4EDCA","dodgerblue2", 
         "#E31A1C","green4","#0dba2f","black","gold1", "skyblue2","#FB9A99","#CAB2D6",
         "blue1","steelblue4","Sky Blue","grey85","#3B9AB2","goldenrod","yellow3",
         "darkorange4","brown","#999999", "#E69F00","#56B4E9","#009E73","#F0E442",
         "#D55E00", "#E69F00","#ba0d42","yellow4","#D55E00", "#CC79A7","darkred",
         "#56B4E9","#86486F","#446455", "bisque4","#F7B1AB","#807182","#E3DDDD",
         "#A45C3D","coral3","#000099","#E2C59F","#B6C5CC", "#8E9CA3","#556670",
         "#771C19", "#AA3929","#E25033","#F27314", "#F8A31B","#2D2D2D","#91D1C2B2",
         "#DC0000B2","#7E6148B2","#E64B35B2", "#4DBBD5B2","#00A087B2","#3C5488B2",
         "#F39B7FB2","#8491B4B2","#FF2700","#62BBA5","#B58900","#80CDC1","#8073AC",
     "#0072B2","#392682","#FA41CA","#0B775E","#02401B","#F4B5BD","#B6854D","deepskyblue");mycola

mycol = c("bisque4", "deeppink", "chartreuse4","blue","coral3","goldenrod","deepskyblue","darkred","#0dba2f", "Orange","Yellow", "#850dba","Blue", "Sky Blue", "#ba0d42");mycol

```



#Load the phyloseq object ps2 contained in the ps2.RData object
#This data has been standardized to a prevalence threshold as 0.25% of total samples
```{r}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")
load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/ps2.RData")

library(phyloseq)
library(ggplot2)
library(microbiome)   # for rarefy_even_depth
library(tidyr)        # for pivoting data
library(dplyr)

# Define rarefaction depths
depths <- c(500, 1000,2500, 5000,10000, 50000,100000)

my_palette=c(
  "100" = "grey35",
  "1000" ="#00AFBB",
  "2500" ="goldenrod",
  "5000" ="#0dba2f",
  "10000" ="#392682",
  "100000" = "coral3");my_palette

# Initialize list to store results
alpha_list <- list()
ps=ps2;ps
# Loop over depths
for (d in depths) {
  # Rarefy phyloseq object
  rarefied <- rarefy_even_depth(ps, sample.size = d, rngseed = 123, verbose = FALSE)
  
  # Estimate alpha diversity
  alpha_df <- estimate_richness(rarefied, measures = c("Observed", "Shannon", "Simpson", "InvSimpson", "Chao1", "ACE"));alpha_df
  
  # Add depth and sample ID
  alpha_df$Depth <- d
  alpha_df$SampleID <- rownames(alpha_df)
  
  # Store in list
  alpha_list[[as.character(d)]] <- alpha_df
}

# Combine all alpha diversity dataframes
alpha_all <- do.call(rbind, alpha_list)

# Reshape for ggplot (long format)
alpha_long <- alpha_all %>%
  pivot_longer(cols = c("Observed", "Shannon", "Simpson", "InvSimpson", "Chao1", "ACE"),
               names_to = "Metric", values_to = "Value");alpha_long
#plot
p <- ggplot(alpha_long, aes(x = factor(Depth), y = Value)) +
  geom_boxplot(aes(fill = factor(Depth))) +
  facet_wrap(~ Metric, scales = "free_y") +
  scale_fill_manual(values = my_palette) +
  labs(x = "Sequencing Depth", y = "Alpha Diversity",
       title = "Alpha Diversity Metrics Across Sequencing Depths") +
  theme_minimal() +
  theme(legend.position = "none")

ggsave("alpha_diversity_plot.tiff", plot = p, width = 12, height = 8, dpi = 300)

```


```{r}
# Load data and libraries
load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/ps2.RData")
library(phyloseq)
library(ggplot2)
library(microbiome)
library(tidyr)
library(dplyr)
library(FSA)
library(multcompView)

# Define rarefaction depths and palette
depths <- c(500, 1000, 2500, 5000, 10000, 50000, 100000)
my_palette <- c("500" = "grey35", "1000" = "#00AFBB", "2500" = "goldenrod",
                "5000" = "#0dba2f", "10000" = "#392682", "50000" = "coral3","1000000" = "#8E9CA3")
# Rarefy and compute alpha diversity
alpha_list <- list()
ps <- ps2
for (d in depths) {
  rarefied <- rarefy_even_depth(ps, sample.size = d, rngseed = 123, verbose = FALSE)
  alpha_df <- estimate_richness(rarefied, measures = c("Observed", "Shannon", "Simpson", "InvSimpson", "Chao1", "ACE"))
  alpha_df$Depth <- d
  alpha_df$SampleID <- rownames(alpha_df)
  alpha_list[[as.character(d)]] <- alpha_df
}
alpha_all <- do.call(rbind, alpha_list);alpha_all

# Reshape for plotting
alpha_long <- alpha_all %>%
  pivot_longer(cols = c("Observed", "Shannon", "Simpson", "InvSimpson", "Chao1", "ACE"),
               names_to = "Metric", values_to = "Value") %>%
  filter(is.finite(Value));alpha_long

# Function to compute significance letters
get_significance_letters <- function(df) {
  if (length(unique(df$Depth)) < 2 || nrow(df) < 2) return(data.frame())
  dunn_res <- dunnTest(Value ~ factor(Depth), data = df, method = "bh")
  pvals <- dunn_res$res$P.adj
  names(pvals) <- dunn_res$res$Comparison
  letters <- multcompLetters(pvals)$Letters
  depth_levels <- sort(unique(df$Depth))
  label_map <- setNames(letters[1:length(depth_levels)], depth_levels)
  df %>%
    group_by(Depth) %>%
    summarise(ypos = max(Value, na.rm = TRUE) + 0.5) %>%
    mutate(label = label_map[as.character(Depth)],
           Metric = unique(df$Metric)) %>%
    filter(!is.na(label), !is.na(ypos))
}

# Compute annotations for all metrics
annot_all <- alpha_long %>%
  group_by(Metric) %>%
  group_split() %>%
  lapply(get_significance_letters) %>%
  bind_rows();annot_all

# Plot with significance letters
p <- ggplot(alpha_long, aes(x = factor(Depth), y = Value)) +
  geom_boxplot(aes(fill = factor(Depth))) +
  geom_text(data = annot_all, aes(x = factor(Depth), y = ypos, label = label),
            color = "black", size = 4) +
  facet_wrap(~ Metric, scales = "free_y") +
  scale_fill_manual(values = my_palette) +
  labs(x = "Sequencing Depth", y = "Alpha Diversity",
       title = "Alpha Diversity Metrics Across Sequencing Depths") +
  theme_minimal() +
  theme(legend.position = "none");p 

#ggsave("alpha_diversity_plot.tiff", plot = p, width = 12, height = 8, dpi = 300)
ggsave(
   "/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/alpha_diversity_plot.tiff",     
  plot = p,
   scale = 1,
 width = 28,
  height = 16,
  units = c("cm"),
 dpi = 600)

```




```{r}
# Load data and libraries
load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/ps2.RData")
library(phyloseq)
library(ggplot2)
library(microbiome)
library(tidyr)
library(dplyr)
library(FSA)
library(multcompView)

# Define rarefaction depths and palette
depths <- c(500, 1000, 2500, 5000, 10000, 50000, 100000)
my_palette <- c("500" = "grey35", "1000" = "#00AFBB", "2500" = "goldenrod",
                "5000" = "#0dba2f", "10000" = "#392682", "50000" = "coral3","1000000" = "#8E9CA3")

# Initialize lists
alpha_list <- list()
lost_samples <- data.frame(Depth = integer(), Retained = integer(), Lost = integer())
lost_sample_names <- list()

# Rarefy and compute alpha diversity
ps <- ps2
total_samples <- nsamples(ps)

for (d in depths) {
  rarefied <- rarefy_even_depth(ps, sample.size = d, rngseed = 123, verbose = FALSE)
  
  retained_samples <- sample_names(rarefied)
  original_samples <- sample_names(ps)
  lost <- setdiff(original_samples, retained_samples)
  
  lost_samples <- rbind(lost_samples, data.frame(Depth = d, Retained = length(retained_samples), Lost = length(lost)))
  lost_sample_names[[as.character(d)]] <- lost
  
  alpha_df <- estimate_richness(rarefied, measures = c("Observed", "Shannon", "Simpson", "InvSimpson", "Chao1", "ACE"))
  alpha_df$Depth <- d
  alpha_df$SampleID <- rownames(alpha_df)
  alpha_list[[as.character(d)]] <- alpha_df
}

# Combine all alpha diversity dataframes
alpha_all <- do.call(rbind, alpha_list)

# Reshape for ggplot (long format)
alpha_long <- alpha_all %>%
  pivot_longer(cols = c("Observed", "Shannon", "Simpson", "InvSimpson", "Chao1", "ACE"),
               names_to = "Metric", values_to = "Value") %>%
  filter(is.finite(Value))

# Function to compute significance letters
get_significance_letters <- function(df) {
  if (length(unique(df$Depth)) < 2 || nrow(df) < 2) return(data.frame())
  dunn_res <- dunnTest(Value ~ factor(Depth), data = df, method = "bh")
  pvals <- dunn_res$res$P.adj
  names(pvals) <- dunn_res$res$Comparison
  letters <- multcompLetters(pvals)$Letters
  depth_levels <- sort(unique(df$Depth))
  label_map <- setNames(letters[1:length(depth_levels)], depth_levels)
  df %>%
    group_by(Depth) %>%
    summarise(ypos = max(Value, na.rm = TRUE) + 0.5) %>%
    mutate(label = label_map[as.character(Depth)],
           Metric = unique(df$Metric)) %>%
    filter(!is.na(label), !is.na(ypos))
}

# Compute annotations for all metrics
annot_all <- alpha_long %>%
  group_by(Metric) %>%
  group_split() %>%
  lapply(get_significance_letters) %>%
  bind_rows()

# Plot with significance letters and sample values
p <- ggplot(alpha_long, aes(x = factor(Depth), y = Value)) +
  geom_boxplot(aes(fill = factor(Depth))) +
  geom_text(data = annot_all, aes(x = factor(Depth), y = ypos, label = label),
            color = "black", size = 4) +
  geom_text(aes(label = round(Value, 1)), position = position_jitter(width = 0.2, height = 0),
            size = 2.5, alpha = 0.7, check_overlap = TRUE) +
  facet_wrap(~ Metric, scales = "free_y") +
  scale_fill_manual(values = my_palette) +
  labs(x = "Sequencing Depth", y = "Alpha Diversity",
       title = "Alpha Diversity Metrics Across Sequencing Depths") +
  theme_minimal() +
  theme(legend.position = "none")

# Save plot
ggsave(
   "/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/alpha_diversity_plot2.tiff",     
  plot = p,
   scale = 1,
 width = 28,
  height = 16,
  units = c("cm"),
 dpi = 600)

# Print lost sample summary
print(lost_samples)

# Print lost sample names per depth
for (depth in names(lost_sample_names)) {
  cat("\nDepth:", depth, "\nLost Samples:", paste(lost_sample_names[[depth]], collapse = ", "), "\n")
}


```


#This analysis focuses exclusively on beta diversity (between-sample diversity) to assess how tomato genotype affects microbial community composition across different Chinese provinces. Three normalization methods are compared: Relative Abundance, CSS, and Hellinger.


#Load the libraries
```{r}
library(phyloseq)
library(vegan)
library(ggplot2)
library(dplyr)
library(purrr)
library(patchwork)
library(metagenomeSeq)
library(knitr)
library(kableExtra)

set.seed(123)
```



#2. Load Data and Select Regions
```{r}
# Load your phyloseq object
# ps2 <- readRDS("your_phyloseq_object.rds")
load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/ps2.RData")

# Check available regions
all_regions <- unique(sample_data(ps2)$Production_site_2);all_regions
cat("Available regions:\n")
print(sort(all_regions))
cat("\n")

# Show sample counts per region
region_summary <- sample_data(ps2) %>%
  data.frame() %>%
  group_by(Production_site_2) %>%
  summarise(n_samples = n(), .groups = "drop") %>%
  arrange(desc(n_samples));region_summary

kable(region_summary, 
      col.names = c("Province/Region", "Number of Samples"),
      caption = "Sample Distribution Across All Regions") %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed"))
```

#2.1 Select Regions to Analyze



```{r}
# OPTION 1: Analyze ALL regions
#regions_to_analyze <- all_regions

# OPTION 2: Select specific regions (uncomment and modify as needed)
 regions_to_analyze <- c("Jiangsu", "Beijing", "Heilongjiang", "Guangxi", "Shaanxi", "Shanxi")

# OPTION 3: Select regions with minimum sample size
# min_samples <- 20
# regions_to_analyze <- region_summary %>%
#   filter(n_samples >= min_samples) %>%
#   pull(Production_site_2)

# OPTION 4: Select by geographic zone
# Northern provinces
# regions_to_analyze <- c("Beijing", "Tianjin", "Heilongjiang", "Liaoning", "Inner Mongolia")

# Eastern provinces
# regions_to_analyze <- c("Jiangsu")

# Southern provinces
# regions_to_analyze <- c("Guangxi", "Yunnan")

cat("Analyzing", length(regions_to_analyze), "regions:\n")
cat(paste(sort(regions_to_analyze), collapse = ", "), "\n\n")

# Subset phyloseq object to selected regions
ps_subset <- subset_samples(ps2, Production_site_2 %in% regions_to_analyze)
ps_subset <- prune_taxa(taxa_sums(ps_subset) > 0, ps_subset)

cat("Filtered dataset:\n")
cat("Total samples:", nsamples(ps_subset), "\n")
cat("Total taxa:", ntaxa(ps_subset), "\n\n")

# Show genotype distribution in selected regions
genotype_summary <- sample_data(ps_subset) %>%
  data.frame() %>%
  group_by(Production_site_2, English_Name_2) %>%
  summarise(n = n(), .groups = "drop") %>%
  arrange(Production_site_2, desc(n));genotype_summary
write.csv(genotype_summary, "genotype_summary.csv", row.names = FALSE)

kable(genotype_summary,
      col.names = c("Province", "Genotype", "Samples"),
      caption = "Genotype Distribution in Selected Regions") %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed"))
```



#3. Normalization Functions

```{r}
# Relative abundance (TSS)
normalize_relative <- function(ps) {
  transform_sample_counts(ps, function(x) x / sum(x))
}

# CSS normalization
normalize_css <- function(ps) {
  mgs <- phyloseq_to_metagenomeSeq(ps)
  p <- cumNormStatFast(mgs)
  mgs <- cumNorm(mgs, p = p)
  norm_counts <- MRcounts(mgs, norm = TRUE)
  otu_table(ps) <- otu_table(norm_counts, taxa_are_rows = TRUE)
  return(ps)
}

# Hellinger transformation
normalize_hellinger <- function(ps) {
  otu_mat <- as(otu_table(ps), "matrix")
  if (taxa_are_rows(ps)) {
    otu_mat <- t(otu_mat)
  }
  otu_hell <- decostand(otu_mat, method = "hellinger")
  if (taxa_are_rows(ps)) {
    otu_hell <- t(otu_hell)
  }
  otu_table(ps) <- otu_table(otu_hell, taxa_are_rows = taxa_are_rows(ps))
  return(ps)
}

```



#4. Beta Diversity Analysis Function

```{r}
analyze_beta_diversity <- function(ps_normalized, region_name, norm_method, metadata) {
  
  # Extract OTU table
  otu_mat <- as(otu_table(ps_normalized), "matrix")
  if (taxa_are_rows(ps_normalized)) {
    otu_mat <- t(otu_mat)
  }
  
  # Calculate Bray-Curtis dissimilarity
  bray_dist <- vegdist(otu_mat, method = "bray")
  
  # PERMANOVA
  permanova_result <- adonis2(bray_dist ~ Genotype, 
                              data = metadata, 
                              permutations = 999)
  
  # PERMDISP (test homogeneity of dispersions)
  disp <- betadisper(bray_dist, metadata$Genotype)
  disp_test <- permutest(disp, permutations = 999)
  
  # PCoA ordination
  pcoa_result <- cmdscale(bray_dist, k = 2, eig = TRUE)
  var_exp <- pcoa_result$eig / sum(pcoa_result$eig) * 100
  
  # Create ordination dataframe
  pcoa_df <- data.frame(
    PC1 = pcoa_result$points[, 1],
    PC2 = pcoa_result$points[, 2],
    Genotype = metadata$Genotype
  )
  
  # Create plot
  p_pcoa <- ggplot(pcoa_df, aes(x = PC1, y = PC2, color = Genotype)) +
    geom_point(size = 3, alpha = 0.7) +
    stat_ellipse(aes(group = Genotype), type = "t", level = 0.95, linewidth = 1) +
    theme_bw() +
    labs(
      title = paste(region_name, "-", toupper(norm_method)),
      subtitle = paste0("PERMANOVA: R² = ", round(permanova_result$R2[1], 3), 
                       ", p = ", round(permanova_result$`Pr(>F)`[1], 3),
                       " | PERMDISP: p = ", round(disp_test$tab$`Pr(>F)`[1], 3)),
      x = paste0("PC1 (", round(var_exp[1], 1), "%)"),
      y = paste0("PC2 (", round(var_exp[2], 1), "%)")
    ) +
    theme(legend.position = "bottom",
          plot.subtitle = element_text(size = 9))
  
  return(list(
    permanova = permanova_result,
    permdisp = disp_test,
    pcoa_plot = p_pcoa,
    distance_matrix = bray_dist
  ))
}

```


#5. Run Beta Diversity Analysis

```{r}
# Define normalization methods
norm_methods <- c("relative", "css", "hellinger")

# Initialize results storage
results_all <- list()

# Loop through each selected region
for (region in regions_to_analyze) {
  
  cat("\n========================================\n")
  cat("ANALYZING:", region, "\n")
  cat("========================================\n")
  
  # Subset by region
  ps_region <- subset_samples(ps_subset, Production_site_2 == region)
  ps_region <- prune_taxa(taxa_sums(ps_region) > 0, ps_region)
  
  n_samples <- nsamples(ps_region)
  n_taxa <- ntaxa(ps_region)
  
  cat("Samples:", n_samples, "\n")
  cat("Taxa:", n_taxa, "\n")
  
  # Get metadata
  metadata <- data.frame(sample_data(ps_region))
  metadata$Genotype <- metadata$English_Name_2
  
  genotype_counts <- table(metadata$Genotype)
  cat("Genotypes:\n")
  print(genotype_counts)
  
  # Check if analysis is possible
  if (length(unique(metadata$Genotype)) < 2) {
    cat("⚠ Skipping: Only 1 genotype in this region\n\n")
    next
  }
  
  if (n_samples < 6) {
    cat("⚠ Warning: Low sample size (n=", n_samples, "). Results may be unreliable.\n")
  }
  
  cat("\n")
  
  # Analyze with each normalization method
  for (norm_method in norm_methods) {
    
    result_key <- paste(region, norm_method, sep = "_")
    
    tryCatch({
      # Apply normalization
      if (norm_method == "css") {
        ps_norm <- normalize_css(ps_region)
      } else if (norm_method == "hellinger") {
        ps_norm <- normalize_hellinger(ps_region)
      } else {
        ps_norm <- normalize_relative(ps_region)
      }
      
      # Analyze beta diversity
      results_all[[result_key]] <- analyze_beta_diversity(
        ps_norm, region, norm_method, metadata
      )
      results_all[[result_key]]$region <- region
      results_all[[result_key]]$normalization <- norm_method
      
      cat("  ✓", norm_method, "complete\n")
      
    }, error = function(e) {
      cat("  ✗ ERROR in", norm_method, ":", e$message, "\n")
    })
  }
  
  cat("\n")
}

cat("========================================\n")
cat("Analysis complete!\n")
cat("========================================\n")

```



#6. Results Summary Table


```{r}
# Extract PERMANOVA results
permanova_summary <- map_df(names(results_all), function(key) {
  result <- results_all[[key]]
  perm <- result$permanova
  disp <- result$permdisp
  
  data.frame(
    Region = result$region,
    Normalization = result$normalization,
    R2 = round(perm$R2[1], 4),
    F_statistic = round(perm$F[1], 3),
    p_value = round(perm$`Pr(>F)`[1], 4),
    PERMDISP_p = round(disp$tab$`Pr(>F)`[1], 4),
    Significant = ifelse(perm$`Pr(>F)`[1] < 0.001, "***",
                        ifelse(perm$`Pr(>F)`[1] < 0.01, "**",
                        ifelse(perm$`Pr(>F)`[1] < 0.05, "*", "ns"))),
    Dispersion_Valid = ifelse(disp$tab$`Pr(>F)`[1] >= 0.05, "✓", "⚠")
  )
});permanova_summary
write.csv(permanova_summary, "permanova_summary.csv", row.names = FALSE)
kable(permanova_summary, 
      caption = "Beta Diversity Results: PERMANOVA Analysis of Genotype Effects",
      col.names = c("Region", "Normalization", "R²", "F", "p-value", 
                    "PERMDISP p", "Sig.", "Valid")) %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed"), 
                full_width = FALSE) %>%
  add_header_above(c(" " = 2, "PERMANOVA" = 3, "Homogeneity Test" = 2, " " = 1)) %>%
  footnote(general = "Significance: *** p<0.001, ** p<0.01, * p<0.05, ns = not significant. Valid (✓) = homogeneous dispersions.",
           general_title = "Note:") %>% save_kable("kable.png")
```

#7. Visualizations
#7.1 Effect Size Comparison


```{r}
# R² comparison across normalizations
mycolor <- c(
  "Normalization1" = "#1b9e77",
  "Normalization2" = "#d95f02",
  "Normalization3" = "#7570b3"
)


p_r2 <- ggplot(permanova_summary, 
               aes(x = reorder(Region, -R2), y = R2, fill = Normalization)) +
  geom_bar(stat = "identity", position = "dodge", alpha = 0.8) +
  geom_text(aes(label = Significant), 
            position = position_dodge(width = 0.9), 
            vjust = -0.5, size = 4) +
  theme_bw() +
  labs(title = "Genotype Effect Size (R²) by Region and Normalization",
       subtitle = "Higher R² indicates stronger genotype effect on community composition",
       y = "R² (Proportion of Variance Explained)",
       x = "Province/Region") +
  scale_fill_brewer(palette = "Set2") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "bottom",
        plot.title = element_text(face = "bold")) +
  ylim(0, max(permanova_summary$R2) * 1.15)

print(p_r2)


#ggsave("alpha_diversity_plot.tiff", plot = p, width = 12, height = 8, dpi = 300)
ggsave(
   "/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/Beta_diversity_regional_specific.tiff",     
  plot = p_r2,
   scale = 1,
 width = 28,
  height = 16,
  units = c("cm"),
 dpi = 600)

```


#7.2 Significance Comparison



```{r}
# p-value comparison
p_sig <- ggplot(permanova_summary, 
                aes(x = reorder(Region, -R2), y = -log10(p_value), 
                    fill = Normalization)) +
  geom_bar(stat = "identity", position = "dodge", alpha = 0.8) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", 
             color = "red", linewidth = 1) +
  geom_hline(yintercept = -log10(0.01), linetype = "dashed", 
             color = "darkred", linewidth = 0.5) +
  theme_bw() +
  labs(title = "Statistical Significance by Region and Normalization",
       subtitle = "Horizontal lines: p = 0.05 (red) and p = 0.01 (dark red)",
       y = "-log10(p-value) [Higher = More Significant]",
       x = "Province/Region") +
  scale_fill_brewer(palette = "Set2") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "bottom",
        plot.title = element_text(face = "bold"))

print(p_sig)

```



#8. PCoA Ordinations

```{r}
#8.1 Relative Abundance {.tabset}
for (region in regions_to_analyze) {
  key <- paste(region, "relative", sep = "_")
  if (key %in% names(results_all)) {
    cat("\n\n###", region, "\n\n")
    
    p <- results_all[[key]]$pcoa_plot +
      scale_color_manual(values = mycola) +
      scale_fill_manual(values = mycola)
    
    # Remove ellipses layers if present
    p$layers <- Filter(function(l) !inherits(l$geom, "GeomPolygon"), p$layers)
    
    print(p)
  }
}

#8.2 CSS Normalization {.tabset}
for (region in regions_to_analyze) {
  key <- paste(region, "css", sep = "_")
  if (key %in% names(results_all)) {
    cat("\n\n###", region, "\n\n")
    
    p <- results_all[[key]]$pcoa_plot +
      scale_color_manual(values = mycola) +
      scale_fill_manual(values = mycola)
    
    p$layers <- Filter(function(l) !inherits(l$geom, "GeomPolygon"), p$layers)
    
    print(p)
  }
}

#8.3 Hellinger Transformation {.tabset}
for (region in regions_to_analyze) {
  key <- paste(region, "hellinger", sep = "_")
  if (key %in% names(results_all)) {
    cat("\n\n###", region, "\n\n")
    
    p <- results_all[[key]]$pcoa_plot +
      scale_color_manual(values = mycola) +
      scale_fill_manual(values = mycola)
    
    p$layers <- Filter(function(l) !inherits(l$geom, "GeomPolygon"), p$layers)
    
    print(p)
  }
}

```






```{r}
library(ggplot2)
library(purrr)

# Helper function to apply custom palette, remove ellipses, and remove grids
process_plot <- function(p) {
  p <- p +
    scale_color_manual(values = mycola) +
    scale_fill_manual(values = mycola) +
    theme_classic() +                       # cleaner background
    theme(
      panel.grid = element_blank(),         # remove all grids
      panel.border = element_blank()        # optional: remove border if desired
    )
  
  # Remove ellipses (GeomPolygon layers)
  p$layers <- Filter(function(l) !inherits(l$geom, "GeomPolygon"), p$layers)
  
  return(p)
}

# Define normalizations
norm_types <- list(
  "Relative_Abundance" = "_relative$",
  "CSS_Normalization" = "_css$",
  "Hellinger_Transformation" = "_hellinger$"
)

# Directory to save plots
output_dir <- "PCoA_individual_plots"
if (!dir.exists(output_dir)) dir.create(output_dir)

# Loop through each normalization
for (norm_name in names(norm_types)) {
  pattern <- norm_types[[norm_name]]
  
  # Get relevant plots
  plots_list <- results_all[grep(pattern, names(results_all))]
  
  if (length(plots_list) > 0) {
    for (region_key in names(plots_list)) {
      # Process plot
      p <- process_plot(plots_list[[region_key]]$pcoa_plot) +
        ggtitle(paste(region_key, "-", norm_name))
      
      # Construct file name
      file_name <- paste0(output_dir, "/", region_key, "_", norm_name, ".png")
      
      # Save plot as PNG with 14 x 12 inches
      ggsave(filename = file_name, plot = p, width = 5, height = 5, dpi = 300)
    }
  }
}


```

#no legends
```{r}
library(ggplot2)
library(purrr)

# Helper function to apply custom palette, remove ellipses, remove grids, and remove legends
process_plot <- function(p) {
  p <- p +
    scale_color_manual(values = mycola) +
    scale_fill_manual(values = mycola) +
    theme_classic() +                       # cleaner background
    theme(
      panel.grid = element_blank(),         # remove all grids
      panel.border = element_blank(),       # optional: remove border
      legend.position = "none"              # remove legend
    )
  
  # Remove ellipses (GeomPolygon layers)
  p$layers <- Filter(function(l) !inherits(l$geom, "GeomPolygon"), p$layers)
  
  return(p)
}

# Define normalizations
norm_types <- list(
  "Relative_Abundance" = "_relative$",
  "CSS_Normalization" = "_css$",
  "Hellinger_Transformation" = "_hellinger$"
)

# Directory to save plots
output_dir <- "PCoA_individual_plots"
if (!dir.exists(output_dir)) dir.create(output_dir)

# Loop through each normalization
for (norm_name in names(norm_types)) {
  pattern <- norm_types[[norm_name]]
  
  # Get relevant plots
  plots_list <- results_all[grep(pattern, names(results_all))]
  
  if (length(plots_list) > 0) {
    for (region_key in names(plots_list)) {
      # Process plot
      p <- process_plot(plots_list[[region_key]]$pcoa_plot) +
        ggtitle(paste(region_key, "-", norm_name))
      
      # Construct file name
      file_name <- paste0(output_dir, "/", region_key, "_", norm_name, ".png")
      
      # Save plot as PNG (adjust width/height as desired)
      ggsave(filename = file_name, plot = p, width = 5, height = 5, dpi = 300)
    }
  }
}

```

#10. Robustness Analysis

```{r}
# Identify regions with consistent effects across normalizations
robust_effects <- permanova_summary %>%
  group_by(Region) %>%
  summarise(
    n_significant = sum(p_value < 0.05),
    all_significant = all(p_value < 0.05),
    mean_R2 = mean(R2),
    sd_R2 = sd(R2),
    consistent_dispersion = all(PERMDISP_p >= 0.05),
    .groups = "drop"
  ) %>%
  arrange(desc(mean_R2))

kable(robust_effects,
      caption = "Robustness of Genotype Effects Across Normalization Methods",
      col.names = c("Region", "# Sig. Methods (of 3)", "All Sig.?", 
                    "Mean R²", "SD R²", "Valid Dispersions?"),
      digits = 4) %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed")) %>%
  footnote(general = "Regions with 'TRUE' in 'All Sig.?' show robust effects. 'Valid Dispersions?' checks PERMDISP assumptions.")
```


#11. Save Results


```{r}
# Create output directory
if (!dir.exists("outputs_beta")) {
  dir.create("outputs_beta")
}

# Save R objects
saveRDS(results_all, "outputs_beta/beta_diversity_results.rds")
write.csv(permanova_summary, "outputs_beta/permanova_summary.csv", row.names = FALSE)
write.csv(robust_effects, "outputs_beta/robustness_summary.csv", row.names = FALSE)

# Save individual distance matrices
for (key in names(results_all)) {
  dist_mat <- as.matrix(results_all[[key]]$distance_matrix)
  write.csv(dist_mat, 
            paste0("outputs_beta/distance_matrix_", key, ".csv"),
            row.names = TRUE)
}

cat("✓ Results saved to 'outputs_beta/' directory\n")
```






---
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/Random_for") #<--- CHANGE ACCORDINGLY

```


# Define prevalence of each taxa; # (in how many samples did each taxa appear at least once)
```{r,echo=FALSE,warning=FALSE}
library(doParallel);packageVersion("doParallel")#‘1.0.17’
library(foreach);packageVersion("foreach")#‘1.5.2’
library(tidyverse);packageVersion("tidyverse")#‘2.0.0’
library(ggplot2);packageVersion("ggplot2")#‘3.5.0’
library(vegan);packageVersion("vegan")#‘2.6.4’
library(reshape2);packageVersion("reshape2")#‘1.4.4’
library(phyloseq);packageVersion("phyloseq")#‘1.46.0’
library(randomForest);packageVersion("randomForest")
library(doBy);packageVersion("doBy")#‘4.7.1.1’
library(plotrix);packageVersion("plotrix")#‘3.8.4’
library(ggpubr);packageVersion("ggpubr")#‘0.6.0’

```



#Training model and Random forest classifications
#x="/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Random_for/Table_rf.txt"

```{r,echo=FALSE,warning=FALSE}
Bacteria_otu <- read.table("Table.txt", header=TRUE,row.names="OTUID");Bacteria_otu
Bacteria_otu <- otu_table(Bacteria_otu, taxa_are_rows = TRUE);Bacteria_otu 

## Imported taxonomy
#Bacteria_tax <- read.csv("Taxonomy.csv",header=TRUE,row.names="OTUID");Bacteria_tax
Bacteria_tax <- read.table("Taxonomy.txt", header=TRUE, sep="\t",row.names="OTUID");Bacteria_tax
Bacteria_tax <- as.matrix(Bacteria_tax);Bacteria_tax
Bacteria_tax <- tax_table(Bacteria_tax);Bacteria_tax

## Imported metadata file
metadata <- read.table("Metadata.txt", header=TRUE, sep="\t",row.names="SampleID");metadata
#metadata <- read.csv("Metadata.csv",header=TRUE,row.names="SampleID");metadata

#rownames(metadata) <- metadata[,1]
metadata <- sample_data(metadata);metadata
#tree
#tree <- read.tree(file = "tree.nwk");tree
#tree <- phy_tree(tree)

all(rownames(metadata) %in% colnames(Bacteria_otu))
#Convert to phyloseq object
Bacteria <- merge_phyloseq(Bacteria_otu,Bacteria_tax, metadata);Bacteria#[ 16460 taxa and 1197 samples ]
Bacteria = subset_taxa(Bacteria, Kingdom=="Bacteria");Bacteria# [ 6448 taxa and 84 samples ]
Bacteria = subset_taxa(Bacteria, Phylum!="CyanoBacteria");Bacteria#  [ 15588 taxa and 1197 samples ]
Bacteria = subset_taxa(Bacteria, Order!="Chloroplast");Bacteria#  [ 15588 taxa and 1197 samples ]
Bacteria = subset_taxa(Bacteria, Family!="Mitochondria");Bacteria# [ 15588 taxa and 1197 samples ]
Bacteria <- prune_taxa(taxa_sums(Bacteria)>0, Bacteria);Bacteria# [ 15588 taxa and 1197 samples ]
#OTU_table_Bacteria = as(otu_table(Bacteria), "matrix")
#########################################################################################################
#Summarize 
microbiome::summarize_phyloseq(Bacteria)
#Number of singletons = 68"
#Sparsity = 0.995179763625127"
#Average number of reads = 22794.9197994987"
#Total number of reads = 27285519"
#Max. number of reads = 225892"
#Min. number of reads = 68"
sums_Bacteria <- sample_sums(Bacteria);sums_Bacteria
sums_Bacteria <- data.frame(sums_Bacteria);sums_Bacteria
sums_Bacteria <- as.matrix(sums_Bacteria);sums_Bacteria
colSums(sums_Bacteria) #  27281011      
max(sums_Bacteria) #225892
min(sums_Bacteria) # 68
sums_Bacteria
#https://david-barnett.github.io/microViz/articles/web-only/tax-fixing.html #fixing the table
#https://rpubs.com/bioguo/680490 (Random forest classifications)
library(devtools);packageVersion("devtools")# ‘2.4.5’# Load the devtools package
#install_github("guokai8/microbial") # Install the package (Microbial package)
library(microbial);packageVersion("microbial")#‘0.0.22’####
ps=Bacteria
save(ps, file = "/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/Random_for/ps.RData")
```


#Genotype

```{r}
load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/Random_for/ps.RData")

# Load required libraries
library(randomForest)
library(caret)
library(reshape2)
library(ggplot2)

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps)/nsamples(ps) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps)

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$English_Name_2);response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.9
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```


#Production_site_2
```{r}

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps)/nsamples(ps) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps)

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$Production_site_2);response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.816 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```




#Fruit_color
```{r}
ps_Fruit_color <- subset_samples(ps, !is.na(Fruit_color));ps_Fruit_color

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps_Fruit_color = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps_Fruit_color <- taxa_sums(ps_Fruit_color)/nsamples(ps_Fruit_color) # average number of reads for each otu
sites.prune.ps_Fruit_color <- prune_taxa(tax.mean.ps_Fruit_color > prunescale.ps_Fruit_color, ps_Fruit_color)

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps_Fruit_color<- t(otu_table(sites.prune.ps_Fruit_color));predictors.ps_Fruit_color
dim(predictors.ps_Fruit_color)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps_Fruit_color <- as.factor(sample_data(sites.prune.ps_Fruit_color)$Fruit_color);response.ps_Fruit_color
# Combine them into 1 data frame
rf.data.ps_Fruit_color <- data.frame(response.ps_Fruit_color, predictors.ps_Fruit_color);rf.data.ps_Fruit_color
str(rf.data.ps_Fruit_color)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps_Fruit_color$response.ps_Fruit_color, p = 0.7, list = FALSE)
trainData <- rf.data.ps_Fruit_color[trainIndex, ]
testData <- rf.data.ps_Fruit_color[-trainIndex, ]

# Train model
ps_Fruit_color.classify <- randomForest(response.ps_Fruit_color ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps_Fruit_color.classify

# Predict on test set
predicted.labels <- predict(ps_Fruit_color.classify, testData)
true.labels <- testData$response.ps_Fruit_color

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.773 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```




#Fruit_shape
```{r}
ps_Fruit_shape <- subset_samples(ps, !is.na(Fruit_shape));ps_Fruit_shape

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_Fruit_shape)/nsamples(ps_Fruit_shape) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_Fruit_shape)

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$Fruit_shape);response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.841 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```




#Fruit_shape
```{r}
ps_Fruit_shape <- subset_samples(ps, !is.na(Fruit_shape));ps_Fruit_shape

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_Fruit_shape)/nsamples(ps_Fruit_shape) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_Fruit_shape)

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$Fruit_shape);response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.841 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```


##TMV_resistance 
```{r}
ps_TMV_resistance  <- subset_samples(ps, !is.na(TMV_resistance ));ps_TMV_resistance 

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_TMV_resistance )/nsamples(ps_TMV_resistance ) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_TMV_resistance )

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$TMV_resistance );response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.827 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```




##Insect_resistance 
```{r}
ps_Insect_resistance  <- subset_samples(ps, !is.na(Insect_resistance ));ps_Insect_resistance 

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_Insect_resistance )/nsamples(ps_Insect_resistance ) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_Insect_resistance )

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$Insect_resistance );response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy# 0.819 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```




##yield 
```{r}
ps_yield <- subset_samples(ps, !is.na(yield));ps_yield

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_yield )/nsamples(ps_yield ) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_yield )

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$yield );response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.798 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```




##seedweight_1000 
```{r}
ps_seedweight_1000 <- subset_samples(ps, !is.na(seedweight_1000));ps_seedweight_1000

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_seedweight_1000 )/nsamples(ps_seedweight_1000 ) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_seedweight_1000 )

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$seedweight_1000 );response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#  0.87 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```





##ovaries 
```{r}
ps_ovaries <- subset_samples(ps, !is.na(ovaries));ps_ovaries

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_ovaries )/nsamples(ps_ovaries ) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_ovaries )

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$ovaries );response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy# 0.85 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```




##Fruit_long_diameter 
```{r}
ps_Fruit_long_diameter <- subset_samples(ps, !is.na(Fruit_long_diameter));ps_Fruit_long_diameter

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_Fruit_long_diameter )/nsamples(ps_Fruit_long_diameter ) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_Fruit_long_diameter )

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$Fruit_long_diameter );response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.864 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```





##Fruit_hori_diameter 
```{r}
ps_Fruit_hori_diameter <- subset_samples(ps, !is.na(Fruit_hori_diameter));ps_Fruit_hori_diameter

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_Fruit_hori_diameter )/nsamples(ps_Fruit_hori_diameter ) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_Fruit_hori_diameter )

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$Fruit_hori_diameter );response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy# 0.802 
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```





##Fruit_taste 
```{r}
ps_Fruit_taste <- subset_samples(ps, Fruit_taste %in% c("Sweet", "Sour"));ps_Fruit_taste#[ 15588 taxa and 1102 samples ]

# All OTUs that do not average > prunescale across samples will be dropped.
prunescale.ps = 0.05

# Prune out rare OTUs by mean relative abundance set by prunescale
tax.mean.ps <- taxa_sums(ps_Fruit_taste )/nsamples(ps_Fruit_taste ) # average number of reads for each otu
sites.prune.ps <- prune_taxa(tax.mean.ps > prunescale.ps, ps_Fruit_taste )

#Next, we are going to format a dataframe of predictors (OTUs) and responses (States)
# Make a dataframe of training data with OTUs as column and samples as rows
predictors.ps<- t(otu_table(sites.prune.ps));predictors.ps
dim(predictors.ps)
# 1197 1655

# Make one column for our outcome/response variable 
response.ps <- as.factor(sample_data(sites.prune.ps)$Fruit_taste );response.ps
# Combine them into 1 data frame
rf.data.ps <- data.frame(response.ps, predictors.ps);rf.data.ps
str(rf.data.ps)

# Split data into training and test sets
set.seed(579383)
trainIndex <- createDataPartition(rf.data.ps$response.ps, p = 0.7, list = FALSE)
trainData <- rf.data.ps[trainIndex, ]
testData <- rf.data.ps[-trainIndex, ]

# Train model
ps.classify <- randomForest(response.ps ~ ., data = trainData, ntree = 500, proximity = TRUE, mtry = 200);ps.classify

# Predict on test set
predicted.labels <- predict(ps.classify, testData)
true.labels <- testData$response.ps

# Compute confusion matrix
conf.mat <- confusionMatrix(predicted.labels, true.labels)
print(conf.mat)

# Extract matrix and accuracy
cm <- conf.mat$table
accuracy <- round(conf.mat$overall['Accuracy'], 3);accuracy#0.839
train_size <- nrow(trainData);train_size
test_size <- nrow(testData);test_size

# Visualize confusion matrix
Data_frame <- reshape2::melt(cm)
colnames(Data_frame) <- c("Actual", "Predicted", "Count")

ggplot(Data_frame, aes(x = Predicted, y = Actual, fill = Count)) +
  geom_tile(color = "white") +
  geom_text(aes(label = Count), size = 6) +
  scale_fill_gradient(low = "lightblue", high = "steelblue") +
  labs(title = paste("Confusion Matrix (Accuracy:", accuracy,
                     "| Train:", train_size, "Test:", test_size, ")"),
       x = "Predicted Label", y = "Actual Label") +
  theme_minimal(base_size = 10)
```














---
title: "Beta Diversity Analysis using Jaccard Distance"
author: "Microbiome Analysis"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
    code_folding: show
    theme: flatly
    highlight: tango
    fig_width: 12
    fig_height: 8
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center",
  cache = FALSE
)
```

# Introduction

This report demonstrates **beta diversity analysis** using **Jaccard distance** on presence-absence transformed microbiome data. Jaccard distance is particularly appropriate for:

- Presence-absence (binary) data
- Focus on shared species between samples
- Community composition comparisons
- Handling sparse microbiome datasets

---

# Setup and Load Data

## Load Required Libraries

```{r libraries}
# Load required packages
library(phyloseq)
library(ggplot2)
library(vegan)
library(gridExtra)
library(pheatmap)
library(RColorBrewer)
library(dendextend)
library(knitr)
library(dplyr)
library(tidyr)

# Set random seed for reproducibility
set.seed(42)
```

## Load Phyloseq Object

```{r load-data}

# Assuming ps2 is already loaded in your environment
# If not, load it:
# ps2 <- readRDS("path/to/ps2.rds")
load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/ps2.RData")

cat("Phyloseq Object: ps2\n")
print(ps2)

# Dataset summary
cat("\nDataset Summary:\n")
cat("Number of samples:", nsamples(ps2), "\n")
cat("Number of OTUs/ASVs:", ntaxa(ps2), "\n")
cat("Total sequences:", sum(otu_table(ps2)), "\n")
cat("Sparsity:", round(sum(otu_table(ps2) == 0) / length(otu_table(ps2)) * 100, 2), "%\n")
```

## Check Sample Metadata

```{r metadata-check}
# View sample variables
cat("Sample variables available:\n")
print(sample_variables(ps2))

# Preview metadata
cat("\nSample metadata (first 10 samples):\n")
metadata <- as(sample_data(ps2), "data.frame")
kable(head(metadata, 10))

# Check for English_Name_2 column
if ("English_Name_2" %in% colnames(metadata)) {
  cat("\n✓ English_Name_2 column found\n")
  cat("Unique groups:", length(unique(metadata$English_Name_2)), "\n")
  cat("Group counts:\n")
  print(table(metadata$English_Name_2))
} else {
  cat("\n⚠ English_Name_2 not found. Available columns:\n")
  print(colnames(metadata))
}
```

## Setup Color Palette

```{r color-palette}
# Define or load your custom color palette
# If you have mycola already defined, use it
# Otherwise, create one:

n_groups <- length(unique(metadata$English_Name_2))

# Option 1: Use your existing mycola
# mycola <- your_predefined_colors

# Option 2: Create extended palette if needed
if (exists("mycola") && length(mycola) >= n_groups) {
  cat("Using existing mycola palette\n")
} else {
  # Create a color palette with enough colors
  library(RColorBrewer)
  
  # Get all available colors from multiple palettes
  qual_col_pals <- brewer.pal.info[brewer.pal.info$category == 'qual',]
  mycola <- unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
  
  # If still not enough, extend with colorRampPalette
  if (length(mycola) < n_groups) {
    mycola <- colorRampPalette(brewer.pal(12, "Set3"))(n_groups)
  }
  
  cat("Created color palette with", length(mycola), "colors\n")
}

# Show color palette
barplot(rep(1, min(20, length(mycola))), col = mycola[1:min(20, length(mycola))], 
        main = "Color Palette (first 20 colors)", axes = FALSE)
```

---

# Presence-Absence Transformation

## Convert to Binary Data

```{r pa-transformation}
# Transform abundance data to presence-absence (PA)
ps2_pa <- transform_sample_counts(ps2, function(x) ifelse(x > 0, 1, 0))

cat("Presence-Absence Transformation Complete\n")
cat("Original data type: abundance counts\n")
cat("Transformed data type: binary (0/1)\n")

# Compare before and after for one OTU
example_otu <- taxa_names(ps2)[1]
comparison <- data.frame(
  Sample = sample_names(ps2)[1:10],
  Original = as.vector(otu_table(ps2)[example_otu, 1:10]),
  PA = as.vector(otu_table(ps2_pa)[example_otu, 1:10])
)

kable(comparison, caption = paste("Example transformation for", example_otu))
```

## Calculate OTU Richness

```{r richness-calculation}
# Calculate richness (number of present OTUs per sample)
richness_df <- data.frame(
  Sample = sample_names(ps2_pa),
  Richness = sample_sums(ps2_pa),
  English_Name_2 = metadata$English_Name_2
)

# Summary by group
richness_summary <- richness_df %>%
  group_by(English_Name_2) %>%
  summarise(
    N = n(),
    Mean = round(mean(Richness), 2),
    SD = round(sd(Richness), 2),
    Min = min(Richness),
    Max = max(Richness)
  ) %>%
  arrange(desc(Mean))

kable(richness_summary, caption = "OTU Richness Summary by Group")
```


```{r}
PA = ggplot(richness_df, aes(
  x = reorder(English_Name_2, Richness, median),
  y = Richness,
  fill = English_Name_2
)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(width = 0.2, alpha = 0.4, size = 1) +
  scale_fill_manual(values = mycola) +
  theme_bw() +
  labs(
    title = "OTU Richness Distribution by Group",
    x = "Group",
    y = "Number of Present OTUs"
  ) +
  theme(
    plot.title = element_text(face = "bold", size = 14),
    legend.position = "none",
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 7),
    axis.text.y = element_text(size = 7)
  )

ggsave(
  "/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/PA.tiff",
  plot = PA,
  scale = 1,
  width = 18,
  height = 9,
  units = "cm",
  dpi = 600
)

```


---

# Calculate Jaccard Distance

## Distance Matrix Calculation

```{r jaccard-calculation}
library(phyloseq)
getAnywhere(distance)

# Calculate Jaccard distance
# Jaccard = 1 - (intersection / union) for binary data
#jaccard_dist <- distance(ps2_pa, method = "jaccard", binary = TRUE)
library(phyloseq)
jaccard_dist <- phyloseq::distance(ps2_pa, method = "jaccard")

cat("Jaccard Distance Matrix Calculated\n")
cat("Matrix dimensions:", attr(jaccard_dist, "Size"), "x", attr(jaccard_dist, "Size"), "\n")
cat("Distance range:", round(min(jaccard_dist), 3), "-", round(max(jaccard_dist), 3), "\n")
cat("Mean distance:", round(mean(jaccard_dist), 3), "\n")
```

## Distance Matrix Preview

```{r distance-preview}
# Show subset of distance matrix
dist_matrix <- as.matrix(jaccard_dist)
kable(dist_matrix[1:8, 1:8], 
      caption = "Jaccard Distance Matrix (first 8x8 samples)",
      digits = 3)
```

## Distance Distribution

```{r distance-distribution, fig.height=5}
# Plot distribution of distances
dist_df <- data.frame(Distance = as.vector(jaccard_dist))

ggplot(dist_df, aes(x = Distance)) +
  geom_histogram(bins = 50, fill = "steelblue", color = "black", alpha = 0.7) +
  geom_vline(xintercept = mean(jaccard_dist), 
             color = "red", linetype = "dashed", size = 1) +
  annotate("text", x = mean(jaccard_dist), y = Inf, 
           label = paste("Mean =", round(mean(jaccard_dist), 3)),
           vjust = 1.5, hjust = -0.1, color = "red") +
  theme_bw() +
  labs(title = "Distribution of Jaccard Distances",
       x = "Jaccard Distance",
       y = "Frequency") +
  theme(plot.title = element_text(face = "bold", size = 14))
```

---

# Ordination Analysis

## Principal Coordinates Analysis (PCoA)

```{r pcoa}
# Perform PCoA
ord_pcoa <- ordinate(ps2_pa, method = "PCoA", distance = jaccard_dist)

# Extract variance explained
eig <- ord_pcoa$values$Eigenvalues
var_exp <- round(eig / sum(eig) * 100, 2)

cat("PCoA Analysis Complete\n")
cat("Variance explained by PC1:", var_exp[1], "%\n")
cat("Variance explained by PC2:", var_exp[2], "%\n")
cat("Cumulative variance (PC1+PC2):", sum(var_exp[1:2]), "%\n")
```

```{r pcoa-plot, fig.height=8}
# Plot PCoA
p_pcoa <- plot_ordination(ps2_pa, ord_pcoa, 
                          color = "English_Name_2",
                          title = "PCoA - Jaccard Distance") +
  geom_point(size = 3, alpha = 0.7) +
  scale_color_manual(values = mycola) +
  theme_bw() +
  labs(color = "Group") +
  theme(plot.title = element_text(face = "bold", size = 14),
        legend.position = "right",
        legend.text = element_text(size = 8)) +
  guides(color = guide_legend(ncol =1))

print(p_pcoa)

ggsave(
  "/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/p_pcoa_jaccard.tiff",
  plot = p_pcoa,
  scale = 1,
  width = 20,
  height = 16,
  units = "cm",
  dpi = 600
)
```




```{r pcoa-plot, fig.height=8}
# Plot PCoA
p_pcoa <- plot_ordination(ps2_pa, ord_pcoa, 
                          color = "Production_site_2",
                          title = "PCoA - Jaccard Distance") +
  geom_point(size = 3, alpha = 0.7) +
  scale_color_manual(values = mycola) +
  theme_bw() +
  labs(color = "Group") +
  theme(plot.title = element_text(face = "bold", size = 14),
        legend.position = "right",
        legend.text = element_text(size = 8)) +
  guides(color = guide_legend(ncol =1))

print(p_pcoa)

ggsave(
  "/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/Production_site_2_jaccard.tiff",
  plot = p_pcoa,
  scale = 1,
  width = 20,
  height = 16,
  units = "cm",
  dpi = 600
)
```





```{r pcoa-ellipse, fig.height=8}
# PCoA with confidence ellipses
p_pcoa_ellipse <- plot_ordination(ps2_pa, ord_pcoa, 
                                   color = "English_Name_2") +
  geom_point(size = 3, alpha = 0.7) +
  stat_ellipse(type = "norm", linetype = 2, size = 0.8) +
  scale_color_manual(values = mycola) +
  theme_bw() +
  labs(title = "PCoA with 95% Confidence Ellipses",
       color = "Group") +
  theme(plot.title = element_text(face = "bold", size = 14),
        legend.position = "right",
        legend.text = element_text(size = 8)) +
  guides(color = guide_legend(ncol = 1))

print(p_pcoa_ellipse)
```

## Non-metric Multidimensional Scaling (NMDS)

```{r nmds}
# Perform NMDS
set.seed(42)
ord_nmds <- ordinate(ps2_pa, method = "NMDS", distance = jaccard_dist)

cat("NMDS Analysis Complete\n")
cat("NMDS Stress:", round(ord_nmds$stress, 4), "\n")
cat("Interpretation:\n")
if (ord_nmds$stress < 0.05) {
  cat("  Excellent representation (stress < 0.05)\n")
} else if (ord_nmds$stress < 0.1) {
  cat("  Good representation (stress < 0.1)\n")
} else if (ord_nmds$stress < 0.2) {
  cat("  Acceptable representation (stress < 0.2)\n")
} else {
  cat("  Poor representation (stress > 0.2) - interpret with caution\n")
}
```

```{r nmds-plot, fig.height=8}
# Plot NMDS
p_nmds <- plot_ordination(ps2_pa, ord_nmds, 
                          color = "English_Name_2",
                          title = paste0("NMDS - Jaccard Distance (Stress = ", 
                                        round(ord_nmds$stress, 3), ")")) +
  geom_point(size = 3, alpha = 0.7) +
  stat_ellipse(type = "norm", linetype = 2, size = 0.8) +
  scale_color_manual(values = mycola) +
  theme_bw() +
  labs(color = "Group") +
  theme(plot.title = element_text(face = "bold", size = 14),
        legend.position = "right",
        legend.text = element_text(size = 8)) +
  guides(color = guide_legend(ncol = 1))

print(p_nmds)
```

---

# Statistical Testing

## PERMANOVA

```{r permanova}
# PERMANOVA - test for differences between groups
set.seed(42)
permanova_result <- adonis2(jaccard_dist ~ English_Name_2, 
                            data = metadata,
                            permutations = 999)

cat("PERMANOVA Results\n")
cat("=" , rep("=", 50), "\n", sep = "")
print(permanova_result)

# Extract key statistics
r2 <- round(permanova_result$R2[1], 4)
pval <- permanova_result$`Pr(>F)`[1]

cat("\nInterpretation:\n")
cat("R² =", r2, "- proportion of variance explained by groups\n")
cat("P-value =", pval, "\n")
if (pval < 0.001) {
  cat("*** Highly significant difference between groups (P < 0.001)\n")
} else if (pval < 0.01) {
  cat("** Significant difference between groups (P < 0.01)\n")
} else if (pval < 0.05) {
  cat("* Significant difference between groups (P < 0.05)\n")
} else {
  cat("No significant difference between groups (P ≥ 0.05)\n")
}
```


## PERMANOVA

```{r permanova}
# PERMANOVA - test for differences between groups
set.seed(42)
permanova_result <- adonis2(jaccard_dist ~ Production_site_2, 
                            data = metadata,
                            permutations = 999)

cat("PERMANOVA Results\n")
cat("=" , rep("=", 50), "\n", sep = "")
print(permanova_result)

# Extract key statistics
r2 <- round(permanova_result$R2[1], 4)
pval <- permanova_result$`Pr(>F)`[1]

cat("\nInterpretation:\n")
cat("R² =", r2, "- proportion of variance explained by groups\n")
cat("P-value =", pval, "\n")
if (pval < 0.001) {
  cat("*** Highly significant difference between groups (P < 0.001)\n")
} else if (pval < 0.01) {
  cat("** Significant difference between groups (P < 0.01)\n")
} else if (pval < 0.05) {
  cat("* Significant difference between groups (P < 0.05)\n")
} else {
  cat("No significant difference between groups (P ≥ 0.05)\n")
}
```
## Test for Homogeneity of Dispersion (BETADISPER)

```{r betadisper}
# BETADISPER - test assumption of homogeneous dispersion
beta_disp <- betadisper(jaccard_dist, metadata$English_Name_2)

# Permutation test
set.seed(42)
disp_test <- permutest(beta_disp, permutations = 999)

cat("BETADISPER Results\n")
cat("=" , rep("=", 50), "\n", sep = "")
print(disp_test)

cat("\nInterpretation:\n")
if (disp_test$tab$`Pr(>F)`[1] < 0.05) {
  cat("⚠ Warning: Significant difference in dispersion between groups (P < 0.05)\n")
  cat("PERMANOVA results should be interpreted with caution.\n")
  cat("Differences may be due to dispersion rather than location.\n")
} else {
  cat("✓ No significant difference in dispersion (P ≥ 0.05)\n")
  cat("PERMANOVA assumption satisfied.\n")
}
```


```{r betadisper-plot, fig.height=10}
# Visualize dispersion
par(mfrow = c(2, 1), mar = c(5, 4, 4, 2))

# Plot 1: PCoA of distances to centroids
plot(beta_disp, main = "PCoA of Distances to Group Centroids",
     hull = FALSE, ellipse = TRUE, conf = 0.95,
     segments = TRUE, col = mycola[1:length(unique(metadata$English_Name_2))])

# Plot 2: Boxplot of distances to centroids
boxplot(beta_disp, 
        col = mycola[1:length(unique(metadata$English_Name_2))],
        main = "Distance to Centroid by Group",
        ylab = "Distance to Centroid",
        las = 2,
        cex.axis = 0.7)
```

---

# Pairwise Comparisons

```{r pairwise-permanova}
# Pairwise PERMANOVA between groups
groups <- unique(metadata$English_Name_2)
n_groups <- length(groups)

if (n_groups > 10) {
  cat("Note: Only showing comparisons for first 10 groups due to computational time\n")
  groups <- groups[1:10]
}

pairwise_results <- data.frame()

cat("Running pairwise PERMANOVA...\n")
pb <- txtProgressBar(min = 0, max = choose(length(groups), 2), style = 3)
counter <- 0

for (i in 1:(length(groups)-1)) {
  for (j in (i+1):length(groups)) {
    counter <- counter + 1
    setTxtProgressBar(pb, counter)
    
    # Subset samples
    group_subset <- metadata$English_Name_2 %in% c(groups[i], groups[j])
    samples_subset <- rownames(metadata)[group_subset]
    
    # Subset distance matrix
    dist_subset <- as.dist(as.matrix(jaccard_dist)[samples_subset, samples_subset])
    meta_subset <- metadata[group_subset, , drop = FALSE]
    
    # Run PERMANOVA
    set.seed(42)
    perm_test <- adonis2(dist_subset ~ English_Name_2, 
                         data = meta_subset, 
                         permutations = 999)
    
    pairwise_results <- rbind(pairwise_results, 
                              data.frame(
                                Group1 = groups[i],
                                Group2 = groups[j],
                                R2 = round(perm_test$R2[1], 4),
                                F_stat = round(perm_test$F[1], 4),
                                P_value = perm_test$`Pr(>F)`[1]
                              ))
  }
}
close(pb)

# Adjust p-values for multiple testing
pairwise_results$P_adjusted <- p.adjust(pairwise_results$P_value, method = "BH")

# Sort by p-value
pairwise_results <- pairwise_results %>% arrange(P_value)

cat("\nPairwise PERMANOVA Complete\n")
cat("Total comparisons:", nrow(pairwise_results), "\n")
cat("Significant (P < 0.05):", sum(pairwise_results$P_value < 0.05), "\n")
cat("Significant after adjustment (Padj < 0.05):", sum(pairwise_results$P_adjusted < 0.05), "\n")
```

```{r pairwise-table}
# Show top pairwise comparisons
kable(head(pairwise_results, 20), 
      caption = "Top 20 Pairwise Group Comparisons (sorted by P-value)",
      digits = 4)
```

---

# Distance Analysis by Group

## Within vs Between Group Distances

```{r group-distances}
# Calculate within and between group distances
calculate_group_distances <- function(dist_matrix, groups) {
  dist_mat <- as.matrix(dist_matrix)
  results <- data.frame()
  
  # Within-group distances
  for (group in unique(groups)) {
    group_samples <- which(groups == group)
    if (length(group_samples) > 1) {
      within_dist <- dist_mat[group_samples, group_samples]
      within_dist <- within_dist[lower.tri(within_dist)]
      
      results <- rbind(results, data.frame(
        Group = group,
        Type = "Within",
        N_comparisons = length(within_dist),
        Mean = mean(within_dist),
        Median = median(within_dist),
        SD = sd(within_dist),
        Min = min(within_dist),
        Max = max(within_dist)
      ))
    }
  }
  
  # Between-group distances
  group_levels <- unique(groups)
  for (i in 1:(length(group_levels)-1)) {
    for (j in (i+1):length(group_levels)) {
      group1 <- group_levels[i]
      group2 <- group_levels[j]
      
      samples1 <- which(groups == group1)
      samples2 <- which(groups == group2)
      
      between_dist <- as.vector(dist_mat[samples1, samples2])
      
      results <- rbind(results, data.frame(
        Group = paste(group1, "vs", group2),
        Type = "Between",
        N_comparisons = length(between_dist),
        Mean = mean(between_dist),
        Median = median(between_dist),
        SD = sd(between_dist),
        Min = min(between_dist),
        Max = max(between_dist)
      ))
    }
  }
  
  return(results)
}

# Calculate distances (limit to first 10 groups if too many)
if (n_groups > 10) {
  groups_subset <- unique(metadata$English_Name_2)[1:10]
  metadata_subset <- metadata[metadata$English_Name_2 %in% groups_subset, ]
  samples_subset <- rownames(metadata_subset)
  dist_subset <- as.dist(as.matrix(jaccard_dist)[samples_subset, samples_subset])
  
  group_distances <- calculate_group_distances(dist_subset, metadata_subset$English_Name_2)
  cat("Note: Analysis limited to first 10 groups\n")
} else {
  group_distances <- calculate_group_distances(jaccard_dist, metadata$English_Name_2)
}

# Show summary
kable(head(group_distances %>% arrange(Mean), 20), 
      caption = "Within and Between Group Distances (Top 20 by Mean Distance)",
      digits = 4)
```

```{r distance-plot, fig.height=8}
# Plot within vs between distances
within_data <- group_distances %>% filter(Type == "Within")
between_data <- group_distances %>% filter(Type == "Between") %>% head(20)

p1 <- ggplot(within_data, aes(x = reorder(Group, Mean), y = Mean, fill = Group)) +
  geom_bar(stat = "identity", alpha = 0.7) +
  geom_errorbar(aes(ymin = Mean - SD, ymax = Mean + SD), width = 0.2) +
  scale_fill_manual(values = mycola) +
  coord_flip() +
  theme_bw() +
  labs(title = "Within-Group Mean Jaccard Distance",
       x = "Group",
       y = "Mean Distance") +
  theme(plot.title = element_text(face = "bold", size = 12),
        legend.position = "none",
        axis.text.y = element_text(size = 8))

print(p1)
```

---

# Hierarchical Clustering

## Dendrogram

```{r hclust, fig.height=10}
# Perform hierarchical clustering
hclust_result <- hclust(jaccard_dist, method = "average")

# Convert to dendrogram
dend <- as.dendrogram(hclust_result)

# Color by groups
group_colors <- mycola[as.numeric(as.factor(metadata$English_Name_2))]
labels_colors(dend) <- group_colors[order.dendrogram(dend)]

# Plot
par(mar = c(10, 4, 4, 2))
plot(dend, 
     main = "Hierarchical Clustering (Jaccard Distance)",
     ylab = "Jaccard Distance",
     cex = 0.4,
     cex.main = 1.5)
```

## Distance Matrix Heatmap

```{r heatmap, fig.height=12, fig.width=12}
# Prepare annotations
annotation_df <- data.frame(
  Group = metadata$English_Name_2,
  row.names = rownames(metadata)
)

# Limit colors if too many groups
n_unique_groups <- length(unique(metadata$English_Name_2))
annotation_colors <- list(
  Group = setNames(mycola[1:n_unique_groups], 
                   unique(metadata$English_Name_2))
)

# Create heatmap
pheatmap(as.matrix(jaccard_dist),
         clustering_distance_rows = jaccard_dist,
         clustering_distance_cols = jaccard_dist,
         annotation_row = annotation_df,
         annotation_col = annotation_df,
         annotation_colors = annotation_colors,
         color = colorRampPalette(c("blue", "white", "red"))(100),
         main = "Jaccard Distance Heatmap",
         show_rownames = FALSE,
         show_colnames = FALSE,
         fontsize = 10)
```

---

# Export Results

```{r export}
# Create output directory
dir.create("beta_diversity_results", showWarnings = FALSE)

# Export distance matrix
write.csv(as.matrix(jaccard_dist), 
          "beta_diversity_results/jaccard_distance_matrix.csv",
          row.names = TRUE)

# Export ordination coordinates
write.csv(ord_pcoa$vectors, 
          "beta_diversity_results/pcoa_coordinates.csv",
          row.names = TRUE)

write.csv(ord_nmds$points, 
          "beta_diversity_results/nmds_coordinates.csv",
          row.names = TRUE)

# Export PERMANOVA results
write.csv(as.data.frame(permanova_result), 
          "beta_diversity_results/permanova_results.csv",
          row.names = TRUE)

# Export pairwise results
write.csv(pairwise_results, 
          "beta_diversity_results/pairwise_permanova_results.csv",
          row.names = FALSE)

# Export group distances
write.csv(group_distances, 
          "beta_diversity_results/group_distances.csv",
          row.names = FALSE)

cat("\n✓ Results exported to 'beta_diversity_results/' directory\n")
cat("\nFiles created:\n")
cat("  - jaccard_distance_matrix.csv\n")
cat("  - pcoa_coordinates.csv\n")
cat("  - nmds_coordinates.csv\n")
cat("  - permanova_results.csv\n")
cat("  - pairwise_permanova_results.csv\n")
cat("  - group_distances.csv\n")
```

---

# Summary and Conclusions

```{r summary}
cat("BETA DIVERSITY ANALYSIS SUMMARY\n")
cat(rep("=", 60), "\n", sep = "")
cat("Dataset:", deparse(substitute(ps2)), "\n")
cat("Number of samples:", nsamples(ps2_pa), "\n")
cat("Number of OTUs/ASVs:", ntaxa(ps2_pa), "\n")
cat("Number of groups:", length(unique(metadata$English_Name_2)), "\n\n")

cat("ORDINATION RESULTS:\n")
cat("PCoA variance explained (PC1+PC2):", sum(var_exp[1:2]), "%\n")
cat("NMDS stress:", round(ord_nmds$stress, 4), "\n\n")

cat("STATISTICAL TESTING:\n")
cat("PERMANOVA R²:", round(permanova_result$R2[1], 4), "\n")
cat("PERMANOVA P-value:", permanova_result$`Pr(>F)`[1], "\n")
cat("Betadisper P-value:", disp_test$tab$`Pr(>F)`[1], "\n\n")

cat("PAIRWISE COMPARISONS:\n")
cat("Total pairwise tests:", nrow(pairwise_results), "\n")
cat("Significant (P < 0.05):", sum(pairwise_results$P_value < 0.05), "\n")
cat("Significant after FDR adjustment:", sum(pairwise_results$P_adjusted < 0.05), "\n")
```

## Key Findings

**Jaccard Distance Analysis:**

- Jaccard distance measures dissimilarity based on presence-absence of OTUs
- Values range from 0 (identical communities) to 1 (completely different)
- Mean Jaccard distance: `r round(mean(jaccard_dist), 3)`

**Group Differentiation:**

```{r key-findings}
if (permanova_result$`Pr(>F)`[1] < 0.05) {
  cat("✓ Significant differences detected between groups (PERMANOVA P < 0.05)\n")
  cat("✓ Group membership explains", round(permanova_result$R2[1] * 100, 2), 
      "% of variance in community composition\n")
} else {
  cat("✗ No significant differences detected between groups (PERMANOVA P ≥ 0.05)\n")
}

if (disp_test$tab$`Pr(>F)`[1] < 0.05) {
  cat("⚠ Groups show different levels of dispersion (betadisper P < 0.05)\n")
  cat("  Interpret PERMANOVA results with caution\n")
} else {
  cat("✓ Groups show similar dispersion (betadisper P ≥ 0.05)\n")
  cat("  PERMANOVA results are reliable\n")
}
```

---

# Session Information

```{r session-info}
sessionInfo()
```

---

# References

1. Jaccard, P. (1912). The distribution of the flora in the alpine zone. *New Phytologist*, 11(2), 37-50.
2. Anderson, M. J. (2001). A new method for non-parametric multivariate analysis of variance. *Austral Ecology*, 26(1), 32-46.
3. Oksanen, J., et al. (2020). vegan: Community Ecology Package. R package version 2.5-7.
4. McMurdie, P. J., & Holmes, S. (2013). phyloseq: An R package for reproducible interactive analysis and graphics of microbiome census data. *PLoS ONE*, 8(4), e61217.



###plot rarefaction curves 
#code based on https://microsud.github.io/microbiomeutilities/articles/microbiomeutilities.html
```{r, echo=FALSE,warning = FALSE}
load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/ps2.RData")
library(microbiomeutilities)
# set seed
set.seed(1)
subsamples <- seq(0, 50000, by=1000)[-1]
#subsamples = c(10, 5000, 10000, 20000, 30000)

p <- plot_alpha_rcurve(ps2, index="observed",
            subsamples = subsamples,
            lower.conf = 0.025, 
            upper.conf = 0.975,
            group="Production_site_2",
            label.color = "brown3",
            label.size = 3,
            label.min = TRUE);p

p <- p + scale_color_manual(values = mycola) + 
  scale_fill_manual(values = mycola)+ 
  theme(axis.text=element_text(size=8), panel.grid=element_blank(),
        strip.background = element_rect(colour=NA,fill="grey33"),
        strip.text.x = element_text(face="bold"))+
  theme(text=element_text(size=14,  family="sans"))+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=6))+
  theme(legend.position="none")+
  labs(tag = "")
print(p)

tiff("p.tiff", units="in", width=10, height=6, res=300)
ggsave("p.TIFF", plot = p)

######################
p2<- plot_alpha_rcurve(ps2, index="observed",
                     subsamples = subsamples,
                     lower.conf = 0.025, 
                     upper.conf = 0.975,
                     group="English_Name_2",
                     label.color = "brown3",
                     label.size = 3,
                     label.min = TRUE) 

p2 <- p2 + scale_color_manual(values = mycola) + 
  scale_fill_manual(values = mycola)+
  theme(axis.text=element_text(size=8), panel.grid=element_blank(),
        strip.background = element_rect(colour=NA,fill="grey33"),
        strip.text.x = element_text(face="bold"))+
  theme(text=element_text(size=16,  family="sans"))+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=6))+
  theme(legend.position="none")+labs(tag = "")
print(p2)

tiff("p2.tiff", units="in", width=10, height=6, res=300)
ggsave("p2.TIFF", plot = p2)
```


#Ccaluclate alpha diversity

```{r,echo=FALSE,warning=FALSE}
load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/ps2.RData")

#calculate community phylogenetic diversity & merge it to the figure above
summary(sample_sums(ps2))#[ 15421 taxa & 1197 samples ]
Bacteria_subsamp <- rarefy_even_depth(ps2, sample.size=1000 , rngseed=5163, replace=FALSE, trimOTUs=TRUE);Bacteria_subsamp#[ 1128 samples by 20 sample variables ]
#Lost Samples after rarefaction: 69 samples

pd=estimate_pd(Bacteria_subsamp);pd
write.csv(pd, file="pd_1000.csv")

```




#STATISTICS FOR ALPHA DIVERSITY
##Import the rarefied dataset & compute the statistics
#Load additional libraries
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")

library(microViz)
library(ggtext)
library(microViz)
library(ggraph)
library(DT)
library(corncob)
library(ggplot2)
library(patchwork)
library(ggplot2) # 3.2.1
library(dplyr)
library(lubridate)
library(cowplot) # 1.0.0
library(egg) # 0.4.5
library(microbiome)
library(knitr)
library(tidyplots)
library(rcompanion)
library(multcompView)
library(tidyverse)
library(ggpubr)
library(dunn.test)
library(FSA)
#install.packages("explore")
library(explore)

#Diversity was computed for using the functions mentioned in Rhea package considering ASVs with abundance >0.25%
#Load the alpha diversity data (rarefied at 1000 depth)#alpha-diversity_1000.tab
#Combine the pd on to  dataframe for other alpha diversity at 1000 sampling depth (alpha-diversity_1000.tab)
alpha_1000_PD <- read.table ("alpha-diversity_1000.tab.txt", check.names = FALSE, header = TRUE, dec = ".", sep = "\t", row.names = 1, comment.char = "");alpha_1000_PD
alpha_1000_PD$yield <- cut(alpha_1000_PD$`Yield_(kg/sq_meter)`, breaks = seq(0,150,20), include.lowest = TRUE)
alpha_1000_PD$yield
alpha_1000_PD$Fruit_long_diameter=cut(alpha_1000_PD$Fruit_longitudinal_diamete, breaks = seq(0,10,3), include.lowest = TRUE)
alpha_1000_PD$Fruit_long_diameter
alpha_1000_PD$Fruit_hori_diameter=cut(alpha_1000_PD$Fruit_horizontal_diameter, breaks = seq(0,10,3), include.lowest = TRUE)
alpha_1000_PD$Fruit_hori_diameter
alpha_1000_PD$seedweight_1000=cut(alpha_1000_PD$`1000_seeds_weight_(units)`, breaks = seq(0,5,1), include.lowest = TRUE)
alpha_1000_PD$seedweight_1000
alpha_1000_PD$ovaries =cut(alpha_1000_PD$Number_of_ovary, breaks = seq(0,12,4), include.lowest = TRUE)
alpha_1000_PD$ovaries
#write_csv(alpha_1000_PD,"alpha_1000_PD.csv")
#subset for the different parameters
alpha_1000_PD_Fruit_taste=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_taste),];alpha_1000_PD_Fruit_taste
alpha_1000_PD_Fruit_color=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_color),];alpha_1000_PD_Fruit_color
alpha_1000_PD_Fruit_shape=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_shape),];alpha_1000_PD_Fruit_shape
alpha_1000_PD_TMV_resistance=alpha_1000_PD[!is.na(alpha_1000_PD$TMV_resistance),];alpha_1000_PD_TMV_resistance
alpha_1000_PD_Insect_resistance=alpha_1000_PD[!is.na(alpha_1000_PD$Insect_resistance),];alpha_1000_PD_Insect_resistance
alpha_1000_PD_yield=alpha_1000_PD[!is.na(alpha_1000_PD$yield),];alpha_1000_PD_yield
alpha_1000_PD_seedweight_1000=alpha_1000_PD[!is.na(alpha_1000_PD$seedweight_1000),];alpha_1000_PD_seedweight_1000
alpha_1000_PD_ovaries=alpha_1000_PD[!is.na(alpha_1000_PD$ovaries),];alpha_1000_PD_ovaries
alpha_1000_PD_ovaries=alpha_1000_PD[!is.na(alpha_1000_PD$ovaries),];alpha_1000_PD_ovaries
alpha_1000_PD_Fruit_hori_diameter=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_hori_diameter),];alpha_1000_PD_Fruit_hori_diameter
alpha_1000_PD_Fruit_long_diameter=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_long_diameter),];alpha_1000_PD_Fruit_long_diameter

attach(alpha_1000_PD)
########################################################################################################################
#https://www.rdocumentation.org/packages/rstatix/versions/0.7.2
#citation("rstatix ")

#explore(alpha_1000_PD)
# report of all variables
#alpha_1000_PD %>% report(output_file = "report.html", output_dir = tempdir())
#alpha_1000_PD %>% get_summary_stats(type = "common")
##########################################################################################################################
x=alpha_1000_PD %>%
  group_by(English_Name_2, Production_site_2) %>%
  get_summary_stats(Shannon_Index, type = "common");x
write_csv(x,"x.csv")


alpha_1000_PD %>%
    get_summary_stats(Shannon_Index, show = c("mean", "sd", "median", "iqr"))
alpha_1000_PD %>%
    get_summary_stats(Richness, show = c("mean", "sd", "median", "iqr"))

alpha_1000_PD$Shannon_Index
```

#consider rhe richness index
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")

# RICHNESS
kruskal.test(Richness ~ English_Name_2, data = alpha_1000_PD)
#Kruskal-Wallis chi-squared = 747.66, df = 99, p-value < 2.2e-16
kruskal.test(Richness ~ Production_site_2, data = alpha_1000_PD)
#Kruskal-Wallis chi-squared = 179.67, df = 11, p-value < 2.2e-16
kruskal.test(Richness ~ Fruit_taste, data = alpha_1000_PD_Fruit_taste)
#Kruskal-Wallis chi-squared = 7.1908, df = 1, p-value = 0.007328
kruskal.test(Richness ~ Fruit_color, data = alpha_1000_PD_Fruit_color)
#Kruskal-Wallis chi-squared = 16.484, df = 4, p-value = 0.002434
kruskal.test(Richness ~ Fruit_shape, data = alpha_1000_PD_Fruit_shape)
#Kruskal-Wallis chi-squared = 45.906, df = 2, p-value = 1.076e-10
kruskal.test(Richness ~ TMV_resistance, data = alpha_1000_PD_TMV_resistance)
#Kruskal-Wallis chi-squared = 14.735, df = 2, p-value = 0.0006313
kruskal.test(Richness ~ Insect_resistance, data = alpha_1000_PD_Insect_resistance)
#Kruskal-Wallis chi-squared = 15.113, df = 5, p-value = 0.009889
kruskal.test(Richness ~ seedweight_1000, data = alpha_1000_PD_seedweight_1000)
#Kruskal-Wallis chi-squared = 3.1461, df = 3, p-value = 0.3696
kruskal.test(Richness ~ ovaries, data = alpha_1000_PD_ovaries)
#Kruskal-Wallis chi-squared = 49.287, df = 2, p-value = 1.983e-11
#yield
kruskal.test(Richness ~ yield, data = alpha_1000_PD_yield)
#Kruskal-Wallis chi-squared = 18.581, df = 4, p-value = 0.0009496
#Fruit_hori_diameter
kruskal.test(Richness ~ Fruit_hori_diameter, data = alpha_1000_PD_Fruit_hori_diameter)
#Kruskal-Wallis chi-squared = 7.6737, df = 2, p-value = 0.02156
#Fruit_long_diameter
kruskal.test(Richness ~ Fruit_long_diameter, data = alpha_1000_PD_Fruit_long_diameter)
#Kruskal-Wallis chi-squared = 5.3711, df = 2, p-value = 0.06818
#POST-HOC ANALYSIS WITH DUNN'S TEST
```


```{r,echo=T,warning=T}
#Region of production
Dunn_tomato_richness_Production_site_2=dunn_test(alpha_1000_PD, Richness ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_richness_Production_site_2
write_csv(Dunn_tomato_richness_Production_site_2, "Dunn_tomato_richness_Production_site_2.csv")
Dunn_tomato_richness_Production_site_2 <- dunnTest(alpha_1000_PD$Richness, alpha_1000_PD$Production_site_2, method = "bonferroni");Dunn_tomato_richness_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_richness_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_richness_Production_site_2$res);CLD_Dunn_tomato_richness_Production_site_2
```


```{r,echo=FALSE,warning=FALSE}
#cultivars
Dunn_tomato_richness_English_Name_2=dunn_test(alpha_1000_PD, Richness ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_richness_English_Name_2
write_csv(Dunn_tomato_richness_English_Name_2, "Dunn_tomato_richness_English_Name_2.csv")
Dunn_tomato_richness_English_Name_2 <- dunnTest(alpha_1000_PD$Richness, alpha_1000_PD$English_Name, method = "bonferroni");Dunn_tomato_richness_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_richness_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_richness_English_Name_2$res);CLD_Dunn_tomato_richness_English_Name_2
write_csv(CLD_Dunn_tomato_richness_English_Name_2,"CLD_Dunn_tomato_richness_English_Name_2.csv")
```


```{r,echo=FALSE,warning=FALSE}
#Taste
Dunn_tomato_Richness_Fruit_taste=dunn_test(alpha_1000_PD_Fruit_taste, Richness ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_taste
write_csv(Dunn_tomato_Richness_Fruit_taste, "Dunn_tomato_Richness_Fruit_taste.csv")
Dunn_tomato_Richness_Fruit_taste <- dunnTest(alpha_1000_PD_Fruit_taste$Richness, alpha_1000_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Richness_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_taste$res);CLD_Dunn_tomato_Richness_Fruit_taste
```


```{r,echo=FALSE,warning=FALSE}
#COLOR
Dunn_tomato_Richness_Fruit_color=dunn_test(alpha_1000_PD_Fruit_color, Richness ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_color
write_csv(Dunn_tomato_Richness_Fruit_color, "Dunn_tomato_Richness_Fruit_color.csv")
Dunn_tomato_Richness_Fruit_color <- dunnTest(alpha_1000_PD_Fruit_color$Richness, alpha_1000_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Richness_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_color$res);CLD_Dunn_tomato_Richness_Fruit_color
```


```{r,echo=FALSE,warning=FALSE}
#insect resistance
Dunn_tomato_Richness_Insect_resistance=dunn_test(alpha_1000_PD_Insect_resistance, Richness ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Insect_resistance
write_csv(Dunn_tomato_Richness_Insect_resistance, "Dunn_tomato_Richness_Insect_resistance.csv")
Dunn_tomato_Richness_Insect_resistance <- dunnTest(alpha_1000_PD_Insect_resistance$Richness, alpha_1000_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Richness_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Insect_resistance$res);CLD_Dunn_tomato_Richness_Insect_resistance
write_csv(CLD_Dunn_tomato_Richness_Insect_resistance,"CLD_Dunn_tomato_Richness_Insect_resistance.csv")

```


```{r,echo=FALSE,warning=FALSE}
#TMV RESISTANCE
Dunn_tomato_Richness_TMV_resistance=dunn_test(alpha_1000_PD_TMV_resistance, Richness ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_TMV_resistance
write_csv(Dunn_tomato_Richness_TMV_resistance, "Dunn_tomato_Richness_TMV_resistance.csv")
Dunn_tomato_Richness_TMV_resistance <- dunnTest(alpha_1000_PD_TMV_resistance$Richness, alpha_1000_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Richness_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_TMV_resistance$res);CLD_Dunn_tomato_Richness_TMV_resistance
write_csv(CLD_Dunn_tomato_Richness_TMV_resistance,"CLD_Dunn_tomato_Richness_TMV_resistance.csv")

```


```{r,echo=FALSE,warning=FALSE}
#seed weight
Dunn_tomato_Richness_seedweight_1000=dunn_test(alpha_1000_PD_seedweight_1000, Richness ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_seedweight_1000
write_csv(Dunn_tomato_Richness_seedweight_1000, "Dunn_tomato_Richness_seedweight_1000.csv")
Dunn_tomato_Richness_seedweight_1000 <- dunnTest(alpha_1000_PD_seedweight_1000$Richness, alpha_1000_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Richness_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_seedweight_1000$res);CLD_Dunn_tomato_Richness_seedweight_1000
```


```{r,echo=FALSE,warning=FALSE}
#seed weight
Dunn_tomato_Richness_Fruit_shape=dunn_test(alpha_1000_PD_Fruit_shape, Richness ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_shape
write_csv(Dunn_tomato_Richness_Fruit_shape, "Dunn_tomato_Richness_Fruit_shape.csv")
Dunn_tomato_Richness_Fruit_shape <- dunnTest(alpha_1000_PD_Fruit_shape$Richness, alpha_1000_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Richness_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_shape$res);CLD_Dunn_tomato_Richness_Fruit_shape
```


```{r,echo=FALSE,warning=FALSE}
#ovaries

Dunn_tomato_Richness_ovaries=dunn_test(alpha_1000_PD_ovaries, Richness ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_ovaries
write_csv(Dunn_tomato_Richness_ovaries, "Dunn_tomato_Richness_ovaries.csv")
Dunn_tomato_Richness_ovaries <- dunnTest(alpha_1000_PD_ovaries$Richness, alpha_1000_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Richness_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_ovaries$res);CLD_Dunn_tomato_Richness_ovaries
write_csv(CLD_Dunn_tomato_Richness_ovaries,"CLD_Dunn_tomato_Richness_ovaries.csv")

```


```{r,echo=FALSE,warning=FALSE}
#yield
Dunn_tomato_Richness_yield=dunn_test(alpha_1000_PD_yield, Richness ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_yield
write_csv(Dunn_tomato_Richness_yield, "Dunn_tomato_Richness_yield.csv")
Dunn_tomato_Richness_yield <- dunnTest(alpha_1000_PD_yield$Richness, alpha_1000_PD_yield$yield, method = "bonferroni");Dunn_tomato_Richness_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_yield$res);CLD_Dunn_tomato_Richness_yield
```


```{r,echo=FALSE,warning=FALSE}
#Fruit_hori_diameter
Dunn_tomato_Richness_Fruit_hori_diameter=dunn_test(alpha_1000_PD_Fruit_hori_diameter, Richness ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_hori_diameter
write_csv(Dunn_tomato_Richness_Fruit_hori_diameter, "Dunn_tomato_Richness_Fruit_hori_diameter.csv")
Dunn_tomato_Richness_Fruit_hori_diameter <- dunnTest(alpha_1000_PD_Fruit_hori_diameter$Richness, alpha_1000_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Richness_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_hori_diameter$res);CLD_Dunn_tomato_Richness_Fruit_hori_diameter
```


```{r,echo=FALSE,warning=FALSE}
#Fruit_long_diameter
Dunn_tomato_Richness_Fruit_long_diameter=dunn_test(alpha_1000_PD_Fruit_long_diameter, Richness ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_long_diameter
write_csv(Dunn_tomato_Richness_Fruit_long_diameter, "Dunn_tomato_Richness_Fruit_long_diameter.csv")
Dunn_tomato_Richness_Fruit_long_diameter <- dunnTest(alpha_1000_PD_Fruit_long_diameter$Richness, alpha_1000_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Richness_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_long_diameter$res);CLD_Dunn_tomato_Richness_Fruit_long_diameter 

```
 
 
#EFFECTIVE RICHNESS
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")

kruskal.test(Effective_Richness ~ English_Name_2, data = alpha_1000_PD)

kruskal.test(Effective_Richness ~ Production_site_2, data = alpha_1000_PD)

kruskal.test(Effective_Richness ~ Fruit_taste, data = alpha_1000_PD_Fruit_taste)

kruskal.test(Effective_Richness ~ Fruit_color, data = alpha_1000_PD_Fruit_color)

kruskal.test(Effective_Richness ~ Fruit_shape, data = alpha_1000_PD_Fruit_shape)

kruskal.test(Effective_Richness ~ TMV_resistance, data = alpha_1000_PD_TMV_resistance)

kruskal.test(Effective_Richness ~ Insect_resistance, data = alpha_1000_PD_Insect_resistance)

kruskal.test(Effective_Richness ~ seedweight_1000, data = alpha_1000_PD_seedweight_1000)

kruskal.test(Effective_Richness ~ ovaries, data = alpha_1000_PD_ovaries)

kruskal.test(Effective_Richness ~ yield, data = alpha_1000_PD_yield)

kruskal.test(Effective_Richness ~ Fruit_hori_diameter, data = alpha_1000_PD_Fruit_hori_diameter)

kruskal.test(Effective_Richness ~ Fruit_long_diameter, data = alpha_1000_PD_Fruit_long_diameter)
```


```{r,echo=FALSE,warning=FALSE}
#Post-hoc tests (Dunn's test)

#Region of production)
Dunn_tomato_Effective_Richness_Production_site_2=dunn_test(alpha_1000_PD, Effective_Richness ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Production_site_2
write_csv(Dunn_tomato_Effective_Richness_Production_site_2, "Dunn_tomato_Effective_Richness_Production_site_2.csv")
Dunn_tomato_Effective_Richness_Production_site_2 <- dunnTest(alpha_1000_PD$Effective_Richness, alpha_1000_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Effective_Richness_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Production_site_2$res);CLD_Dunn_tomato_Effective_Richness_Production_site_2 
write_csv(CLD_Dunn_tomato_Effective_Richness_Production_site_2,"CLD_Dunn_tomato_Effective_Richness_Production_site_2.csv")

#Cultivar
Dunn_tomato_Effective_Richness_English_Name_2=dunn_test(alpha_1000_PD, Effective_Richness ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_English_Name_2
write_csv(Dunn_tomato_Effective_Richness_English_Name_2, "Dunn_tomato_Effective_Richness_English_Name_2.csv")
Dunn_tomato_Effective_Richness_English_Name_2 <- dunnTest(alpha_1000_PD$Effective_Richness, alpha_1000_PD$English_Name, method = "bonferroni");Dunn_tomato_Effective_Richness_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_English_Name_2$res);CLD_Dunn_tomato_Effective_Richness_English_Name_2 
write_csv(CLD_Dunn_tomato_Effective_Richness_English_Name_2,"CLD_Dunn_tomato_Effective_Richness_English_Name_2.csv")

#taste
Dunn_tomato_Effective_Richness_Fruit_taste=dunn_test(alpha_1000_PD_Fruit_taste, Effective_Richness ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_taste
write_csv(Dunn_tomato_Effective_Richness_Fruit_taste, "Dunn_tomato_Effective_Richness_Fruit_taste.csv")
Dunn_tomato_Effective_Richness_Fruit_taste <- dunnTest(alpha_1000_PD_Fruit_taste$Effective_Richness, alpha_1000_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_taste$res);CLD_Dunn_tomato_Effective_Richness_Fruit_taste 
write_csv(CLD_Dunn_tomato_Effective_Richness_Fruit_taste,"CLD_Dunn_tomato_Effective_Richness_Fruit_taste.csv")

#color
Dunn_tomato_Effective_Richness_Fruit_color=dunn_test(alpha_1000_PD_Fruit_color, Effective_Richness ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_color
write_csv(Dunn_tomato_Effective_Richness_Fruit_color, "Dunn_tomato_Effective_Richness_Fruit_color.csv")
Dunn_tomato_Effective_Richness_Fruit_color <- dunnTest(alpha_1000_PD_Fruit_color$Effective_Richness, alpha_1000_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_color$res);CLD_Dunn_tomato_Effective_Richness_Fruit_color 
write_csv(CLD_Dunn_tomato_Effective_Richness_Fruit_color,"CLD_Dunn_tomato_Effective_Richness_Fruit_color.csv")

#insect resistance
Dunn_tomato_Effective_Richness_Insect_resistance=dunn_test(alpha_1000_PD_Insect_resistance, Effective_Richness ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Insect_resistance
write_csv(Dunn_tomato_Effective_Richness_Insect_resistance, "Dunn_tomato_Effective_Richness_Insect_resistance.csv")
Dunn_tomato_Effective_Richness_Insect_resistance <- dunnTest(alpha_1000_PD_Insect_resistance$Effective_Richness, alpha_1000_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Effective_Richness_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Insect_resistance$res);CLD_Dunn_tomato_Effective_Richness_Insect_resistance 
write_csv(CLD_Dunn_tomato_Effective_Richness_Insect_resistance,"CLD_Dunn_tomato_Effective_Richness_Insect_resistance.csv")

#TMV RESISTANCE
Dunn_tomato_Effective_Richness_TMV_resistance=dunn_test(alpha_1000_PD_TMV_resistance, Effective_Richness ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_TMV_resistance
write_csv(Dunn_tomato_Effective_Richness_TMV_resistance, "Dunn_tomato_Effective_Richness_TMV_resistance.csv")
Dunn_tomato_Effective_Richness_TMV_resistance <- dunnTest(alpha_1000_PD_TMV_resistance$Effective_Richness, alpha_1000_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Effective_Richness_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_TMV_resistance$res);CLD_Dunn_tomato_Effective_Richness_TMV_resistance 
write_csv(CLD_Dunn_tomato_Effective_Richness_TMV_resistance,"CLD_Dunn_tomato_Effective_Richness_TMV_resistance.csv")

#seed weight
Dunn_tomato_Effective_Richness_seedweight_1000=dunn_test(alpha_1000_PD_seedweight_1000, Effective_Richness ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_seedweight_1000
write_csv(Dunn_tomato_Effective_Richness_seedweight_1000, "Dunn_tomato_Effective_Richness_seedweight_1000.csv")
Dunn_tomato_Effective_Richness_seedweight_1000 <- dunnTest(alpha_1000_PD_seedweight_1000$Effective_Richness, alpha_1000_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Effective_Richness_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_seedweight_1000$res);CLD_Dunn_tomato_Effective_Richness_seedweight_1000 
write_csv(CLD_Dunn_tomato_Effective_Richness_seedweight_1000,"CLD_Dunn_tomato_Effective_Richness_seedweight_1000.csv")

#seed SHAPE
Dunn_tomato_Effective_Richness_Fruit_shape=dunn_test(alpha_1000_PD_Fruit_shape, Effective_Richness ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_shape
write_csv(Dunn_tomato_Effective_Richness_Fruit_shape, "Dunn_tomato_Effective_Richness_Fruit_shape.csv")
Dunn_tomato_Effective_Richness_Fruit_shape <- dunnTest(alpha_1000_PD_Fruit_shape$Effective_Richness, alpha_1000_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_shape$res);CLD_Dunn_tomato_Effective_Richness_Fruit_shape 
write_csv(CLD_Dunn_tomato_Effective_Richness_Fruit_shape,"CLD_Dunn_tomato_Effective_Richness_Fruit_shape.csv")

#ovaries
Dunn_tomato_Effective_Richness_ovaries=dunn_test(alpha_1000_PD_ovaries, Effective_Richness ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_ovaries
write_csv(Dunn_tomato_Effective_Richness_ovaries, "Dunn_tomato_Effective_Richness_ovaries.csv")
Dunn_tomato_Effective_Richness_ovaries <- dunnTest(alpha_1000_PD_ovaries$Effective_Richness, alpha_1000_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Effective_Richness_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_ovaries$res);CLD_Dunn_tomato_Effective_Richness_ovaries 
write_csv(CLD_Dunn_tomato_Effective_Richness_ovaries,"CLD_Dunn_tomato_Effective_Richness_ovaries.csv")

#yield
Dunn_tomato_Effective_Richness_yield=dunn_test(alpha_1000_PD_yield, Effective_Richness ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_yield
write_csv(Dunn_tomato_Effective_Richness_yield, "Dunn_tomato_Effective_Richness_yield.csv")
Dunn_tomato_Effective_Richness_yield <- dunnTest(alpha_1000_PD_yield$Effective_Richness, alpha_1000_PD_yield$yield, method = "bonferroni");Dunn_tomato_Effective_Richness_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_yield$res);CLD_Dunn_tomato_Effective_Richness_yield 
write_csv(CLD_Dunn_tomato_Effective_Richness_yield,"CLD_Dunn_tomato_Effective_Richness_yield.csv")

#Fruit_hori_diameter
Dunn_tomato_Effective_Richness_Fruit_hori_diameter=dunn_test(alpha_1000_PD_Fruit_hori_diameter, Effective_Richness ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_hori_diameter
write_csv(Dunn_tomato_Effective_Richness_Fruit_hori_diameter, "Dunn_tomato_Effective_Richness_Fruit_hori_diameter.csv")
Dunn_tomato_Effective_Richness_Fruit_hori_diameter <- dunnTest(alpha_1000_PD_Fruit_hori_diameter$Effective_Richness, alpha_1000_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_hori_diameter$res);CLD_Dunn_tomato_Effective_Richness_Fruit_hori_diameter 
write_csv(CLD_Dunn_tomato_Effective_Richness_Fruit_hori_diameter,"CLD_Dunn_tomato_Effective_Richness_Fruit_hori_diameter.csv")

#Fruit_long_diameter
Dunn_tomato_Effective_Richness_Fruit_long_diameter=dunn_test(alpha_1000_PD_Fruit_long_diameter, Effective_Richness ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_long_diameter
write_csv(Dunn_tomato_Effective_Richness_Fruit_long_diameter, "Dunn_tomato_Effective_Richness_Fruit_long_diameter.csv")
Dunn_tomato_Effective_Richness_Fruit_long_diameter <- dunnTest(alpha_1000_PD_Fruit_long_diameter$Effective_Richness, alpha_1000_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_long_diameter$res);CLD_Dunn_tomato_Effective_Richness_Fruit_long_diameter 
###########################################################################################

```
 
 
#SHANNON INDEX
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")
kruskal.test(Shannon_Index ~ English_Name_2, data = alpha_1000_PD)

kruskal.test(Shannon_Index ~ Production_site_2, data = alpha_1000_PD)

kruskal.test(Shannon_Index ~ Fruit_taste, data = alpha_1000_PD_Fruit_taste)

kruskal.test(Shannon_Index ~ Fruit_color, data = alpha_1000_PD_Fruit_color)

kruskal.test(Shannon_Index ~ Fruit_shape, data = alpha_1000_PD_Fruit_shape)

kruskal.test(Shannon_Index ~ TMV_resistance, data = alpha_1000_PD_TMV_resistance)

kruskal.test(Shannon_Index ~ Insect_resistance, data = alpha_1000_PD_Insect_resistance)

kruskal.test(Shannon_Index ~ seedweight_1000, data = alpha_1000_PD_seedweight_1000)

kruskal.test(Shannon_Index ~ ovaries, data = alpha_1000_PD_ovaries)

kruskal.test(Shannon_Index ~ yield, data = alpha_1000_PD_yield)

kruskal.test(Shannon_Index ~ Fruit_hori_diameter, data = alpha_1000_PD_Fruit_hori_diameter)

kruskal.test(Shannon_Index ~ Fruit_long_diameter, data = alpha_1000_PD_Fruit_long_diameter)
```


```{r,echo=FALSE,warning=FALSE}
#Post-hoc (Dunn's Test)

#Region of seed production
Dunn_tomato_Shannon_Index_Production_site_2=dunn_test(alpha_1000_PD, Shannon_Index ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Production_site_2
write_csv(Dunn_tomato_Shannon_Index_Production_site_2, "Dunn_tomato_Shannon_Index_Production_site_2.csv")
Dunn_tomato_Shannon_Index_Production_site_2 <- dunnTest(alpha_1000_PD$Shannon_Index, alpha_1000_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Shannon_Index_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Production_site_2$res);CLD_Dunn_tomato_Shannon_Index_Production_site_2 
write_csv(CLD_Dunn_tomato_Shannon_Index_Production_site_2,"CLD_Dunn_tomato_Shannon_Index_Production_site_2.csv")

#cultivar
Dunn_tomato_Shannon_Index_English_Name_2=dunn_test(alpha_1000_PD, Shannon_Index ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_English_Name_2
write_csv(Dunn_tomato_Shannon_Index_English_Name_2, "Dunn_tomato_Shannon_Index_English_Name_2.csv")
Dunn_tomato_Shannon_Index_English_Name_2 <- dunnTest(alpha_1000_PD$Shannon_Index, alpha_1000_PD$English_Name, method = "bonferroni");Dunn_tomato_Shannon_Index_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_English_Name_2$res);CLD_Dunn_tomato_Shannon_Index_English_Name_2 
write_csv(CLD_Dunn_tomato_Shannon_Index_English_Name_2, "CLD_Dunn_tomato_Shannon_Index_English_Name_2.csv")

#Taste
Dunn_tomato_Shannon_Index_Fruit_taste=dunn_test(alpha_1000_PD_Fruit_taste, Shannon_Index ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_taste
write_csv(Dunn_tomato_Shannon_Index_Fruit_taste, "Dunn_tomato_Shannon_Index_Fruit_taste.csv")
Dunn_tomato_Shannon_Index_Fruit_taste <- dunnTest(alpha_1000_PD_Fruit_taste$Shannon_Index, alpha_1000_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_taste$res);CLD_Dunn_tomato_Shannon_Index_Fruit_taste 
write_csv(CLD_Dunn_tomato_Shannon_Index_Fruit_taste,"CLD_Dunn_tomato_Shannon_Index_Fruit_taste.csv")


#color
Dunn_tomato_Shannon_Index_Fruit_color=dunn_test(alpha_1000_PD_Fruit_color, Shannon_Index ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_color
write_csv(Dunn_tomato_Shannon_Index_Fruit_color, "Dunn_tomato_Shannon_Index_Fruit_color.csv")
Dunn_tomato_Shannon_Index_Fruit_color <- dunnTest(alpha_1000_PD_Fruit_color$Shannon_Index, alpha_1000_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_color$res);CLD_Dunn_tomato_Shannon_Index_Fruit_color 
write_csv(CLD_Dunn_tomato_Shannon_Index_Fruit_color,"CLD_Dunn_tomato_Shannon_Index_Fruit_color.csv")

#insect resistance
Dunn_tomato_Shannon_Index_Insect_resistance=dunn_test(alpha_1000_PD_Insect_resistance, Shannon_Index ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Insect_resistance
write_csv(Dunn_tomato_Shannon_Index_Insect_resistance, "Dunn_tomato_Shannon_Index_Insect_resistance.csv")
Dunn_tomato_Shannon_Index_Insect_resistance <- dunnTest(alpha_1000_PD_Insect_resistance$Shannon_Index, alpha_1000_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Shannon_Index_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Insect_resistance$res);CLD_Dunn_tomato_Shannon_Index_Insect_resistance 
write_csv(CLD_Dunn_tomato_Shannon_Index_Insect_resistance,"CLD_Dunn_tomato_Shannon_Index_Insect_resistance.csv")


#TMV RESISTANCE
Dunn_tomato_Shannon_Index_TMV_resistance=dunn_test(alpha_1000_PD_TMV_resistance, Shannon_Index ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_TMV_resistance
write_csv(Dunn_tomato_Shannon_Index_TMV_resistance, "Dunn_tomato_Shannon_Index_TMV_resistance.csv")
Dunn_tomato_Shannon_Index_TMV_resistance <- dunnTest(alpha_1000_PD_TMV_resistance$Shannon_Index, alpha_1000_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Shannon_Index_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_TMV_resistance$res);CLD_Dunn_tomato_Shannon_Index_TMV_resistance 
write_csv(CLD_Dunn_tomato_Shannon_Index_TMV_resistance,"CLD_Dunn_tomato_Shannon_Index_TMV_resistance.csv")

#seed weight
Dunn_tomato_Shannon_Index_seedweight_1000=dunn_test(alpha_1000_PD_seedweight_1000, Shannon_Index ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_seedweight_1000
write_csv(Dunn_tomato_Shannon_Index_seedweight_1000, "Dunn_tomato_Shannon_Index_seedweight_1000.csv")
Dunn_tomato_Shannon_Index_seedweight_1000 <- dunnTest(alpha_1000_PD_seedweight_1000$Shannon_Index, alpha_1000_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Shannon_Index_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_seedweight_1000$res);CLD_Dunn_tomato_Shannon_Index_seedweight_1000 
write_csv(CLD_Dunn_tomato_Shannon_Index_seedweight_1000,"CLD_Dunn_tomato_Shannon_Index_seedweight_1000.csv")


#seed SHAPE
Dunn_tomato_Shannon_Index_Fruit_shape=dunn_test(alpha_1000_PD_Fruit_shape, Shannon_Index ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_shape
write_csv(Dunn_tomato_Shannon_Index_Fruit_shape, "Dunn_tomato_Shannon_Index_Fruit_shape.csv")
Dunn_tomato_Shannon_Index_Fruit_shape <- dunnTest(alpha_1000_PD_Fruit_shape$Shannon_Index, alpha_1000_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_shape$res);CLD_Dunn_tomato_Shannon_Index_Fruit_shape 
write_csv(CLD_Dunn_tomato_Shannon_Index_Fruit_shape, "CLD_Dunn_tomato_Shannon_Index_Fruit_shape.csv")

#ovaries
Dunn_tomato_Shannon_Index_ovaries=dunn_test(alpha_1000_PD_ovaries, Shannon_Index ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_ovaries
write_csv(Dunn_tomato_Shannon_Index_ovaries, "Dunn_tomato_Shannon_Index_ovaries.csv")
Dunn_tomato_Shannon_Index_ovaries <- dunnTest(alpha_1000_PD_ovaries$Shannon_Index, alpha_1000_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Shannon_Index_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_ovaries$res);CLD_Dunn_tomato_Shannon_Index_ovaries 
write_csv(CLD_Dunn_tomato_Shannon_Index_ovaries, "CLD_Dunn_tomato_Shannon_Index_ovaries.csv")

#yield
Dunn_tomato_Shannon_Index_yield=dunn_test(alpha_1000_PD_yield, Shannon_Index ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_yield
write_csv(Dunn_tomato_Shannon_Index_yield, "Dunn_tomato_Shannon_Index_yield.csv")
Dunn_tomato_Shannon_Index_yield <- dunnTest(alpha_1000_PD_yield$Shannon_Index, alpha_1000_PD_yield$yield, method = "bonferroni");Dunn_tomato_Shannon_Index_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_yield$res);CLD_Dunn_tomato_Shannon_Index_yield 
write_csv(CLD_Dunn_tomato_Shannon_Index_yield,"CLD_Dunn_tomato_Shannon_Index_yield.csv")

#Fruit_hori_diameter
Dunn_tomato_Shannon_Index_Fruit_hori_diameter=dunn_test(alpha_1000_PD_Fruit_hori_diameter, Shannon_Index ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_hori_diameter
write_csv(Dunn_tomato_Shannon_Index_Fruit_hori_diameter, "Dunn_tomato_Shannon_Index_Fruit_hori_diameter.csv")
Dunn_tomato_Shannon_Index_Fruit_hori_diameter <- dunnTest(alpha_1000_PD_Fruit_hori_diameter$Shannon_Index, alpha_1000_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_hori_diameter$res);CLD_Dunn_tomato_Shannon_Index_Fruit_hori_diameter 

#Fruit_long_diameter
Dunn_tomato_Shannon_Index_Fruit_long_diameter=dunn_test(alpha_1000_PD_Fruit_long_diameter, Shannon_Index ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_long_diameter
write_csv(Dunn_tomato_Shannon_Index_Fruit_long_diameter, "Dunn_tomato_Shannon_Index_Fruit_long_diameter.csv")
Dunn_tomato_Shannon_Index_Fruit_long_diameter <- dunnTest(alpha_1000_PD_Fruit_long_diameter$Shannon_Index, alpha_1000_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_long_diameter$res);CLD_Dunn_tomato_Shannon_Index_Fruit_long_diameter 
```
 

#Effective Shannon-diversity index
 
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")

kruskal.test(Shannon_Effective ~ English_Name_2, data = alpha_1000_PD)

kruskal.test(Shannon_Effective ~ Production_site_2, data = alpha_1000_PD)

kruskal.test(Shannon_Effective ~ Fruit_taste, data = alpha_1000_PD_Fruit_taste)

kruskal.test(Shannon_Effective ~ Fruit_color, data = alpha_1000_PD_Fruit_color)

kruskal.test(Shannon_Effective ~ Fruit_shape, data = alpha_1000_PD_Fruit_shape)

kruskal.test(Shannon_Effective ~ TMV_resistance, data = alpha_1000_PD_TMV_resistance)

kruskal.test(Shannon_Effective ~ Insect_resistance, data = alpha_1000_PD_Insect_resistance)

kruskal.test(Shannon_Effective ~ seedweight_1000, data = alpha_1000_PD_seedweight_1000)

kruskal.test(Shannon_Effective ~ ovaries, data = alpha_1000_PD_ovaries)

kruskal.test(Shannon_Effective ~ yield, data = alpha_1000_PD_yield)

kruskal.test(Shannon_Effective ~ Fruit_hori_diameter, data = alpha_1000_PD_Fruit_hori_diameter)

kruskal.test(Shannon_Effective ~ Fruit_long_diameter, data = alpha_1000_PD_Fruit_long_diameter)
```


```{r,echo=FALSE,warning=FALSE}
#Region of tomato seed production
# Post-hoc (Dunn's Test)
Dunn_tomato_Shannon_Effective_Production_site_2=dunn_test(alpha_1000_PD, Shannon_Effective ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Production_site_2
write_csv(Dunn_tomato_Shannon_Effective_Production_site_2, "Dunn_tomato_Shannon_Effective_Production_site_2.csv")
Dunn_tomato_Shannon_Effective_Production_site_2 <- dunnTest(alpha_1000_PD$Shannon_Effective, alpha_1000_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Shannon_Effective_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Production_site_2$res);CLD_Dunn_tomato_Shannon_Effective_Production_site_2 
write_csv(CLD_Dunn_tomato_Shannon_Effective_Production_site_2, "CLD_Dunn_tomato_Shannon_Effective_Production_site_2.csv")
#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_Shannon_Effective_English_Name_2=dunn_test(alpha_1000_PD, Shannon_Effective ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_English_Name_2
write_csv(Dunn_tomato_Shannon_Effective_English_Name_2, "Dunn_tomato_Shannon_Effective_English_Name_2.csv")
Dunn_tomato_Shannon_Effective_English_Name_2 <- dunnTest(alpha_1000_PD$Shannon_Effective, alpha_1000_PD$English_Name, method = "bonferroni");Dunn_tomato_Shannon_Effective_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_English_Name_2$res);CLD_Dunn_tomato_Shannon_Effective_English_Name_2 
write_csv(CLD_Dunn_tomato_Shannon_Effective_English_Name_2, "CLD_Dunn_tomato_Shannon_Effective_English_Name_2.csv")

#Taste
Dunn_tomato_Shannon_Effective_Fruit_taste=dunn_test(alpha_1000_PD_Fruit_taste, Shannon_Effective ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_taste
write_csv(Dunn_tomato_Shannon_Effective_Fruit_taste, "Dunn_tomato_Shannon_Effective_Fruit_taste.csv")
Dunn_tomato_Shannon_Effective_Fruit_taste <- dunnTest(alpha_1000_PD_Fruit_taste$Shannon_Effective, alpha_1000_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_taste$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_taste 
write_csv(CLD_Dunn_tomato_Shannon_Effective_Fruit_taste, "CLD_Dunn_tomato_Shannon_Effective_Fruit_taste.csv")

#Color
Dunn_tomato_Shannon_Effective_Fruit_color=dunn_test(alpha_1000_PD_Fruit_color, Shannon_Effective ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_color
write_csv(Dunn_tomato_Shannon_Effective_Fruit_color, "Dunn_tomato_Shannon_Effective_Fruit_color.csv")
Dunn_tomato_Shannon_Effective_Fruit_color <- dunnTest(alpha_1000_PD_Fruit_color$Shannon_Effective, alpha_1000_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_color$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_color 
write_csv(CLD_Dunn_tomato_Shannon_Effective_Fruit_color, "CLD_Dunn_tomato_Shannon_Effective_Fruit_color.csv")

#insect resistance
Dunn_tomato_Shannon_Effective_Insect_resistance=dunn_test(alpha_1000_PD_Insect_resistance, Shannon_Effective ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Insect_resistance
write_csv(Dunn_tomato_Shannon_Effective_Insect_resistance, "Dunn_tomato_Shannon_Effective_Insect_resistance.csv")
Dunn_tomato_Shannon_Effective_Insect_resistance <- dunnTest(alpha_1000_PD_Insect_resistance$Shannon_Effective, alpha_1000_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Shannon_Effective_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Insect_resistance$res);CLD_Dunn_tomato_Shannon_Effective_Insect_resistance 
write_csv(CLD_Dunn_tomato_Shannon_Effective_Insect_resistance, "CLD_Dunn_tomato_Shannon_Effective_Insect_resistance.csv")

#TMV RESISTANCE
Dunn_tomato_Shannon_Effective_TMV_resistance=dunn_test(alpha_1000_PD_TMV_resistance, Shannon_Effective ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_TMV_resistance
write_csv(Dunn_tomato_Shannon_Effective_TMV_resistance, "Dunn_tomato_Shannon_Effective_TMV_resistance.csv")
Dunn_tomato_Shannon_Effective_TMV_resistance <- dunnTest(alpha_1000_PD_TMV_resistance$Shannon_Effective, alpha_1000_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Shannon_Effective_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_TMV_resistance$res);CLD_Dunn_tomato_Shannon_Effective_TMV_resistance 
write_csv(CLD_Dunn_tomato_Shannon_Effective_TMV_resistance, "CLD_Dunn_tomato_Shannon_Effective_TMV_resistance.csv")

#seed weight
Dunn_tomato_Shannon_Effective_seedweight_1000=dunn_test(alpha_1000_PD_seedweight_1000, Shannon_Effective ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_seedweight_1000
write_csv(Dunn_tomato_Shannon_Effective_seedweight_1000, "Dunn_tomato_Shannon_Effective_seedweight_1000.csv")
Dunn_tomato_Shannon_Effective_seedweight_1000 <- dunnTest(alpha_1000_PD_seedweight_1000$Shannon_Effective, alpha_1000_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Shannon_Effective_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_seedweight_1000$res);CLD_Dunn_tomato_Shannon_Effective_seedweight_1000 
write_csv(CLD_Dunn_tomato_Shannon_Effective_seedweight_1000, "CLD_Dunn_tomato_Shannon_Effective_seedweight_1000.csv")

#seed SHAPE
Dunn_tomato_Shannon_Effective_Fruit_shape=dunn_test(alpha_1000_PD_Fruit_shape, Shannon_Effective ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_shape
write_csv(Dunn_tomato_Shannon_Effective_Fruit_shape, "Dunn_tomato_Shannon_Effective_Fruit_shape.csv")
Dunn_tomato_Shannon_Effective_Fruit_shape <- dunnTest(alpha_1000_PD_Fruit_shape$Shannon_Effective, alpha_1000_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_shape$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_shape 
write_csv(CLD_Dunn_tomato_Shannon_Effective_Fruit_shape, "CLD_Dunn_tomato_Shannon_Effective_Fruit_shape.csv")

#ovaries
Dunn_tomato_Shannon_Effective_ovaries=dunn_test(alpha_1000_PD_ovaries, Shannon_Effective ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_ovaries
write_csv(Dunn_tomato_Shannon_Effective_ovaries, "Dunn_tomato_Shannon_Effective_ovaries.csv")
Dunn_tomato_Shannon_Effective_ovaries <- dunnTest(alpha_1000_PD_ovaries$Shannon_Effective, alpha_1000_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Shannon_Effective_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_ovaries$res);CLD_Dunn_tomato_Shannon_Effective_ovaries 
write_csv(CLD_Dunn_tomato_Shannon_Effective_ovaries, "CLD_Dunn_tomato_Shannon_Effective_ovaries.csv")

#yield
Dunn_tomato_Shannon_Effective_yield=dunn_test(alpha_1000_PD_yield, Shannon_Effective ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_yield
write_csv(Dunn_tomato_Shannon_Effective_yield, "Dunn_tomato_Shannon_Effective_yield.csv")
Dunn_tomato_Shannon_Effective_yield <- dunnTest(alpha_1000_PD_yield$Shannon_Effective, alpha_1000_PD_yield$yield, method = "bonferroni");Dunn_tomato_Shannon_Effective_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_yield$res);CLD_Dunn_tomato_Shannon_Effective_yield
write_csv(CLD_Dunn_tomato_Shannon_Effective_yield, "CLD_Dunn_tomato_Shannon_Effective_yield.csv")

#Fruit_hori_diameter
Dunn_tomato_Shannon_Effective_Fruit_hori_diameter=dunn_test(alpha_1000_PD_Fruit_hori_diameter, Shannon_Effective ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_hori_diameter
write_csv(Dunn_tomato_Shannon_Effective_Fruit_hori_diameter, "Dunn_tomato_Shannon_Effective_Fruit_hori_diameter.csv")
Dunn_tomato_Shannon_Effective_Fruit_hori_diameter <- dunnTest(alpha_1000_PD_Fruit_hori_diameter$Shannon_Effective, alpha_1000_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_hori_diameter$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_hori_diameter

#Fruit_long_diameter
Dunn_tomato_Shannon_Effective_Fruit_long_diameter=dunn_test(alpha_1000_PD_Fruit_long_diameter, Shannon_Effective ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_long_diameter
write_csv(Dunn_tomato_Shannon_Effective_Fruit_long_diameter, "Dunn_tomato_Shannon_Effective_Fruit_long_diameter.csv")
Dunn_tomato_Shannon_Effective_Fruit_long_diameter <- dunnTest(alpha_1000_PD_Fruit_long_diameter$Shannon_Effective, alpha_1000_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_long_diameter$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_long_diameter
```
 
 
#PHYLOGENETIC DIVERSITY
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")

kruskal.test(PD ~ English_Name_2, data = alpha_1000_PD)

kruskal.test(PD ~ Production_site_2, data = alpha_1000_PD)

kruskal.test(PD ~ Fruit_taste, data = alpha_1000_PD_Fruit_taste)

kruskal.test(PD ~ Fruit_color, data = alpha_1000_PD_Fruit_color)

kruskal.test(PD ~ Fruit_shape, data = alpha_1000_PD_Fruit_shape)

kruskal.test(PD ~ TMV_resistance, data = alpha_1000_PD_TMV_resistance)

kruskal.test(PD ~ Insect_resistance, data = alpha_1000_PD_Insect_resistance)

kruskal.test(PD ~ seedweight_1000, data = alpha_1000_PD_seedweight_1000)

kruskal.test(PD ~ ovaries, data = alpha_1000_PD_ovaries)

kruskal.test(PD ~ yield, data = alpha_1000_PD_yield)

kruskal.test(PD ~ Fruit_hori_diameter, data = alpha_1000_PD_Fruit_hori_diameter)

kruskal.test(PD ~ Fruit_long_diameter, data = alpha_1000_PD_Fruit_long_diameter)
```


```{r,echo=FALSE,warning=FALSE}
# Post-hoc Dunn's Test 
#Region of tomato production

Dunn_tomato_PD_Production_site_2=dunn_test(alpha_1000_PD, PD ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Production_site_2
write_csv(Dunn_tomato_PD_Production_site_2, "Dunn_tomato_PD_Production_site_2.csv")
library(FSA)
library(rcompanion)
# Convert PD to numeric
alpha_1000_PD$PD <- as.numeric(as.character(alpha_1000_PD$PD))
# Remove any NAs
alpha_1000_PD <- alpha_1000_PD[!is.na(alpha_1000_PD$PD), ]
# Use formula notation (preferred)
Dunn_tomato_PD_Production_site_2 <- dunnTest(PD ~ Production_site_2, 
                                              data = alpha_1000_PD, 
                                              method = "bonferroni")
# View results
Dunn_tomato_PD_Production_site_2
# Get significance letters
CLD_Dunn_tomato_PD_Production_site_2 <- cldList(P.adj ~ Comparison, 
                                              data = Dunn_tomato_PD_Production_site_2$res)
write_csv(CLD_Dunn_tomato_PD_Production_site_2,"CLD_Dunn_tomato_PD_Production_site_2.csv") 


#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_PD_English_Name_2=dunn_test(alpha_1000_PD, PD ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_English_Name_2
write_csv(Dunn_tomato_PD_English_Name_2, "Dunn_tomato_PD_English_Name_2.csv")
Dunn_tomato_PD_English_Name_2 <- dunnTest(alpha_1000_PD$PD, alpha_1000_PD$English_Name, method = "bonferroni");Dunn_tomato_PD_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_English_Name_2$res);CLD_Dunn_tomato_PD_English_Name_2 
write_csv(CLD_Dunn_tomato_PD_English_Name_2, "CLD_Dunn_tomato_PD_English_Name_2.csv")

#Taste
Dunn_tomato_PD_Fruit_taste=dunn_test(alpha_1000_PD_Fruit_taste, PD ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_taste
write_csv(Dunn_tomato_PD_Fruit_taste, "Dunn_tomato_PD_Fruit_taste.csv")
class(alpha_1000_PD_Fruit_taste$PD)
# Convert to numeric
alpha_1000_PD_Fruit_taste$PD <- as.numeric(as.character(alpha_1000_PD_Fruit_taste$PD))
# Now try again with FORMULA notation (easier)
library(FSA)
Dunn_tomato_PD_Fruit_taste <- dunnTest(PD ~ Fruit_taste, 
                                        data = alpha_1000_PD_Fruit_taste, 
                                        method = "bonferroni"); Dunn_tomato_PD_Fruit_taste
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_taste$res);CLD_Dunn_tomato_PD_Fruit_taste 
write_csv(CLD_Dunn_tomato_PD_Fruit_taste, "CLD_Dunn_tomato_PD_Fruit_taste.csv")


#color
Dunn_tomato_PD_Fruit_color=dunn_test(alpha_1000_PD_Fruit_color, PD ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_color
write_csv(Dunn_tomato_PD_Fruit_color, "Dunn_tomato_PD_Fruit_color.csv")
library(FSA)
# Convert PD to numeric
alpha_1000_PD_Fruit_color$PD <- as.numeric(as.character(alpha_1000_PD_Fruit_color$PD))
# Use FORMULA notation (correct way)
Dunn_tomato_PD_Fruit_color <- dunnTest(PD ~ Fruit_color, 
                                        data = alpha_1000_PD_Fruit_color, 
                                        method = "bonferroni");Dunn_tomato_PD_Fruit_color
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_color$res);CLD_Dunn_tomato_PD_Fruit_color 
write_csv(CLD_Dunn_tomato_PD_Fruit_color, "CLD_Dunn_tomato_PD_Fruit_color.csv")


#insect resistance
Dunn_tomato_PD_Insect_resistance=dunn_test(alpha_1000_PD_Insect_resistance, PD ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Insect_resistance
write_csv(Dunn_tomato_PD_Insect_resistance, "Dunn_tomato_PD_Insect_resistance.csv")
library(FSA)
# Convert PD to numeric
alpha_1000_PD_Insect_resistance$PD <- as.numeric(as.character(alpha_1000_PD_Insect_resistance$PD))
# Use FORMULA notation (correct way)
Dunn_tomato_PD_Insect_resistance <- dunnTest(PD ~ Insect_resistance, 
                                        data = alpha_1000_PD_Insect_resistance, 
                                        method = "bonferroni");Dunn_tomato_PD_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Insect_resistance$res);CLD_Dunn_tomato_PD_Insect_resistance 
write_csv(CLD_Dunn_tomato_PD_Insect_resistance, "CLD_Dunn_tomato_PD_Insect_resistance.csv")


#TMV RESISTANCE
Dunn_tomato_PD_TMV_resistance=dunn_test(alpha_1000_PD_TMV_resistance, PD ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_TMV_resistance
write_csv(Dunn_tomato_PD_TMV_resistance, "Dunn_tomato_PD_TMV_resistance.csv")
library(FSA)
# Convert PD to numeric
alpha_1000_PD_TMV_resistance$PD <- as.numeric(as.character(alpha_1000_PD_TMV_resistance$PD))
# Use FORMULA notation (correct way)
Dunn_tomato_PD_TMV_resistance <- dunnTest(PD ~ TMV_resistance, 
                                        data = alpha_1000_PD_TMV_resistance, 
                                        method = "bonferroni");Dunn_tomato_PD_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_TMV_resistance$res);CLD_Dunn_tomato_PD_TMV_resistance 
write_csv(CLD_Dunn_tomato_PD_TMV_resistance, "CLD_Dunn_tomato_PD_TMV_resistance.csv")


#seed weight
Dunn_tomato_PD_seedweight_1000=dunn_test(alpha_1000_PD_seedweight_1000, PD ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_seedweight_1000
write_csv(Dunn_tomato_PD_seedweight_1000, "Dunn_tomato_PD_seedweight_1000.csv")
library(FSA)
# Convert PD to numeric
alpha_1000_PD_seedweight_1000$PD <- as.numeric(as.character(alpha_1000_PD_seedweight_1000$PD))
# Use FORMULA notation (correct way)
Dunn_tomato_PD_seedweight_1000 <- dunnTest(PD ~ seedweight_1000, 
                                        data = alpha_1000_PD_seedweight_1000, 
                                        method = "bonferroni");Dunn_tomato_PD_seedweight_1000
library(rcompanion)
CLD_Dunn_tomato_PD_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_seedweight_1000$res);CLD_Dunn_tomato_PD_seedweight_1000 
write_csv(CLD_Dunn_tomato_PD_seedweight_1000,"CLD_Dunn_tomato_PD_seedweight_1000.csv") 


#seed SHAPE
Dunn_tomato_PD_Fruit_shape=dunn_test(alpha_1000_PD_Fruit_shape, PD ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_shape
write_csv(Dunn_tomato_PD_Fruit_shape, "Dunn_tomato_PD_Fruit_shape.csv")
library(FSA)
# Convert PD to numeric
alpha_1000_PD_Fruit_shape$PD <- as.numeric(as.character(alpha_1000_PD_Fruit_shape$PD))
# Use FORMULA notation (correct way)
Dunn_tomato_PD_Fruit_shape <- dunnTest(PD ~ Fruit_shape, 
                                        data = alpha_1000_PD_Fruit_shape, 
                                        method = "bonferroni");Dunn_tomato_PD_Fruit_shape

# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_shape$res);CLD_Dunn_tomato_PD_Fruit_shape 
write_csv(CLD_Dunn_tomato_PD_Fruit_shape,"CLD_Dunn_tomato_PD_Fruit_shape.csv") 


#ovaries
Dunn_tomato_PD_ovaries=dunn_test(alpha_1000_PD_ovaries, PD ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_ovaries
write_csv(Dunn_tomato_PD_ovaries, "Dunn_tomato_PD_ovaries.csv")
library(FSA)
# Convert PD to numeric
alpha_1000_PD_ovaries$PD <- as.numeric(as.character(alpha_1000_PD_ovaries$PD))
# Use FORMULA notation (correct way)
Dunn_tomato_PD_ovaries <- dunnTest(PD ~ ovaries, 
                                        data = alpha_1000_PD_ovaries, 
                                        method = "bonferroni");Dunn_tomato_PD_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_ovaries$res);CLD_Dunn_tomato_PD_ovaries 
write_csv(CLD_Dunn_tomato_PD_ovaries, "CLD_Dunn_tomato_PD_ovaries.csv")


#yield
Dunn_tomato_PD_yield=dunn_test(alpha_1000_PD_yield, PD ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_yield
write_csv(Dunn_tomato_PD_yield, "Dunn_tomato_PD_yield.csv")

library(FSA)
alpha_1000_PD_yield$PD <- as.numeric(as.character(alpha_1000_PD_yield$PD))
Dunn_tomato_PD_yield <- dunnTest(alpha_1000_PD_yield$PD, alpha_1000_PD_yield$yield, method = "bonferroni");Dunn_tomato_PD_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_yield$res);CLD_Dunn_tomato_PD_yield 
write_csv(CLD_Dunn_tomato_PD_yield, "CLD_Dunn_tomato_PD_yield.csv")


#Fruit_hori_diameter
Dunn_tomato_PD_Fruit_hori_diameter=dunn_test(alpha_1000_PD_Fruit_hori_diameter, PD ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_hori_diameter
write_csv(Dunn_tomato_PD_Fruit_hori_diameter, "Dunn_tomato_PD_Fruit_hori_diameter.csv")

library(FSA)
alpha_1000_PD_Fruit_hori_diameter$PD <- as.numeric(as.character(alpha_1000_PD_Fruit_hori_diameter$PD))
Dunn_tomato_PD_Fruit_hori_diameter <- dunnTest(alpha_1000_PD_Fruit_hori_diameter$PD, alpha_1000_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_PD_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_hori_diameter$res);CLD_Dunn_tomato_PD_Fruit_hori_diameter 


#Fruit_long_diameter
Dunn_tomato_PD_Fruit_long_diameter=dunn_test(alpha_1000_PD_Fruit_long_diameter, PD ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_long_diameter
write_csv(Dunn_tomato_PD_Fruit_long_diameter, "Dunn_tomato_PD_Fruit_long_diameter.csv")
library(FSA)
alpha_1000_PD_Fruit_long_diameter$PD <- as.numeric(as.character(alpha_1000_PD_Fruit_long_diameter$PD))
Dunn_tomato_PD_Fruit_long_diameter <- dunnTest(alpha_1000_PD_Fruit_long_diameter$PD, alpha_1000_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_PD_Fruit_long_diameter

# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_long_diameter$res);CLD_Dunn_tomato_PD_Fruit_long_diameter 
###########################################################################################################################
```



#Simpson_Index
```{r,echo=FALSE,warning=F}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")


kruskal.test(Simpson_Index ~ English_Name_2, data = alpha_1000_PD)

kruskal.test(Simpson_Index ~ Production_site_2, data = alpha_1000_PD)


kruskal.test(Simpson_Index ~ Fruit_taste, data = alpha_1000_PD_Fruit_taste)

kruskal.test(Simpson_Index ~ Fruit_color, data = alpha_1000_PD_Fruit_color)

kruskal.test(Simpson_Index ~ Fruit_shape, data = alpha_1000_PD_Fruit_shape)

kruskal.test(Simpson_Index ~ TMV_resistance, data = alpha_1000_PD_TMV_resistance)

kruskal.test(Simpson_Index ~ Insect_resistance, data = alpha_1000_PD_Insect_resistance)

kruskal.test(Simpson_Index ~ seedweight_1000, data = alpha_1000_PD_seedweight_1000)

kruskal.test(Simpson_Index ~ ovaries, data = alpha_1000_PD_ovaries)

kruskal.test(Simpson_Index ~ yield, data = alpha_1000_PD_yield)

kruskal.test(Simpson_Index ~ Fruit_hori_diameter, data = alpha_1000_PD_Fruit_hori_diameter)

kruskal.test(Simpson_Index ~ Fruit_long_diameter, data = alpha_1000_PD_Fruit_long_diameter)
```


```{r,echo=FALSE,warning=F}
#Region of tomato production
# Dunn's Test 
Dunn_tomato_Simpson_Index_Production_site_2=dunn_test(alpha_1000_PD, Simpson_Index ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Production_site_2
write_csv(Dunn_tomato_Simpson_Index_Production_site_2, "Dunn_tomato_Simpson_Index_Production_site_2.csv")
Dunn_tomato_Simpson_Index_Production_site_2 <- dunnTest(alpha_1000_PD$Simpson_Index, alpha_1000_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Simpson_Index_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Production_site_2$res);CLD_Dunn_tomato_Simpson_Index_Production_site_2 
write_csv(CLD_Dunn_tomato_Simpson_Index_Production_site_2,"CLD_Dunn_tomato_Simpson_Index_Production_site_2.csv")

#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_Simpson_Index_English_Name_2=dunn_test(alpha_1000_PD, Simpson_Index ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_English_Name_2
write_csv(Dunn_tomato_Simpson_Index_English_Name_2, "Dunn_tomato_Simpson_Index_English_Name_2.csv")
Dunn_tomato_Simpson_Index_English_Name_2 <- dunnTest(alpha_1000_PD$Simpson_Index, alpha_1000_PD$English_Name, method = "bonferroni");Dunn_tomato_Simpson_Index_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_English_Name_2$res);CLD_Dunn_tomato_Simpson_Index_English_Name_2 
write_csv(CLD_Dunn_tomato_Simpson_Index_English_Name_2,"CLD_Dunn_tomato_Simpson_Index_English_Name_2.csv")

#Taste
Dunn_tomato_Simpson_Index_Fruit_taste=dunn_test(alpha_1000_PD_Fruit_taste, Simpson_Index ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_taste
write_csv(Dunn_tomato_Simpson_Index_Fruit_taste, "Dunn_tomato_Simpson_Index_Fruit_taste.csv")
Dunn_tomato_Simpson_Index_Fruit_taste <- dunnTest(alpha_1000_PD_Fruit_taste$Simpson_Index, alpha_1000_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_taste$res);CLD_Dunn_tomato_Simpson_Index_Fruit_taste 
write_csv(CLD_Dunn_tomato_Simpson_Index_Fruit_taste,"CLD_Dunn_tomato_Simpson_Index_Fruit_taste.csv")

#Colour
Dunn_tomato_Simpson_Index_Fruit_color=dunn_test(alpha_1000_PD_Fruit_color, Simpson_Index ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_color
write_csv(Dunn_tomato_Simpson_Index_Fruit_color, "Dunn_tomato_Simpson_Index_Fruit_color.csv")
Dunn_tomato_Simpson_Index_Fruit_color <- dunnTest(alpha_1000_PD_Fruit_color$Simpson_Index, alpha_1000_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_color
# letters of significance 
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_color$res);CLD_Dunn_tomato_Simpson_Index_Fruit_color 
write_csv(CLD_Dunn_tomato_Simpson_Index_Fruit_color,"CLD_Dunn_tomato_Simpson_Index_Fruit_color.csv")


#insect resistance
Dunn_tomato_Simpson_Index_Insect_resistance=dunn_test(alpha_1000_PD_Insect_resistance, Simpson_Index ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Insect_resistance
write_csv(Dunn_tomato_Simpson_Index_Insect_resistance, "Dunn_tomato_Simpson_Index_Insect_resistance.csv")
Dunn_tomato_Simpson_Index_Insect_resistance <- dunnTest(alpha_1000_PD_Insect_resistance$Simpson_Index, alpha_1000_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Simpson_Index_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Insect_resistance$res);CLD_Dunn_tomato_Simpson_Index_Insect_resistance 
write_csv(CLD_Dunn_tomato_Simpson_Index_Insect_resistance,"CLD_Dunn_tomato_Simpson_Index_Insect_resistance.csv")

#TMV RESISTANCE
Dunn_tomato_Simpson_Index_TMV_resistance=dunn_test(alpha_1000_PD_TMV_resistance, Simpson_Index ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_TMV_resistance
write_csv(Dunn_tomato_Simpson_Index_TMV_resistance, "Dunn_tomato_Simpson_Index_TMV_resistance.csv")
Dunn_tomato_Simpson_Index_TMV_resistance <- dunnTest(alpha_1000_PD_TMV_resistance$Simpson_Index, alpha_1000_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Simpson_Index_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_TMV_resistance$res);CLD_Dunn_tomato_Simpson_Index_TMV_resistance 
write_csv(CLD_Dunn_tomato_Simpson_Index_TMV_resistance,"CLD_Dunn_tomato_Simpson_Index_TMV_resistance.csv")

#seed weight
Dunn_tomato_Simpson_Index_seedweight_1000=dunn_test(alpha_1000_PD_seedweight_1000, Simpson_Index ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_seedweight_1000
write_csv(Dunn_tomato_Simpson_Index_seedweight_1000, "Dunn_tomato_Simpson_Index_seedweight_1000.csv")
Dunn_tomato_Simpson_Index_seedweight_1000 <- dunnTest(alpha_1000_PD_seedweight_1000$Simpson_Index, alpha_1000_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Simpson_Index_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_seedweight_1000$res);CLD_Dunn_tomato_Simpson_Index_seedweight_1000 
write_csv(CLD_Dunn_tomato_Simpson_Index_seedweight_1000,"CLD_Dunn_tomato_Simpson_Index_seedweight_1000.csv")

#seed SHAPE
Dunn_tomato_Simpson_Index_Fruit_shape=dunn_test(alpha_1000_PD_Fruit_shape, Simpson_Index ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_shape
write_csv(Dunn_tomato_Simpson_Index_Fruit_shape, "Dunn_tomato_Simpson_Index_Fruit_shape.csv")
Dunn_tomato_Simpson_Index_Fruit_shape <- dunnTest(alpha_1000_PD_Fruit_shape$Simpson_Index, alpha_1000_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_shape$res);CLD_Dunn_tomato_Simpson_Index_Fruit_shape 
write_csv(CLD_Dunn_tomato_Simpson_Index_Fruit_shape,"CLD_Dunn_tomato_Simpson_Index_Fruit_shape.csv")

#ovaries
Dunn_tomato_Simpson_Index_ovaries=dunn_test(alpha_1000_PD_ovaries, Simpson_Index ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_ovaries
write_csv(Dunn_tomato_Simpson_Index_ovaries, "Dunn_tomato_Simpson_Index_ovaries.csv")
Dunn_tomato_Simpson_Index_ovaries <- dunnTest(alpha_1000_PD_ovaries$Simpson_Index, alpha_1000_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Simpson_Index_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_ovaries$res);CLD_Dunn_tomato_Simpson_Index_ovaries 
write_csv(CLD_Dunn_tomato_Simpson_Index_ovaries,"CLD_Dunn_tomato_Simpson_Index_ovaries.csv")

#yield
Dunn_tomato_Simpson_Index_yield=dunn_test(alpha_1000_PD_yield, Simpson_Index ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_yield
write_csv(Dunn_tomato_Simpson_Index_yield, "Dunn_tomato_Simpson_Index_yield.csv")
Dunn_tomato_Simpson_Index_yield <- dunnTest(alpha_1000_PD_yield$Simpson_Index, alpha_1000_PD_yield$yield, method = "bonferroni");Dunn_tomato_Simpson_Index_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_yield$res);CLD_Dunn_tomato_Simpson_Index_yield 
write_csv(CLD_Dunn_tomato_Simpson_Index_yield,"CLD_Dunn_tomato_Simpson_Index_yield.csv")

#Fruit_hori_diameter
Dunn_tomato_Simpson_Index_Fruit_hori_diameter=dunn_test(alpha_1000_PD_Fruit_hori_diameter, Simpson_Index ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_hori_diameter
write_csv(Dunn_tomato_Simpson_Index_Fruit_hori_diameter, "Dunn_tomato_Simpson_Index_Fruit_hori_diameter.csv")
Dunn_tomato_Simpson_Index_Fruit_hori_diameter <- dunnTest(alpha_1000_PD_Fruit_hori_diameter$Simpson_Index, alpha_1000_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_hori_diameter$res);CLD_Dunn_tomato_Simpson_Index_Fruit_hori_diameter 
write_csv(CLD_Dunn_tomato_Simpson_Index_Fruit_hori_diameter,"CLD_Dunn_tomato_Simpson_Index_Fruit_hori_diameter.csv")

#Fruit_long_diameter
Dunn_tomato_Simpson_Index_Fruit_long_diameter=dunn_test(alpha_1000_PD_Fruit_long_diameter, Simpson_Index ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_long_diameter
write_csv(Dunn_tomato_Simpson_Index_Fruit_long_diameter, "Dunn_tomato_Simpson_Index_Fruit_long_diameter.csv")
Dunn_tomato_Simpson_Index_Fruit_long_diameter <- dunnTest(alpha_1000_PD_Fruit_long_diameter$Simpson_Index, alpha_1000_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_long_diameter$res);CLD_Dunn_tomato_Simpson_Index_Fruit_long_diameter 
```

#Effective simpson index
```{r,echo=FALSE,warning=FALSE}
########################################################################################################
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")

kruskal.test(Simpson_Effective ~English_Name_2, data = alpha_1000_PD)

kruskal.test(Simpson_Effective ~ Production_site_2, data = alpha_1000_PD)

kruskal.test(Simpson_Effective ~ Fruit_taste, data = alpha_1000_PD_Fruit_taste)

kruskal.test(Simpson_Effective ~ Fruit_color, data = alpha_1000_PD_Fruit_color)

kruskal.test(Simpson_Effective ~ Fruit_shape, data = alpha_1000_PD_Fruit_shape)

kruskal.test(Simpson_Effective ~ TMV_resistance, data = alpha_1000_PD_TMV_resistance)

kruskal.test(Simpson_Effective ~ Insect_resistance, data = alpha_1000_PD_Insect_resistance)

kruskal.test(Simpson_Effective ~ seedweight_1000, data = alpha_1000_PD_seedweight_1000)

kruskal.test(Simpson_Effective ~ ovaries, data = alpha_1000_PD_ovaries)

kruskal.test(Simpson_Effective ~ yield, data = alpha_1000_PD_yield)

kruskal.test(Simpson_Effective ~ Fruit_hori_diameter, data = alpha_1000_PD_Fruit_hori_diameter)

kruskal.test(Simpson_Effective ~ Fruit_long_diameter, data = alpha_1000_PD_Fruit_long_diameter)
```


```{r,echo=FALSE,warning=FALSE}
#Region of production
# Dunn's Test 
Dunn_tomato_Simpson_Effective_Production_site_2=dunn_test(alpha_1000_PD, Simpson_Effective ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Production_site_2
write_csv(Dunn_tomato_Simpson_Effective_Production_site_2, "Dunn_tomato_Simpson_Effective_Production_site_2.csv")
Dunn_tomato_Simpson_Effective_Production_site_2 <- dunnTest(alpha_1000_PD$Simpson_Effective, alpha_1000_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Simpson_Effective_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Production_site_2$res);CLD_Dunn_tomato_Simpson_Effective_Production_site_2 
write_csv(CLD_Dunn_tomato_Simpson_Effective_Production_site_2,"CLD_Dunn_tomato_Simpson_Effective_Production_site_2.csv")

#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_Simpson_Effective_English_Name_2=dunn_test(alpha_1000_PD, Simpson_Effective ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_English_Name_2
write_csv(Dunn_tomato_Simpson_Effective_English_Name_2, "Dunn_tomato_Simpson_Effective_English_Name_2.csv")
Dunn_tomato_Simpson_Effective_English_Name_2 <- dunnTest(alpha_1000_PD$Simpson_Effective, alpha_1000_PD$English_Name, method = "bonferroni");Dunn_tomato_Simpson_Effective_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_English_Name_2$res);CLD_Dunn_tomato_Simpson_Effective_English_Name_2 
write_csv(CLD_Dunn_tomato_Simpson_Effective_English_Name_2,"CLD_Dunn_tomato_Simpson_Effective_English_Name_2.csv")

#Taste
Dunn_tomato_Simpson_Effective_Fruit_taste=dunn_test(alpha_1000_PD_Fruit_taste, Simpson_Effective ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_taste
write_csv(Dunn_tomato_Simpson_Effective_Fruit_taste, "Dunn_tomato_Simpson_Effective_Fruit_taste.csv")
Dunn_tomato_Simpson_Effective_Fruit_taste <- dunnTest(alpha_1000_PD_Fruit_taste$Simpson_Effective, alpha_1000_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_taste$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_taste 
write_csv(CLD_Dunn_tomato_Simpson_Effective_Fruit_taste,"CLD_Dunn_tomato_Simpson_Effective_Fruit_taste.csv")

#Color
Dunn_tomato_Simpson_Effective_Fruit_color=dunn_test(alpha_1000_PD_Fruit_color, Simpson_Effective ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_color
write_csv(Dunn_tomato_Simpson_Effective_Fruit_color, "Dunn_tomato_Simpson_Effective_Fruit_color.csv")
Dunn_tomato_Simpson_Effective_Fruit_color <- dunnTest(alpha_1000_PD_Fruit_color$Simpson_Effective, alpha_1000_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_color$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_color 
write_csv(CLD_Dunn_tomato_Simpson_Effective_Fruit_color,"CLD_Dunn_tomato_Simpson_Effective_Fruit_color.csv")


#insect resistance
Dunn_tomato_Simpson_Effective_Insect_resistance=dunn_test(alpha_1000_PD_Insect_resistance, Simpson_Effective ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Insect_resistance
write_csv(Dunn_tomato_Simpson_Effective_Insect_resistance, "Dunn_tomato_Simpson_Effective_Insect_resistance.csv")
Dunn_tomato_Simpson_Effective_Insect_resistance <- dunnTest(alpha_1000_PD_Insect_resistance$Simpson_Effective, alpha_1000_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Simpson_Effective_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Insect_resistance$res);CLD_Dunn_tomato_Simpson_Effective_Insect_resistance 
write_csv(CLD_Dunn_tomato_Simpson_Effective_Insect_resistance,"CLD_Dunn_tomato_Simpson_Effective_Insect_resistance.csv")


#TMV RESISTANCE
Dunn_tomato_Simpson_Effective_TMV_resistance=dunn_test(alpha_1000_PD_TMV_resistance, Simpson_Effective ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_TMV_resistance
write_csv(Dunn_tomato_Simpson_Effective_TMV_resistance, "Dunn_tomato_Simpson_Effective_TMV_resistance.csv")
Dunn_tomato_Simpson_Effective_TMV_resistance <- dunnTest(alpha_1000_PD_TMV_resistance$Simpson_Effective, alpha_1000_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Simpson_Effective_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_TMV_resistance$res);CLD_Dunn_tomato_Simpson_Effective_TMV_resistance 
write_csv(CLD_Dunn_tomato_Simpson_Effective_TMV_resistance,"CLD_Dunn_tomato_Simpson_Effective_TMV_resistance.csv")


#seed weight
Dunn_tomato_Simpson_Effective_seedweight_1000=dunn_test(alpha_1000_PD_seedweight_1000, Simpson_Effective ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_seedweight_1000
write_csv(Dunn_tomato_Simpson_Effective_seedweight_1000, "Dunn_tomato_Simpson_Effective_seedweight_1000.csv")
Dunn_tomato_Simpson_Effective_seedweight_1000 <- dunnTest(alpha_1000_PD_seedweight_1000$Simpson_Effective, alpha_1000_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Simpson_Effective_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_seedweight_1000$res);CLD_Dunn_tomato_Simpson_Effective_seedweight_1000 
write_csv(CLD_Dunn_tomato_Simpson_Effective_seedweight_1000,"CLD_Dunn_tomato_Simpson_Effective_seedweight_1000.csv")

#seed SHAPE
Dunn_tomato_Simpson_Effective_Fruit_shape=dunn_test(alpha_1000_PD_Fruit_shape, Simpson_Effective ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_shape
write_csv(Dunn_tomato_Simpson_Effective_Fruit_shape, "Dunn_tomato_Simpson_Effective_Fruit_shape.csv")
Dunn_tomato_Simpson_Effective_Fruit_shape <- dunnTest(alpha_1000_PD_Fruit_shape$Simpson_Effective, alpha_1000_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_shape$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_shape
write_csv(CLD_Dunn_tomato_Simpson_Effective_Fruit_shape,"CLD_Dunn_tomato_Simpson_Effective_Fruit_shape.csv")

#ovaries
Dunn_tomato_Simpson_Effective_ovaries=dunn_test(alpha_1000_PD_ovaries, Simpson_Effective ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_ovaries
write_csv(Dunn_tomato_Simpson_Effective_ovaries, "Dunn_tomato_Simpson_Effective_ovaries.csv")
Dunn_tomato_Simpson_Effective_ovaries <- dunnTest(alpha_1000_PD_ovaries$Simpson_Effective, alpha_1000_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Simpson_Effective_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_ovaries$res);CLD_Dunn_tomato_Simpson_Effective_ovaries
write_csv(CLD_Dunn_tomato_Simpson_Effective_ovaries,"CLD_Dunn_tomato_Simpson_Effective_ovaries.csv")


#yield
Dunn_tomato_Simpson_Effective_yield=dunn_test(alpha_1000_PD_yield, Simpson_Effective ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_yield
write_csv(Dunn_tomato_Simpson_Effective_yield, "Dunn_tomato_Simpson_Effective_yield.csv")
Dunn_tomato_Simpson_Effective_yield <- dunnTest(alpha_1000_PD_yield$Simpson_Effective, alpha_1000_PD_yield$yield, method = "bonferroni");Dunn_tomato_Simpson_Effective_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_yield$res);CLD_Dunn_tomato_Simpson_Effective_yield
write_csv(CLD_Dunn_tomato_Simpson_Effective_yield,"CLD_Dunn_tomato_Simpson_Effective_yield.csv")

#Fruit_hori_diameter
Dunn_tomato_Simpson_Effective_Fruit_hori_diameter=dunn_test(alpha_1000_PD_Fruit_hori_diameter, Simpson_Effective ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_hori_diameter
write_csv(Dunn_tomato_Simpson_Effective_Fruit_hori_diameter, "Dunn_tomato_Simpson_Effective_Fruit_hori_diameter.csv")
Dunn_tomato_Simpson_Effective_Fruit_hori_diameter <- dunnTest(alpha_1000_PD_Fruit_hori_diameter$Simpson_Effective, alpha_1000_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_hori_diameter$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_hori_diameter
write_csv(CLD_Dunn_tomato_Simpson_Effective_Fruit_hori_diameter,"CLD_Dunn_tomato_Simpson_Effective_Fruit_hori_diameter.csv")

#Fruit_long_diameter
Dunn_tomato_Simpson_Effective_Fruit_long_diameter=dunn_test(alpha_1000_PD_Fruit_long_diameter, Simpson_Effective ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_long_diameter
write_csv(Dunn_tomato_Simpson_Effective_Fruit_long_diameter, "Dunn_tomato_Simpson_Effective_Fruit_long_diameter.csv")
Dunn_tomato_Simpson_Effective_Fruit_long_diameter <- dunnTest(alpha_1000_PD_Fruit_long_diameter$Simpson_Effective, alpha_1000_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_long_diameter$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_long_diameter
```


#EVENNESS
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")

kruskal.test(Evenness ~ English_Name_2, data = alpha_1000_PD)

kruskal.test(Evenness ~ Production_site_2, data = alpha_1000_PD)

kruskal.test(Evenness ~ Fruit_taste, data = alpha_1000_PD_Fruit_taste)

kruskal.test(Evenness ~ Fruit_color, data = alpha_1000_PD_Fruit_color)

kruskal.test(Evenness ~ Fruit_shape, data = alpha_1000_PD_Fruit_shape)

kruskal.test(Evenness ~ TMV_resistance, data = alpha_1000_PD_TMV_resistance)

kruskal.test(Evenness ~ Insect_resistance, data = alpha_1000_PD_Insect_resistance)

kruskal.test(Evenness ~ seedweight_1000, data = alpha_1000_PD_seedweight_1000)

kruskal.test(Evenness ~ ovaries, data = alpha_1000_PD_ovaries)

kruskal.test(Evenness ~ yield, data = alpha_1000_PD_yield)

kruskal.test(Evenness ~ Fruit_hori_diameter, data = alpha_1000_PD_Fruit_hori_diameter)

kruskal.test(Evenness ~ Fruit_long_diameter, data = alpha_1000_PD_Fruit_long_diameter)
```


```{r,echo=FALSE,warning=FALSE}
#Post-hoc (Dunn's tests)

# Region of tomato seed production
Dunn_tomato_Evenness_Production_site_2=dunn_test(alpha_1000_PD, Evenness ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Production_site_2
write_csv(Dunn_tomato_Evenness_Production_site_2, "Dunn_tomato_Evenness_Production_site_2.csv")
Dunn_tomato_Evenness_Production_site_2 <- dunnTest(alpha_1000_PD$Evenness, alpha_1000_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Evenness_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Production_site_2$res);CLD_Dunn_tomato_Evenness_Production_site_2 
write_csv(CLD_Dunn_tomato_Evenness_Production_site_2,"CLD_Dunn_tomato_Evenness_Production_site_2.csv")

#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_Evenness_English_Name_2=dunn_test(alpha_1000_PD, Evenness ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_English_Name_2
write_csv(Dunn_tomato_Evenness_English_Name_2, "Dunn_tomato_Evenness_English_Name_2.csv")
Dunn_tomato_Evenness_English_Name_2 <- dunnTest(alpha_1000_PD$Evenness, alpha_1000_PD$English_Name, method = "bonferroni");Dunn_tomato_Evenness_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_English_Name_2$res);CLD_Dunn_tomato_Evenness_English_Name_2 
write_csv(CLD_Dunn_tomato_Evenness_English_Name_2,"CLD_Dunn_tomato_Evenness_English_Name_2.csv")
#Taste
Dunn_tomato_Evenness_Fruit_taste=dunn_test(alpha_1000_PD_Fruit_taste, Evenness ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_taste
write_csv(Dunn_tomato_Evenness_Fruit_taste, "Dunn_tomato_Evenness_Fruit_taste.csv")
Dunn_tomato_Evenness_Fruit_taste <- dunnTest(alpha_1000_PD_Fruit_taste$Evenness, alpha_1000_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Evenness_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_taste$res);CLD_Dunn_tomato_Evenness_Fruit_taste 
write_csv(CLD_Dunn_tomato_Evenness_Fruit_taste,"CLD_Dunn_tomato_Evenness_Fruit_taste.csv")

#color
Dunn_tomato_Evenness_Fruit_color=dunn_test(alpha_1000_PD_Fruit_color, Evenness ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_color
write_csv(Dunn_tomato_Evenness_Fruit_color, "Dunn_tomato_Evenness_Fruit_color.csv")
Dunn_tomato_Evenness_Fruit_color <- dunnTest(alpha_1000_PD_Fruit_color$Evenness, alpha_1000_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Evenness_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_color$res);CLD_Dunn_tomato_Evenness_Fruit_color 
write_csv(CLD_Dunn_tomato_Evenness_Fruit_color,"CLD_Dunn_tomato_Evenness_Fruit_color.csv")

#insect resistance
Dunn_tomato_Evenness_Insect_resistance=dunn_test(alpha_1000_PD_Insect_resistance, Evenness ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Insect_resistance
write_csv(Dunn_tomato_Evenness_Insect_resistance, "Dunn_tomato_Evenness_Insect_resistance.csv")
Dunn_tomato_Evenness_Insect_resistance <- dunnTest(alpha_1000_PD_Insect_resistance$Evenness, alpha_1000_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Evenness_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Insect_resistance$res);CLD_Dunn_tomato_Evenness_Insect_resistance 
write_csv(CLD_Dunn_tomato_Evenness_Insect_resistance,"CLD_Dunn_tomato_Evenness_Insect_resistance.csv")

#TMV RESISTANCE
Dunn_tomato_Evenness_TMV_resistance=dunn_test(alpha_1000_PD_TMV_resistance, Evenness ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_TMV_resistance
write_csv(Dunn_tomato_Evenness_TMV_resistance, "Dunn_tomato_Evenness_TMV_resistance.csv")
Dunn_tomato_Evenness_TMV_resistance <- dunnTest(alpha_1000_PD_TMV_resistance$Evenness, alpha_1000_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Evenness_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_TMV_resistance$res);CLD_Dunn_tomato_Evenness_TMV_resistance 
write_csv(CLD_Dunn_tomato_Evenness_TMV_resistance,"CLD_Dunn_tomato_Evenness_TMV_resistance.csv")

#seed weight
Dunn_tomato_Evenness_seedweight_1000=dunn_test(alpha_1000_PD_seedweight_1000, Evenness ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_seedweight_1000
write_csv(Dunn_tomato_Evenness_seedweight_1000, "Dunn_tomato_Evenness_seedweight_1000.csv")
Dunn_tomato_Evenness_seedweight_1000 <- dunnTest(alpha_1000_PD_seedweight_1000$Evenness, alpha_1000_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Evenness_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_seedweight_1000$res);CLD_Dunn_tomato_Evenness_seedweight_1000 
write_csv(CLD_Dunn_tomato_Evenness_seedweight_1000,"CLD_Dunn_tomato_Evenness_seedweight_1000.csv")

#seed SHAPE
Dunn_tomato_Evenness_Fruit_shape=dunn_test(alpha_1000_PD_Fruit_shape, Evenness ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_shape
write_csv(Dunn_tomato_Evenness_Fruit_shape, "Dunn_tomato_Evenness_Fruit_shape.csv")
Dunn_tomato_Evenness_Fruit_shape <- dunnTest(alpha_1000_PD_Fruit_shape$Evenness, alpha_1000_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Evenness_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_shape$res);CLD_Dunn_tomato_Evenness_Fruit_shape 
write_csv(CLD_Dunn_tomato_Evenness_Fruit_shape,"CLD_Dunn_tomato_Evenness_Fruit_shape.csv")

#ovaries
Dunn_tomato_Evenness_ovaries=dunn_test(alpha_1000_PD_ovaries, Evenness ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_ovaries
write_csv(Dunn_tomato_Evenness_ovaries, "Dunn_tomato_Evenness_ovaries.csv")
Dunn_tomato_Evenness_ovaries <- dunnTest(alpha_1000_PD_ovaries$Evenness, alpha_1000_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Evenness_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_ovaries$res);CLD_Dunn_tomato_Evenness_ovaries 
write_csv(CLD_Dunn_tomato_Evenness_ovaries,"CLD_Dunn_tomato_Evenness_ovaries.csv")

#yield
Dunn_tomato_Evenness_yield=dunn_test(alpha_1000_PD_yield, Evenness ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_yield
write_csv(Dunn_tomato_Evenness_yield, "Dunn_tomato_Evenness_yield.csv")
Dunn_tomato_Evenness_yield <- dunnTest(alpha_1000_PD_yield$Evenness, alpha_1000_PD_yield$yield, method = "bonferroni");Dunn_tomato_Evenness_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_yield$res);CLD_Dunn_tomato_Evenness_yield 
write_csv(CLD_Dunn_tomato_Evenness_yield,"CLD_Dunn_tomato_Evenness_yield.csv")

#Fruit_hori_diameter
Dunn_tomato_Evenness_Fruit_hori_diameter=dunn_test(alpha_1000_PD_Fruit_hori_diameter, Evenness ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_hori_diameter
write_csv(Dunn_tomato_Evenness_Fruit_hori_diameter, "Dunn_tomato_Evenness_Fruit_hori_diameter.csv")
Dunn_tomato_Evenness_Fruit_hori_diameter <- dunnTest(alpha_1000_PD_Fruit_hori_diameter$Evenness, alpha_1000_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Evenness_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_hori_diameter$res);CLD_Dunn_tomato_Evenness_Fruit_hori_diameter 

#Fruit_long_diameter
Dunn_tomato_Evenness_Fruit_long_diameter=dunn_test(alpha_1000_PD_Fruit_long_diameter, Evenness ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_long_diameter
write_csv(Dunn_tomato_Evenness_Fruit_long_diameter, "Dunn_tomato_Evenness_Fruit_long_diameter.csv")
Dunn_tomato_Evenness_Fruit_long_diameter <- dunnTest(alpha_1000_PD_Fruit_long_diameter$Evenness, alpha_1000_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Evenness_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_long_diameter$res);CLD_Dunn_tomato_Evenness_Fruit_long_diameter 
```


#Alpha diversity  graphics  
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome")
#1:richness_shannon_pd_Simpson_eveness.tiff
plot_richness=ggboxplot(alpha_1000_PD, x = "English_Name_2", y = "Richness", fill= "English_Name_2",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=18))+
  theme(text=element_text(size=18,  family="sans"))+ labs(tag = "");plot_richness


#SHANON DIVERSITY

  
plot_Shannon=ggboxplot(alpha_1000_PD, x = "English_Name_2", y = "Shannon_Index", fill= "English_Name_2", ylab = "Shannon Index", xlab = "Tomato cultivars")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_text( colour="gray33", size=18),
        axis.text.y  = element_text( colour="gray33", size=18))+
  theme(text=element_text(size=18,  family="sans"))+ labs(tag = "");plot_Shannon
#ggsave("plot_Shannon.TIFF", plot = plot_Shannon)



#Phylogenetic diversity
plot_PD=ggboxplot(alpha_1000_PD, x = "English_Name_2", y = "PD", fill= "English_Name_2", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=18))+
  theme(text=element_text(size=18,  family="sans"))+ labs(tag = "");plot_PD
#ggsave("plot_PD.TIFF", plot = plot_PD)


#Figure 2A & B
#tiff("richness_shannon_pd_Simpson_eveness.tiff", units="in", width=20, height=18, res=300)
tiff("richness_shannon_pd_Simpson_eveness.tiff", units="in", width=18, height=12, res=300)
richness_shannon_pd_Simpson_eveness.tiff=(plot_richness/plot_Shannon);richness_shannon_pd_Simpson_eveness.tiff
ggsave("richness_shannon_pd_Simpson_eveness.tiff", plot = richness_shannon_pd_Simpson_eveness.tiff)
dev.off()

 
plot_Simpson=ggboxplot(alpha_1000_PD, x = "English_Name_2", y = "Simpson_Index", fill= "English_Name_2", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "");plot_Simpson
#ggsave("plot_Simpson.TIFF", plot = plot_Simpson)

#EVENNESS
 
plot_evenness=ggboxplot(alpha_1000_PD, x = "English_Name_2", y = "Evenness", fill= "English_Name_2", ylab = "Evenness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "");plot_evenness

#Effective_alpha_diversity
tiff("Effective_alpha_diversity.tiff", units="in", width=20, height=15, res=300)
plot_effective_richness=ggboxplot(alpha_1000_PD, x = "English_Name_2", y = "Effective_Richness", fill= "English_Name_2",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=16),
        axis.title.y = element_text(face="bold", colour="gray33", size=16),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "");plot_effective_richness


plot_effective_Shannon=ggboxplot(alpha_1000_PD, x = "English_Name_2", y = "Shannon_Effective", fill= "English_Name_2",  ylab = "Eff. Shannon div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('gray', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=16),
        axis.title.y = element_text(face="bold", colour="gray33", size=16),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "");plot_effective_Shannon


plot_effective_Simpson=ggboxplot(alpha_1000_PD, x = "English_Name_2", y = "Simpson_Effective", fill= "English_Name_2",  ylab = "Eff. Simpson div.", xlab = "Cultivar (n=100)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('gray', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=16),
        axis.title.y = element_text(face="bold", colour="gray33", size=16),
        axis.text.x = element_text( colour="gray33", size=16),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "");plot_effective_Simpson


#Draw and include the rarefaction curves

load("/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome/Tomato-seed-microbiome/ps2.RData")
library(microbiomeutilities)
# set seed
set.seed(1)
subsamples <- seq(0, 50000, by=500)[-1]
#subsamples = c(10, 5000, 10000, 20000, 30000)

p <- plot_alpha_rcurve(ps2, index="observed",
            subsamples = subsamples,
            lower.conf = 0.025, 
            upper.conf = 0.975,
            group="Production_site_2",
            label.color = "brown3",
            label.size = 3,
            label.min = TRUE);p

p <- p + scale_color_manual(values = mycola) + 
  scale_fill_manual(values = mycola)+ 
  theme(axis.text=element_text(size=8), panel.grid=element_blank(),
        strip.background = element_rect(colour=NA,fill="grey33"),
        strip.text.x = element_text(face="bold"))+
  theme(text=element_text(size=14,  family="sans"))+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=6))+
  theme(legend.position="none")+
  labs(tag = "A")
print(p)

tiff("p.tiff", units="in", width=10, height=6, res=300)
ggsave("p.TIFF", plot = p)

######################
p2<- plot_alpha_rcurve(ps2, index="observed",
                     subsamples = subsamples,
                     lower.conf = 0.025, 
                     upper.conf = 0.975,
                     group="English_Name_2",
                     label.color = "brown3",
                     label.size = 3,
                     label.min = TRUE) 

p2 <- p2 + scale_color_manual(values = mycola) + 
  scale_fill_manual(values = mycola)+
  theme(axis.text=element_text(size=8), panel.grid=element_blank(),
        strip.background = element_rect(colour=NA,fill="grey33"),
        strip.text.x = element_text(face="bold"))+
  theme(text=element_text(size=16,  family="sans"))+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=6))+
  theme(legend.position="none")+labs(tag = "")
print(p2)

tiff("p2.tiff", units="in", width=10, height=6, res=300)
ggsave("p2.TIFF", plot = p2)


tiff("pd_Simpson_eveness_Effective_alpha_diversity.tiff", units="in", width=20, height=20, res=300)
pd_Simpson_eveness_Effective_alpha_diversity.tiff=(p2/plot_PD/plot_Simpson/plot_evenness/plot_effective_richness/plot_effective_Shannon/plot_effective_Simpson);pd_Simpson_eveness_Effective_alpha_diversity.tiff
ggsave("pd_Simpson_eveness_Effective_alpha_diversity.tiff", plot = pd_Simpson_eveness_Effective_alpha_diversity.tiff)
dev.off()
```


```{r,echo=FALSE,warning=FALSE}
###########################################################################################################################
#Production site
plot_richness_prod_site=ggboxplot(alpha_1000_PD, x = "Production_site_2", y = "Richness", fill= "Production_site_2",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "A");plot_richness_prod_site

#Effective_alpha_diversity
plot_effective_richness_prod_site=ggboxplot(alpha_1000_PD, x = "Production_site_2", y = "Effective_Richness", fill= "Production_site_2",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "B");plot_effective_richness_prod_site

#SHANON DIVERSITY
plot_Shannon_prod_site=ggboxplot(alpha_1000_PD, x = "Production_site_2", y = "Shannon_Index", fill= "Production_site_2", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "C");plot_Shannon_prod_site


plot_effective_Shannon_prod_site=ggboxplot(alpha_1000_PD, x = "Production_site_2", y = "Shannon_Effective", fill= "Production_site_2",  ylab = "Eff. Shannon div.", xlab = "Region of seed of production")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('gray', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_text( colour="gray33", size=17),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_prod_site

#Phylogenetic diversity
plot_PD_prod_site=ggboxplot(alpha_1000_PD, x = "Production_site_2", y = "PD", fill= "Production_site_2", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "E");plot_PD_prod_site


plot_Simpson_prod_site=ggboxplot(alpha_1000_PD, x = "Production_site_2", y = "Simpson_Index", fill= "Production_site_2", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +
  #scale_color_gradientn(colours = c('grey', 'pink'))+
 scale_color_viridis_b() +
   theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "F");plot_Simpson_prod_site


plot_effective_Simpson_prod_site=ggboxplot(alpha_1000_PD, x = "Production_site_2", y = "Simpson_Effective", fill= "Production_site_2",  ylab = "Effe. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('gray', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_prod_site


#EVENNESS
plot_evenness_prod_site=ggboxplot(alpha_1000_PD, x = "Production_site_2", y = "Evenness", fill= "Production_site_2", ylab = "Evenness", xlab = "Region of seed of production")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_text( colour="gray33", size=17),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "H");plot_evenness_prod_site
#ggsave("plot_evenness.TIFF", plot = plot_evenness)

tiff("richness_shannon_pd_Simpson_eveness_priduction_site.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_priduction_site.tiff=(plot_richness_prod_site+plot_effective_richness_prod_site+plot_Shannon_prod_site+plot_effective_Shannon_prod_site+plot_PD_prod_site+plot_Simpson_prod_site+plot_effective_Simpson_prod_site+plot_evenness_prod_site+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_priduction_site.tiff
ggsave("richness_shannon_pd_Simpson_eveness_priduction_site.tiff", plot = richness_shannon_pd_Simpson_eveness_priduction_site.tiff)
dev.off()
###########################################################################################################################

#Dataframe
alpha_1000_PD_Fruit_color=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_color),];alpha_1000_PD_Fruit_color
#####################################################################################################################################################
myco_col=c("chartreuse4","#FE7F9D","darkturquoise","#F0E442", "#009E73");myco_col
#FRUIT COLOR FRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLOR

#alpha_1000_PD_Fruit_color
plot_richness_Fruit_color=ggboxplot(alpha_1000_PD_Fruit_color, x = "Fruit_color", y = "Richness", fill= "Fruit_color",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_Fruit_color

#Effective_alpha_diversity
plot_effective_richness_Fruit_color=ggboxplot(alpha_1000_PD_Fruit_color, x = "Fruit_color", y = "Effective_Richness", fill= "Fruit_color",  ylab = "Eff.Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_Fruit_color

#SHANON DIVERSITY
plot_Shannon_Fruit_color=ggboxplot(alpha_1000_PD_Fruit_color, x = "Fruit_color", y = "Shannon_Index", fill= "Fruit_color", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_Fruit_color


plot_effective_Shannon_Fruit_color=ggboxplot(alpha_1000_PD_Fruit_color, x = "Fruit_color", y = "Shannon_Effective", fill= "Fruit_color",  ylab = "Eff. Shannon div.", xlab = "Berry Color")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_Fruit_color

#Phylogenetic diversity
plot_PD_Fruit_color=ggboxplot(alpha_1000_PD_Fruit_color, x = "Fruit_color", y = "PD", fill= "Fruit_color", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_Fruit_color

plot_Simpson_Fruit_color=ggboxplot(alpha_1000_PD_Fruit_color, x = "Fruit_color", y = "Simpson_Index", fill= "Fruit_color", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_Fruit_color

plot_effective_Simpson_Fruit_color=ggboxplot(alpha_1000_PD_Fruit_color, x = "Fruit_color", y = "Simpson_Effective", fill= "Fruit_color",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_Fruit_color

#EVENNESS
plot_evenness_Fruit_color=ggboxplot(alpha_1000_PD_Fruit_color, x = "Fruit_color", y = "Evenness", fill= "Fruit_color", ylab = "Evenness", xlab = "Berry Color")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_Fruit_color

tiff("richness_shannon_pd_Simpson_eveness_Fruit_color.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_color.tiff=(plot_richness_Fruit_color+plot_effective_richness_Fruit_color+plot_Shannon_Fruit_color+plot_effective_Shannon_Fruit_color+plot_PD_Fruit_color+plot_Simpson_Fruit_color+plot_effective_Simpson_Fruit_color+plot_evenness_Fruit_color+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_color.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_color.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_color.tiff)
dev.off()

##################################################################################################################
alpha_1000_PD_Fruit_taste=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_taste),];alpha_1000_PD_Fruit_taste
myco_taste=c("chartreuse4","#FE7F9D");myco_taste
##############################################################################################################################################################################################################################################
plot_richness_Fruit_taste=ggboxplot(alpha_1000_PD_Fruit_taste, x = "Fruit_taste", y = "Richness", fill= "Fruit_taste",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_Fruit_taste


#Effective_alpha_diversity
plot_effective_richness_Fruit_taste=ggboxplot(alpha_1000_PD_Fruit_taste, x = "Fruit_taste", y = "Effective_Richness", fill= "Fruit_taste",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_Fruit_taste

#SHANON DIVERSITY
plot_Shannon_Fruit_taste=ggboxplot(alpha_1000_PD_Fruit_taste, x = "Fruit_taste", y = "Shannon_Index", fill= "Fruit_taste", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_Fruit_taste


plot_effective_Shannon_Fruit_taste=ggboxplot(alpha_1000_PD_Fruit_taste, x = "Fruit_taste", y = "Shannon_Effective", fill= "Fruit_taste",  ylab = "Eff. Shannon div.", xlab = "Berry taste")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_Fruit_taste

#Phylogenetic diversity
plot_PD_Fruit_taste=ggboxplot(alpha_1000_PD_Fruit_taste, x = "Fruit_taste", y = "PD", fill= "Fruit_taste", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_Fruit_taste


plot_Simpson_Fruit_taste=ggboxplot(alpha_1000_PD_Fruit_taste, x = "Fruit_taste", y = "Simpson_Index", fill= "Fruit_taste", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_Fruit_taste

plot_effective_Simpson_Fruit_taste=ggboxplot(alpha_1000_PD_Fruit_taste, x = "Fruit_taste", y = "Simpson_Effective", fill= "Fruit_taste",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_Fruit_taste

#EVENNESS
 
plot_evenness_Fruit_taste=ggboxplot(alpha_1000_PD_Fruit_taste, x = "Fruit_taste", y = "Evenness", fill= "Fruit_taste", ylab = "Evenness", xlab = "Berry taste")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_Fruit_taste

tiff("richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff=(plot_richness_Fruit_taste+plot_effective_richness_Fruit_taste+plot_Shannon_Fruit_taste+plot_effective_Shannon_Fruit_taste+plot_PD_Fruit_taste+plot_Simpson_Fruit_taste+plot_effective_Simpson_Fruit_taste+plot_evenness_Fruit_taste+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff)
dev.off()
##############################################################################################################################################################################################################################################

alpha_1000_PD_Fruit_shape=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_shape),];alpha_1000_PD_Fruit_shape
myco_shp=c("chartreuse4","#FE7F9D","darkturquoise");myco_shp
##############################################################################################################################################################################################################################################
alpha_1000_PD_Fruit_shape=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_shape),];alpha_1000_PD_Fruit_shape
tiff("richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff", units="in", width=20, height=20, res=300)
plot_richness_Fruit_shape=ggboxplot(alpha_1000_PD_Fruit_shape, x = "Fruit_shape", y = "Richness", fill= "Fruit_shape",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_Fruit_shape

#Effective_alpha_diversity
plot_effective_richness_Fruit_shape=ggboxplot(alpha_1000_PD_Fruit_shape, x = "Fruit_shape", y = "Effective_Richness", fill= "Fruit_shape",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_Fruit_shape

#SHANON DIVERSITY
plot_Shannon_Fruit_shape=ggboxplot(alpha_1000_PD_Fruit_shape, x = "Fruit_shape", y = "Shannon_Index", fill= "Fruit_shape", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_Fruit_shape


plot_effective_Shannon_Fruit_shape=ggboxplot(alpha_1000_PD_Fruit_shape, x = "Fruit_shape", y = "Shannon_Effective", fill= "Fruit_shape",  ylab = "Eff. Shannon div.", xlab = "Berry shape")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_Fruit_shape

#Phylogenetic diversity
plot_PD_Fruit_shape=ggboxplot(alpha_1000_PD_Fruit_shape, x = "Fruit_shape", y = "PD", fill= "Fruit_shape", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_Fruit_shape


plot_Simpson_Fruit_shape=ggboxplot(alpha_1000_PD_Fruit_shape, x = "Fruit_shape", y = "Simpson_Index", fill= "Fruit_shape", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_Fruit_shape


plot_effective_Simpson_Fruit_shape=ggboxplot(alpha_1000_PD_Fruit_shape, x = "Fruit_shape", y = "Simpson_Effective", fill= "Fruit_shape",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_Fruit_shape

#EVENNESS
plot_evenness_Fruit_shape=ggboxplot(alpha_1000_PD_Fruit_shape, x = "Fruit_shape", y = "Evenness", fill= "Fruit_shape", ylab = "Evenness", xlab = "Berry shape")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_Fruit_shape

tiff("richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff=(plot_richness_Fruit_shape+plot_effective_richness_Fruit_shape+plot_Shannon_Fruit_shape+plot_effective_Shannon_Fruit_shape+plot_PD_Fruit_shape+plot_Simpson_Fruit_shape+plot_effective_Simpson_Fruit_shape+plot_evenness_Fruit_shape+ plot_layout(ncol= 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff)
dev.off()
###########################################################################################################################
alpha_1000_PD_TMV_resistance=alpha_1000_PD[!is.na(alpha_1000_PD$TMV_resistance),];alpha_1000_PD_TMV_resistance
myco_TMV=c("chartreuse4","#FE7F9D","darkturquoise");myco_TMV
###########################################################################################################################
plot_richness_TMV_resistance=ggboxplot(alpha_1000_PD_TMV_resistance, x = "TMV_resistance", y = "Richness", fill= "TMV_resistance",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_TMV_resistance

#Effective_alpha_diversity
plot_effective_richness_TMV_resistance=ggboxplot(alpha_1000_PD_TMV_resistance, x = "TMV_resistance", y = "Effective_Richness", fill= "TMV_resistance",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_TMV_resistance

#SHANON DIVERSITY
plot_Shannon_TMV_resistance=ggboxplot(alpha_1000_PD_TMV_resistance, x = "TMV_resistance", y = "Shannon_Index", fill= "TMV_resistance", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_TMV_resistance


plot_effective_Shannon_TMV_resistance=ggboxplot(alpha_1000_PD_TMV_resistance, x = "TMV_resistance", y = "Shannon_Effective", fill= "TMV_resistance",  ylab = "Eff. Shannon div.", xlab = "TMV resistance")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_TMV_resistance

#Phylogenetic diversity
plot_PD_TMV_resistance=ggboxplot(alpha_1000_PD_TMV_resistance, x = "TMV_resistance", y = "PD", fill= "TMV_resistance", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_TMV_resistance


plot_Simpson_TMV_resistance=ggboxplot(alpha_1000_PD_TMV_resistance, x = "TMV_resistance", y = "Simpson_Index", fill= "TMV_resistance", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_TMV_resistance

plot_effective_Simpson_TMV_resistance=ggboxplot(alpha_1000_PD_TMV_resistance, x = "TMV_resistance", y = "Simpson_Effective", fill= "TMV_resistance",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x =element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_TMV_resistance

#EVENNESS
plot_evenness_TMV_resistance=ggboxplot(alpha_1000_PD_TMV_resistance, x = "TMV_resistance", y = "Evenness", fill= "TMV_resistance", ylab = "Evenness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_TMV_resistance

tiff("richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff=(plot_richness_TMV_resistance+plot_effective_richness_TMV_resistance+plot_Shannon_TMV_resistance+plot_effective_Shannon_TMV_resistance+plot_PD_TMV_resistance+plot_Simpson_TMV_resistance+plot_effective_Simpson_TMV_resistance+plot_evenness_TMV_resistance+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff
ggsave("richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff", plot = richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff)
dev.off()
###########################################################################################################################
alpha_1000_PD_Insect_resistance=alpha_1000_PD[!is.na(alpha_1000_PD$Insect_resistance),];alpha_1000_PD_Insect_resistance
myco_insRes=c("chartreuse4","#FE7F9D","darkturquoise","#F0E442", "#009E73", "#FF0000");myco_insRes
##############################################################################################################################################################################################################################################
plot_richness_Insect_resistance=ggboxplot(alpha_1000_PD_Insect_resistance, x = "Insect_resistance_2", y = "Richness", fill= "Insect_resistance_2",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_Insect_resistance


#Effective_alpha_diversity
plot_effective_richness_Insect_resistance=ggboxplot(alpha_1000_PD_Insect_resistance, x = "Insect_resistance_2", y = "Effective_Richness", fill= "Insect_resistance_2",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_Insect_resistance

#SHANON DIVERSITY
plot_Shannon_Insect_resistance=ggboxplot(alpha_1000_PD_Insect_resistance, x = "Insect_resistance_2", y = "Shannon_Index", fill= "Insect_resistance_2", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_Insect_resistance


plot_effective_Shannon_Insect_resistance=ggboxplot(alpha_1000_PD_Insect_resistance, x = "Insect_resistance_2", y = "Shannon_Effective", fill= "Insect_resistance_2",  ylab = "Eff. Shannon div.", xlab = "Insect resistance")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_Insect_resistance

#Phylogenetic diversity
plot_PD_Insect_resistance=ggboxplot(alpha_1000_PD_Insect_resistance, x = "Insect_resistance_2", y = "PD", fill= "Insect_resistance_2", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_Insect_resistance


plot_Simpson_Insect_resistance=ggboxplot(alpha_1000_PD_Insect_resistance, x = "Insect_resistance_2", y = "Simpson_Index", fill= "Insect_resistance_2", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_Insect_resistance

plot_effective_Simpson_Insect_resistance=ggboxplot(alpha_1000_PD_Insect_resistance, x = "Insect_resistance_2", y = "Simpson_Effective", fill= "Insect_resistance_2",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_Insect_resistance

#EVENNESS
plot_evenness_Insect_resistance=ggboxplot(alpha_1000_PD_Insect_resistance, x = "Insect_resistance_2", y = "Evenness", fill= "Insect_resistance_2", ylab = "Evenness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_Insect_resistance

tiff("richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff=(plot_richness_Insect_resistance+plot_effective_richness_Insect_resistance+plot_Shannon_Insect_resistance+plot_effective_Shannon_Insect_resistance+plot_PD_Insect_resistance+plot_Simpson_Insect_resistance+plot_effective_Simpson_Insect_resistance+plot_evenness_Insect_resistance+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff", plot = richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff)
dev.off()
###########################################################################################################################
alpha_1000_PD_seedweight_1000=alpha_1000_PD[!is.na(alpha_1000_PD$seedweight_1000),];alpha_1000_PD_seedweight_1000
myco_weight=c("#0B775E","#FE7F9D","darkturquoise","#F0E442");myco_weight
##############################################################################################################################################################################################################################################
plot_richness_seedweight_1000=ggboxplot(alpha_1000_PD_seedweight_1000, x = "seedweight_1000", y = "Richness", fill= "seedweight_1000",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_seedweight_1000


#Effective_alpha_diversity
plot_effective_richness_seedweight_1000=ggboxplot(alpha_1000_PD_seedweight_1000, x = "seedweight_1000", y = "Effective_Richness", fill= "seedweight_1000",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_seedweight_1000

#SHANON DIVERSITY
plot_Shannon_seedweight_1000=ggboxplot(alpha_1000_PD_seedweight_1000, x = "seedweight_1000", y = "Shannon_Index", fill= "seedweight_1000", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_seedweight_1000

plot_effective_Shannon_seedweight_1000=ggboxplot(alpha_1000_PD_seedweight_1000, x = "seedweight_1000", y = "Shannon_Effective", fill= "seedweight_1000",  ylab = "Eff. Shannon div.", xlab = "Weight of 1000 seeds (grams)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_seedweight_1000

#Phylogenetic diversity
plot_PD_seedweight_1000=ggboxplot(alpha_1000_PD_seedweight_1000, x = "seedweight_1000", y = "PD", fill= "seedweight_1000", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_seedweight_1000

plot_Simpson_seedweight_1000=ggboxplot(alpha_1000_PD_seedweight_1000, x = "seedweight_1000", y = "Simpson_Index", fill= "seedweight_1000", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_seedweight_1000

plot_effective_Simpson_seedweight_1000=ggboxplot(alpha_1000_PD_seedweight_1000, x = "seedweight_1000", y = "Simpson_Effective", fill= "seedweight_1000",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_seedweight_1000

#EVENNESS
plot_evenness_seedweight_1000=ggboxplot(alpha_1000_PD_seedweight_1000, x = "seedweight_1000", y = "Evenness", fill= "seedweight_1000", ylab = "Evenness", xlab = "Weight of 1000 seeds (grams)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_seedweight_1000

tiff("richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff=(plot_richness_seedweight_1000+plot_effective_richness_seedweight_1000+plot_Shannon_seedweight_1000+plot_effective_Shannon_seedweight_1000+plot_PD_seedweight_1000+plot_Simpson_seedweight_1000+plot_effective_Simpson_seedweight_1000+plot_evenness_seedweight_1000+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff
ggsave("richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff", plot = richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff)
dev.off()
#########################################################################################################################
alpha_1000_PD_yield=alpha_1000_PD[!is.na(alpha_1000_PD$yield),];alpha_1000_PD_yield
myco_yield=c("chartreuse4","#FE7F9D","darkturquoise","#F0E442", "#009E73");myco_yield
#########################################################################################################################
plot_richness_yield=ggboxplot(alpha_1000_PD_yield, x = "yield", y = "Richness", fill= "yield",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_yield


#Effective_alpha_diversity
plot_effective_richness_yield=ggboxplot(alpha_1000_PD_yield, x = "yield", y = "Effective_Richness", fill= "yield",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_yield

#SHANON DIVERSITY
plot_Shannon_yield=ggboxplot(alpha_1000_PD_yield, x = "yield", y = "Shannon_Index", fill= "yield", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_yield


plot_effective_Shannon_yield=ggboxplot(alpha_1000_PD_yield, x = "yield", y = "Shannon_Effective", fill= "yield",  ylab = "Eff. Shannon div.", xlab = "Yield (Kg/Sq.Km)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_yield

#Phylogenetic diversity
plot_PD_yield=ggboxplot(alpha_1000_PD_yield, x = "yield", y = "PD", fill= "yield", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_yield

plot_Simpson_yield=ggboxplot(alpha_1000_PD_yield, x = "yield", y = "Simpson_Index", fill= "yield", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_yield

plot_effective_Simpson_yield=ggboxplot(alpha_1000_PD_yield, x = "yield", y = "Simpson_Effective", fill= "yield",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_yield

#EVENNESS
plot_evenness_yield=ggboxplot(alpha_1000_PD_yield, x = "yield", y = "Evenness", fill= "yield", ylab = "Evenness", xlab = "Yield (Kg/Sq.Km)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_yield

tiff("richness_shannon_pd_Simpson_eveness_yield.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_yield.tiff=(plot_richness_yield+plot_effective_richness_yield+plot_Shannon_yield+plot_effective_Shannon_yield+plot_PD_yield+plot_Simpson_yield+plot_effective_Simpson_yield+plot_evenness_yield+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_yield.tiff
ggsave("richness_shannon_pd_Simpson_eveness_yield.tiff", plot = richness_shannon_pd_Simpson_eveness_yield.tiff)
dev.off()
#########################################################################################################################
#########################################################################################################################
alpha_1000_PD_ovaries=alpha_1000_PD[!is.na(alpha_1000_PD$ovaries),];alpha_1000_PD_ovaries
myco_ovaries=c("chartreuse4","#FE7F9D","darkturquoise");myco_ovaries
#########################################################################################################################
plot_richness_ovaries=ggboxplot(alpha_1000_PD_ovaries, x = "ovaries", y = "Richness", fill= "ovaries",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_ovaries
#ggsave("plot_richness.TIFF", plot = plot_richness)
#+facet_wrap(~ovaries);
#Chao1_simpson.tiff=(plot6 /plot7 );Chao1_simpson.tiff
#ggsave("Chao1_simpson.tiff", plot = Chao1_simpson.tiff)

#Effective_alpha_diversity
plot_effective_richness_ovaries=ggboxplot(alpha_1000_PD_ovaries, x = "ovaries", y = "Effective_Richness", fill= "ovaries",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_ovaries
#ggsave("plot_effective_richness.TIFF", plot = plot_effective_richness)

#SHANON DIVERSITY
plot_Shannon_ovaries=ggboxplot(alpha_1000_PD_ovaries, x = "ovaries", y = "Shannon_Index", fill= "ovaries", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_ovaries
#ggsave("plot_Shannon.TIFF", plot = plot_Shannon)


plot_effective_Shannon_ovaries=ggboxplot(alpha_1000_PD_ovaries, x = "ovaries", y = "Shannon_Effective", fill= "ovaries",  ylab = "Eff. Shannon div.", xlab = "Number of ovaries")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_ovaries
#ggsave("plot_effective_Shannon.TIFF", plot = plot_effective_Shannon)

#Phylogenetic diversity
plot_PD_ovaries=ggboxplot(alpha_1000_PD_ovaries, x = "ovaries", y = "PD", fill= "ovaries", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_ovaries
#ggsave("plot_PD.TIFF", plot = plot_PD)


plot_Simpson_ovaries=ggboxplot(alpha_1000_PD_ovaries, x = "ovaries", y = "Simpson_Index", fill= "ovaries", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_ovaries
#ggsave("plot_Simpson.TIFF", plot = plot_Simpson)

plot_effective_Simpson_ovaries=ggboxplot(alpha_1000_PD_ovaries, x = "ovaries", y = "Simpson_Effective", fill= "ovaries",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_ovaries
#ggsave("plot_effective_Simpson.TIFF", plot = plot_effective_Simpson)

#EVENNESS
 
plot_evenness_ovaries=ggboxplot(alpha_1000_PD_ovaries, x = "ovaries", y = "Evenness", fill= "ovaries", ylab = "Evenness", xlab = "Number of ovaries")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_ovaries
#ggsave("plot_evenness.TIFF", plot = plot_evenness)

tiff("richness_shannon_pd_Simpson_eveness_ovaries.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_ovaries.tiff=(plot_richness_ovaries+plot_effective_richness_ovaries+plot_Shannon_ovaries+plot_effective_Shannon_ovaries+plot_PD_ovaries+plot_Simpson_ovaries+plot_effective_Simpson_ovaries+plot_evenness_ovaries+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_ovaries.tiff
ggsave("richness_shannon_pd_Simpson_eveness_ovaries.tiff", plot = richness_shannon_pd_Simpson_eveness_ovaries.tiff)
dev.off()
##########################################################################################################################
##########################################################################################################################
##########################################################################################################################
##########################################################################################################################
alpha_1000_PD_Fruit_hori_diameter=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_hori_diameter),];alpha_1000_PD_Fruit_hori_diameter
myco_horiz=c("chartreuse4","#FE7F9D","darkturquoise");myco_horiz
#########################################################################################################################
plot_richness_Fruit_hori_diameter=ggboxplot(alpha_1000_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Richness", fill= "Fruit_hori_diameter",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_Fruit_hori_diameter
#ggsave("plot_richness.TIFF", plot = plot_richness)
#+facet_wrap(~Fruit_hori_diameter);
#Chao1_simpson.tiff=(plot6 /plot7 );Chao1_simpson.tiff
#ggsave("Chao1_simpson.tiff", plot = Chao1_simpson.tiff)

#Effective_alpha_diversity
plot_effective_richness_Fruit_hori_diameter=ggboxplot(alpha_1000_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Effective_Richness", fill= "Fruit_hori_diameter",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_Fruit_hori_diameter
#ggsave("plot_effective_richness.TIFF", plot = plot_effective_richness)

#SHANON DIVERSITY
plot_Shannon_Fruit_hori_diameter=ggboxplot(alpha_1000_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Shannon_Index", fill= "Fruit_hori_diameter", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_Fruit_hori_diameter
#ggsave("plot_Shannon.TIFF", plot = plot_Shannon)


plot_effective_Shannon_Fruit_hori_diameter=ggboxplot(alpha_1000_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Shannon_Effective", fill= "Fruit_hori_diameter",  ylab = "Eff. Shannon div.", xlab = "Berry horizontal diameter (cm)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_Fruit_hori_diameter
#ggsave("plot_effective_Shannon.TIFF", plot = plot_effective_Shannon)

#Phylogenetic diversity
plot_PD_Fruit_hori_diameter=ggboxplot(alpha_1000_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "PD", fill= "Fruit_hori_diameter", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_Fruit_hori_diameter
#ggsave("plot_PD.TIFF", plot = plot_PD)


plot_Simpson_Fruit_hori_diameter=ggboxplot(alpha_1000_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Simpson_Index", fill= "Fruit_hori_diameter", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_Fruit_hori_diameter
#ggsave("plot_Simpson.TIFF", plot = plot_Simpson)

plot_effective_Simpson_Fruit_hori_diameter=ggboxplot(alpha_1000_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Simpson_Effective", fill= "Fruit_hori_diameter",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_Fruit_hori_diameter
#ggsave("plot_effective_Simpson.TIFF", plot = plot_effective_Simpson)

#EVENNESS
 
plot_evenness_Fruit_hori_diameter=ggboxplot(alpha_1000_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Evenness", fill= "Fruit_hori_diameter", ylab = "Evenness", xlab = "Berry horizontal diameter (cm)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_Fruit_hori_diameter
#ggsave("plot_evenness.TIFF", plot = plot_evenness)

tiff("richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff=(plot_richness_Fruit_hori_diameter+plot_effective_richness_Fruit_hori_diameter+plot_Shannon_Fruit_hori_diameter+plot_effective_Shannon_Fruit_hori_diameter+plot_PD_Fruit_hori_diameter+plot_Simpson_Fruit_hori_diameter+plot_effective_Simpson_Fruit_hori_diameter+plot_evenness_Fruit_hori_diameter+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff)
dev.off()
#########################################################################################################################
#########################################################################################################################
#########################################################################################################################
#########################################################################################################################
##########################################################################################################################
alpha_1000_PD_Fruit_long_diameter=alpha_1000_PD[!is.na(alpha_1000_PD$Fruit_long_diameter),];alpha_1000_PD_Fruit_long_diameter
myco_longdiam=c("chartreuse4","#00AFBB","#FE7F9D" );myco_longdiam
#########################################################################################################################
plot_richness_Fruit_long_diameter=ggboxplot(alpha_1000_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Richness", fill= "Fruit_long_diameter",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A");plot_richness_Fruit_long_diameter
#ggsave("plot_richness.TIFF", plot = plot_richness)
#+facet_wrap(~Fruit_long_diameter);
#Chao1_simpson.tiff=(plot6 /plot7 );Chao1_simpson.tiff
#ggsave("Chao1_simpson.tiff", plot = Chao1_simpson.tiff)

#Effective_alpha_diversity
plot_effective_richness_Fruit_long_diameter=ggboxplot(alpha_1000_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Effective_Richness", fill= "Fruit_long_diameter",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B");plot_effective_richness_Fruit_long_diameter
#ggsave("plot_effective_richness.TIFF", plot = plot_effective_richness)

#SHANON DIVERSITY
plot_Shannon_Fruit_long_diameter=ggboxplot(alpha_1000_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Shannon_Index", fill= "Fruit_long_diameter", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C");plot_Shannon_Fruit_long_diameter
#ggsave("plot_Shannon.TIFF", plot = plot_Shannon)


plot_effective_Shannon_Fruit_long_diameter=ggboxplot(alpha_1000_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Shannon_Effective", fill= "Fruit_long_diameter",  ylab = "Eff. Shannon div.", xlab = "Berry longitudinal diameter (cm)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D");plot_effective_Shannon_Fruit_long_diameter
#ggsave("plot_effective_Shannon.TIFF", plot = plot_effective_Shannon)

#Phylogenetic diversity
plot_PD_Fruit_long_diameter=ggboxplot(alpha_1000_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "PD", fill= "Fruit_long_diameter", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E");plot_PD_Fruit_long_diameter
#ggsave("plot_PD.TIFF", plot = plot_PD)


plot_Simpson_Fruit_long_diameter=ggboxplot(alpha_1000_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Simpson_Index", fill= "Fruit_long_diameter", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F");plot_Simpson_Fruit_long_diameter
#ggsave("plot_Simpson.TIFF", plot = plot_Simpson)

plot_effective_Simpson_Fruit_long_diameter=ggboxplot(alpha_1000_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Simpson_Effective", fill= "Fruit_long_diameter",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G");plot_effective_Simpson_Fruit_long_diameter
#ggsave("plot_effective_Simpson.TIFF", plot = plot_effective_Simpson)

#EVENNESS
 
plot_evenness_Fruit_long_diameter=ggboxplot(alpha_1000_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Evenness", fill= "Fruit_long_diameter", ylab = "Evenness", xlab = "Berry longitudinal diameter (cm)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H");plot_evenness_Fruit_long_diameter
#ggsave("plot_evenness.TIFF", plot = plot_evenness)

tiff("richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff=(plot_richness_Fruit_long_diameter+plot_effective_richness_Fruit_long_diameter+plot_Shannon_Fruit_long_diameter+plot_effective_Shannon_Fruit_long_diameter+plot_PD_Fruit_long_diameter+plot_Simpson_Fruit_long_diameter+plot_effective_Simpson_Fruit_long_diameter+plot_evenness_Fruit_long_diameter+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff)
dev.off()

```

```{r}

```





