---
title: "Exploring plasticity of the tomato seed microbiome"
author: "Expedito Olimi"
date: "2025-01-05"
output: html_document
---
#Load all the packages
#set working directory

#myworking director(wd)=/Users/expeditoolimi/Documents/Southampton/Global_tomato_seed_Microbiome/Tomato100_Project/Tomato-seed-microbiome

#replace this with your working directory
```{r, echo=F, warning= F}
setwd("wd")
library(phyloseq);packageVersion("phyloseq")#‘1.46.0’
library(vegan);packageVersion("vegan")# ‘2.6.4’
library(genefilter);packageVersion("genefilter")#‘1.84.0’
library(psych);packageVersion("psych")#‘2.4.1’
library(DESeq2);packageVersion("DESeq2")#‘1.42.1’
library(ggplot2);packageVersion("ggplot2")#‘3.5.0’
library(reshape2);packageVersion("reshape2")#‘1.4.4’
library(microbiome);packageVersion("microbiome")#‘1.24.0’
library(ranacapa);packageVersion("ranacapa")#‘0.1.0’
library(S4Vectors);packageVersion("S4Vectors")#‘0.40.2’
library(Biobase);packageVersion("Biobase")#‘2.62.0’
library(BiocGenerics);packageVersion("BiocGenerics")#‘0.48.1’
library(parallel);packageVersion("parallel")#‘4.3.2’
library(ggpubr);packageVersion("ggpubr")#‘0.6.0’
#library(metagMisc)
library(decontam);packageVersion("decontam")#‘1.22.0’
library(RColorBrewer);packageVersion("RColorBrewer")#‘1.1.3’
library(multcompView);packageVersion("multcompView")#‘0.1.10’
library(metagenomeSeq);packageVersion("metagenomeSeq")#‘1.43.0’
#library(devtools)
library(lawstat);packageVersion("lawstat")#‘3.6’
library(DirichletMultinomial);packageVersion("DirichletMultinomial")#‘1.44.0’
library(lattice);packageVersion("lattice")# ‘0.22.5’
library(xtable);packageVersion("xtable")#‘1.8.4’
library(permute);packageVersion("permute")#‘0.9.7’
library(grid);packageVersion("grid")#‘4.3.2’
library("doParallel");packageVersion("doParallel")#‘1.0.17’
#library(igraph)
library(ade4);packageVersion("ade4")#‘1.7.22’
#library(rgl)
library(car);packageVersion("car")#‘3.1.2’
library(gridExtra);packageVersion("gridExtra")# ‘2.3’
library(tidyverse);packageVersion("tidyverse")#‘2.0.0’
library(phia);packageVersion("phia")#‘0.3.1’
library(lme4);packageVersion("lme4")#‘1.1.35.1’
library(broom);packageVersion("broom")#‘1.0.5’
library(AICcmodavg);packageVersion("AICcmodavg")#‘2.3.3’
library(multcomp);packageVersion("multcomp")#‘1.4.25’
library(rcompanion);packageVersion("rcompanion")#‘2.4.35’
library(hrbrthemes);packageVersion("hrbrthemes")#‘0.8.7’
library(viridis);packageVersion("viridis")#‘0.6.5’
library(ape);packageVersion("ape")#‘5.7.1’#loading the tree
library(microbiomeutilities);packageVersion("microbiomeutilities")#‘1.0.17’#process taxonomy table
library(microViz);packageVersion("microViz")#‘0.12.1’#clean taxonomy table to be cited
library(GUniFrac);packageVersion("GUniFrac")#‘1.8’
library(ggtext) ;packageVersion("ggtext")# ‘0.1.2’for rotated labels on ord_plot() 
library(ggraph);packageVersion("ggraph")# ‘2.2.1’for taxatree_plots()
library(DT);packageVersion("DT")# ‘0.32’for tax_fix_interactive()
library(corncob);packageVersion("corncob")#‘0.4.1’ for beta binomial models in tax_model()
library(VennDiagram);packageVersion("VennDiagram")#‘1.7.3’
library(UpSetR);packageVersion("UpSetR")#‘1.4.0’
library(rstatix);packageVersion("rstatix")#‘0.7.2’
library(ALDEx2);packageVersion("ALDEx2")#[1] ‘1.34.0’
library(cowplot);packageVersion("cowplot")#‘1.1.3’
library(HMP);packageVersion("HMP")#HMP: Hypothesis Testing & Power Calculations for Comparing Metagenomic Samples from HMP #‘2.0.1’
library(dendextend);packageVersion("dendextend")#‘1.17.1’
library(dplyr)
library(gcookbook)
library(tidyverse)
library(patchwork)
library(MicrobiotaProcess)
library(rmarkdown)#install.packages("rmarkdown")
library(dada2)
library(Biostrings)
library(ggalluvial)
library(ggh4x)
library(tidyverse)
library(ggrepel)
library(qiime2R)#https://github.com/jbisanz/qiime2R
library(dplyr)
library(pals)
library(ggforce)
library(gridExtra)
library(vegan)
library(data.table)
library(picante)
library(ggtreeExtra)
library(ggpmisc)
library(DECIPHER) #BiocManager::install("DECIPHER")
library(phangorn) #remotes::install_github("KlausVigo/phangorn")
library(ggtree) #BiocManager::install("ggtree")
library(plotly)
library(speedyseq) #remotes::install_github("mikemc/speedyseq")
library(ComplexHeatmap)
library(agricolae)
library(scales)
library(ggpmisc)
library(pairwiseAdonis)#library(devtools) install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
#remotes::install_github("mahendra-mariadassou/phyloseq-extended", ref = "dev")
library(phyloseq.extended)#remotes::install_github(repo = "mahendra-mariadassou/phyloseq-extended@v0.1.0.9000-beta", ref = "dev")

```

#load the scripts 
```{r}
setwd("wd")
source("aggregate_top_taxa.txt")
source("PD.txt")
source("aggregate_rare.txt")
source("plot_composition.txt")
source("RSCRIPT_parwise.adonis.txt")
source("plot_hierarchy.R")
```


#Load colors
```{r,echo = F,warning = FALSE}
mycolors=c("grey35","#00AFBB","goldenrod","#0dba2f","#392682","coral3","#8E9CA3",
           "chartreuse4","grey85","#FF2700","bisque4","#0B775E","darkred","#F39B7FB2","#FA41CA",
           "steelblue","blue","#D16103", "#C3D7A4", "#52854C","deepskyblue","#293352","#FFDB6D",
           "Yellow","#CC79A7","#392682", "#CAB2D6","maroon","orchid1","khaki2","#86486F","#7E6148B2");mycolors

mycola=c("grey35","chartreuse4","#FE7F9D","darkturquoise", "#F0E442","#009E73","#FF0000",
         "Orange","blue","#850dba","green1","#FF6AD5","#6A3D9A","Yellow","Blue",
         "deeppink","#1C0221","#00AFBB","steelblue","#D16103", "#C3D7A4","#52854C",
         "#4E84C4","#293352","#FFDB6D","#CC79A7","#C4961A", "#F4EDCA","dodgerblue2", 
         "#E31A1C","green4","#0dba2f","black","gold1", "skyblue2","#FB9A99","#CAB2D6",
         "blue1","steelblue4","Sky Blue","grey85","#3B9AB2","goldenrod","yellow3",
         "darkorange4","brown","#999999", "#E69F00","#56B4E9","#009E73","#F0E442",
         "#D55E00", "#E69F00","#ba0d42","yellow4","#D55E00", "#CC79A7","darkred",
         "#56B4E9","#86486F","#446455", "bisque4","#F7B1AB","#807182","#E3DDDD",
         "#A45C3D","coral3","#000099","#E2C59F","#B6C5CC", "#8E9CA3","#556670",
         "#771C19", "#AA3929","#E25033","#F27314", "#F8A31B","#2D2D2D","#91D1C2B2",
         "#DC0000B2","#7E6148B2","#E64B35B2", "#4DBBD5B2","#00A087B2","#3C5488B2",
         "#F39B7FB2","#8491B4B2","#FF2700","#62BBA5","#B58900","#80CDC1","#8073AC",
     "#0072B2","#392682","#FA41CA","#0B775E","#02401B","#F4B5BD","#B6854D","deepskyblue");mycola

mycol = c("bisque4", "deeppink", "chartreuse4","blue","coral3","goldenrod","deepskyblue","darkred","#0dba2f", "Orange","Yellow", "#850dba","Blue", "Sky Blue", "#ba0d42");mycol
```


#Load data (ASV table, taxonomy & tree, & metadata)
```{r,echo = F}
##Imported feature table
asv_tab <- read.table("wd/Table.txt", 
              header=TRUE,
              row.names="OTUID");asv_tab
asv_tab <- otu_table(asv_tab, taxa_are_rows = TRUE);asv_tab 

## Imported taxonomy table
tax_tab <- read.table("wd/Taxonomy.txt", 
              header=TRUE,
              sep="\t",
              row.names="OTUID");tax_tab
tax_tab <- as.matrix(tax_tab);tax_tab
tax_tab <- tax_table(tax_tab);tax_tab

## Imported metadata file
metadata <- read.csv("wd/Metadata.csv",
              header=TRUE,
              row.names="SampleID");metadata
#rownames(metadata) <- metadata[,1]
metadata <- sample_data(metadata);metadata

#import rooted tree
rtree <- read.tree(file = "wd/tree.nwk");rtree
tree <- phy_tree(rtree)

#import reference sequences
## Import reference sequences
# rs_table <- Biostrings::readDNAStringSet(
#   "outputs/qiime2/dna-sequences.fasta",
#   format = "fasta",
#   nrec = -1L,
#   skip = 0L,
#   seek.first.rec = F,
#   use.names = T,
#   with.qualities = F)

#proof that the data are of equal dimensions
all(rownames(metadata) %in% colnames(asv_tab))

#Convert to phyloseq object
ps <- merge_phyloseq(
  asv_tab,
  tax_tab, 
  metadata,
  tree);ps#[ 16460 taxa & 1197 samples ]
summarize_phyloseq(ps)
ps0 <- ps # 16460 taxa & 1197 samples
save(ps0,
file = "wd/ps0.RData")

```


#clean the data
```{r,echo = FALSE,warning = FALSE}
load("wd/ps0.RData")

ps_clean = subset_taxa(ps0, Kingdom=="Bacteria");ps_clean# [ 6448 taxa & 84 samples ]
ps_clean = subset_taxa(ps_clean, Phylum!="Cyanobacteria");ps_clean#  [ 15588 taxa & 1197 samples ]
ps_clean = subset_taxa(ps_clean, Order!="Chloroplast");ps_clean#  [ 15588 taxa & 1197 samples ]
ps_clean = subset_taxa(ps_clean, Family!="Mitochondria");ps_clean# [ 15588 taxa & 1197 samples ]
ps_clean <- prune_taxa(taxa_sums(ps_clean)>0, ps_clean);ps_clean# [ 15588 taxa & 1197 samples ]

ps <- ps_clean

sort(sample_sums(ps))
summary(sample_sums(ps)) # mean reads/sample = 22791
summarize_phyloseq(ps) # Average reads = 22791, Total reads = 27281011, Singletons = 38
taxa_names(ps)[1:5]
sample_data(ps)$total_reads <- sample_sums(ps)

save(ps, file = "wd/ps.RData")

K <- as.data.frame(table(tax_table(ps)[,"Kingdom"]));K
P <- as.data.frame(table(tax_table(ps)[,"Phylum"]));P # 38 phyla
C <- as.data.frame(table(tax_table(ps)[,"Class"]));C # 109 classes
O <- as.data.frame(table(tax_table(ps)[,"Order"]));O # 261 orders
Fa <- as.data.frame(table(tax_table(ps)[,"Family"]));Fa # 476 families
G <- as.data.frame(table(tax_table(ps)[,"Genus"]));G # 1187 genera
S <- as.data.frame(table(tax_table(ps)[,"Species"]));S # 1264 species

microbiome::summarize_phyloseq(ps)
#Number of singletons = 68"
#Sparsity = 0.995179763625127"
#Average number of reads = 22794.9197994987"
# Total number of reads = 27281011"
#Max. number of reads = 225892"
#Min. number of reads = 68"
taxonomy <- as.vector(phyloseq::tax_table(ps))
sums_ps <- sample_sums(ps);sums_ps
sums_ps <- data.frame(sums_ps);sums_ps
sums_ps <- as.matrix(sums_ps);sums_ps
colSums(sums_ps) #  27281011      
max(sums_ps) #225892
min(sums_ps) # 68
sums_ps
```
#Filter data at 0.025 prevalence

#Obtained from: https://f1000research.com/articles/5-1492/v1;#One of the reasons to filter by prevalence is to avoid spending #much time analyzing taxa that were only rarely seen. This also turns out to be a useful filter of noise (taxa that are #actually just artifacts of the data collection process)
# Define prevalence of each taxa; # (in how many samples did each taxa appear at least once)

```{r,echo = FALSE,warning = FALSE}
load("wd/ps.RData")

prev0 = apply(X = otu_table(ps),
              MARGIN = ifelse(taxa_are_rows(ps), yes = 1, no = 2),
              FUN = function(x){sum(x > 0)})

prevdf = data.frame(Prevalence = prev0,
                    TotalAbundance = taxa_sums(ps),
                    tax_table(ps))# the most prevalent ASVs present therein the composition

#only for reference, in case we want to filter Phyla that appear less than X times
keepPhyla = table(prevdf$Phylum)[(table(prevdf$Phylum) > 0)]

prevdf1 = subset(prevdf, Phylum %in% names(keepPhyla))

# Define prevalence threshold as 0.25% of total samples# 
prevalenceThreshold = 0.0025 * nsamples(ps)
prevalenceThreshold

# Execute prevalence filter, using `prune_taxa()` function
ps1 = prune_taxa((prev0 > prevalenceThreshold), ps);ps1

# Filter entries with unidentified Phylum.
ps2 = subset_taxa(ps1, Phylum %in% names(keepPhyla));ps2

#Plot prevalence of different phyla
#https://github.com/joey711/phyloseq/issues/1474

ggplot(prevdf1, aes(TotalAbundance, Prevalence, color = Phylum)) +
  geom_hline(yintercept = prevalenceThreshold, alpha = 0.5, linetype = 2) +
  geom_point(size = 2, alpha = 0.65) +
  scale_y_log10() + scale_x_log10() +
  ggtitle(paste0("Prevalence filtering, using a threshold of ", prevalenceThreshold))+
  xlab("Total Abundance") +
  scale_color_manual(values=ocean.balance(38))+
  theme_bw()+
  facet_wrap(~Phylum, nrow = 4)

print(paste0("Initial #ASVs:", ntaxa(ps), ". #ASVs after prevalence filtering:", ntaxa(ps2), ". #ASVs removed: ", ntaxa(ps)-ntaxa(ps2)))
```


# save 
```{r}
save(ps2, file = "wd/ps2.RData")
```


----
#Figure S1A
----
###plot rarefaction curves 
#code based on https://microsud.github.io/microbiomeutilities/articles/microbiomeutilities.html
```{r, echo=FALSE,warning = FALSE}
load("wd/ps2.RData")
library(microbiomeutilities)
# set seed
set.seed(1)
subsamples <- seq(0, 50000, by=500)[-1]
#subsamples = c(10, 5000, 10000, 20000, 30000)
############
p<- plot_alpha_rcurve(ps2, index="observed",
                     subsamples = subsamples,
                     lower.conf = 0.025, 
                     upper.conf = 0.975,
                     group="English_Name_2",
                     label.color = "brown3",
                     label.size = 3,
                     label.min = TRUE) 

p <- p + scale_color_manual(values = mycola) + 
    scale_fill_manual(values = mycola)+
    theme(axis.text=element_text(size=8), panel.grid=element_blank(),
    strip.background = element_rect(colour=NA,fill="grey33"),
    strip.text.x = element_text(face="bold"))+
    theme(text=element_text(size=16,  family="sans"))+
    guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=6))+
    theme(legend.position="none")+labs(tag = "")
    print(p)
    tiff("p.tiff", units="in", width=10, height=6, res=300)
    ggsave("p.TIFF", plot = p)
```
----
#Fig.13 A $ B
----
#Plot correlation of shannon index & total reads (observed ASVs) [since this is common practice]
```{r, echo=FALSE, warning=FALSE}
load("wd/ps2.RData")

corr1=ggplot(data = data.frame("total_reads" =  phyloseq::sample_sums(ps2),
    "observed" = phyloseq::estimate_richness(ps2,
    measures = "Observed")[, 1]),
    aes(x = total_reads, y = observed)) +
    geom_point() +
    geom_smooth(method="lm", se = FALSE) +
    labs(x = "\nTotal Reads", y = "Observed Richness\n")+
    theme(panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(),
    panel.background = element_rect(colour = "gray33", size=0.4))+ 
    theme(axis.title.x = element_text(face="bold", colour="gray33", size=15),
    axis.title.y = element_text(face="bold", colour="gray33", size=15),
    axis.text.x = element_text(colour="gray33", size=15),
    axis.text.y  = element_text(colour="gray33", size=15))+
    labs(tag = "A");corr1

#############
corr2=ggplot(data = data.frame("total_reads" =  phyloseq::sample_sums(ps2),
    "Shannon" = phyloseq::estimate_richness(ps2, measures = "Shannon")[, 1]),
    aes(x = total_reads, y = Shannon)) +
    geom_point() +
    geom_smooth(method="lm", se = FALSE) +
    labs(x = "\nTotal Reads", y = "Shannon index\n")+
    theme(panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(),
    panel.background = element_rect(colour = "gray33", size=0.4))+ 
    theme(axis.title.x = element_text(face="bold", colour="gray33", size=15),
    axis.title.y = element_text(face="bold", colour="gray33", size=15),
    axis.text.x = element_text(colour="gray33", size=15),
    axis.text.y  = element_text(colour="gray33", size=15))+
    labs(tag = "B");corr2

tiff("correlation.tiff", units="in", width=12, height=6, res=300)
correlation.tiff=corr1+corr2+plot_layout(ncol=2);correlation.tiff
ggsave("correlation.tiff", plot = correlation.tiff)
dev.off()
```

#Fig.2A
#Transformed to relative abundances 
#bar blot of the seed microbiome composition in 
#the different cukltivars at the various locations for the top500 ASVs
```{r,echo=FALSE,warning = FALSE}
load("wd/ps2.RData")

# Bar plots:
#Showing individual replicates. Top 500 taxa
    toptax = names(sort(taxa_sums(ps2), decreasing=TRUE))[1:500];toptax
    ps.toptax = transform_sample_counts(ps2, function(x) {x/sum(x)*100});ps.toptax
    ps.toptax <- prune_taxa(toptax, ps.toptax);ps.toptax

#Class level
    psmelt(ps.toptax)
    gg1=ggplot(psmelt(ps.toptax), aes(x=reorder(Sample, English_Name), y=Abundance, fill=Class))+
    geom_bar(stat = "identity", position = "stack", width = 1) +
    scale_fill_manual(values=c(ocean.dense(7), ocean.tempo(8)))+
    #scale_fill_manual(values=c("#25725d", "#439b5b", "#7bc369", "#abd69a", "#d8e8b0",         "#cccda7", "#bfb77b", "#7e9998", "#92B2B2", "#83bbbf",     "#61acc1", "#58a0c4",       "#7087c3","#8973b4", "#512972"))+
    #scale_fill_manual(values = mycola)+
    xlab("") + ylab("Relative abundance (%)") +
    facet_wrap(~reorder(Production_site_2, Sample), scales = "free_x", nrow = 1) + 
    ggtitle("Class, top 500 ASVs") +
    theme_pubclean() + theme(legend.position="bottom", axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=0)) +
    theme(legend.text = element_text(face = c(rep("italic", 5), rep("plain", 5))))+
    guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow = 2))+
    theme(text=element_text(size=8,  family="Arial"))+ labs(tag = "");gg1

    library(ggpubr)
    tiff("gg1_Region_cultivar.tiff", units="in", width=10, height=6, res=300)
    gg1_Region_cultivar.tiff=gg1+plot_layout(nrow = 1);gg1_Region_cultivar.tiff
    ggsave("gg1_Region_cultivar.tiff", plot = gg1_Region_cultivar.tiff)
    dev.off()
```

#Average percentage composition class
```{r,echo=FALSE,warning=FALSE,message=FALSE}
 load("wd/ps.RData")
#Convert to relative abundance or the computation of the micribial composition in the samples
#consider phylum
GPf_Phylum = tax_glom(ps, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})
df_Phylum = data.frame(Phylum = tax_table(GPf_Phylum)[,"Phylum"], 
Mean = rowMeans(otu_table(GPf_Phylum)), row.names = NULL)
df_Phylum = df_Phylum[order(-df_Phylum$Mean),];df_Phylum
head(df_Phylum,20)

#consider Class
GPf_Class = tax_glom(ps, "Class") %>% transform_sample_counts(function(x) {x * 100/sum(x)})
df_Class = data.frame(Class = tax_table(GPf_Class)[,"Class"], 
Mean = rowMeans(otu_table(GPf_Class)), row.names = NULL)
df_Class = df_Class[order(-df_Class$Mean),];df_Class
head(df_Class,20)

#consider Order
GPf_Order = tax_glom(ps, "Order") %>% transform_sample_counts(function(x) {x * 100/sum(x)})
df_Order = data.frame(Order = tax_table(GPf_Order)[,"Order"],
Mean = rowMeans(otu_table(GPf_Order)), row.names = NULL)
df_Order = df_Order[order(-df_Order$Mean),];df_Order
head(df_Order,20)

#consider Family
GPf_Family = tax_glom(ps, "Family") %>% transform_sample_counts(function(x) {x * 100/sum(x)})
df_Family = data.frame(Family = tax_table(GPf_Family)[,"Family"],
Mean = rowMeans(otu_table(GPf_Family)), row.names = NULL)
df_Family = df_Family[order(-df_Family$Mean),];df_Family
head(df_Family,20)

#consider Genus
GPf_Genus = tax_glom(ps, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)})
df_Genus = data.frame(Genus = tax_table(GPf_Genus)[,"Genus"],
Mean = rowMeans(otu_table(GPf_Genus)), row.names = NULL)
df_Genus = df_Genus[order(-df_Genus$Mean),];df_Genus
head(df_Genus,20)
```

#Beta-diversity dispersion
#Hellinger transformed data for beta diversity analyses (i.e., transformation:method="hellinger")
```{r, echo=FALSE, warning=FALSE}
#Hellinger transform the data
load("wd/ps2.RData")
(ps_hellinger <- microbiome::transform(ps2, "hellinger"))  
save(ps_hellinger, file = "wd/ps_hellinger.RData")

#Load the hellinger transformed data
load ("wd/ps_hellinger.RData")
#Generate distance matrix
hellinger_dist_matrix <- phyloseq::distance(ps_hellinger, method = "bray");hellinger_dist_matrix
#The dispersion attributed to variation in culltivars 
dispr <- vegan::betadisper(hellinger_dist_matrix, phyloseq::sample_data(ps_hellinger)$English_Name_2);dispr #calculate the dispersion

data_fram_cultivar <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_cultivar
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_cultivar)
#data:  Richness by Production_site_2
#Kruskal-Wallis chi-squared = 638.86, df = 99, p-value < 2.2e-16

library(FSA)
# Dunn's Test 
Dunn_dispr_culrivar=dunn_test(data_fram_cultivar, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_culrivar
write_csv(Dunn_dispr_culrivar, "Dunn_dispr_culrivar.csv")
Dunn_dispr_culrivar <- dunnTest(data_fram_cultivar$Distance_to_centroid, data_fram_cultivar$Group, method = "bonferroni");Dunn_dispr_culrivar
# Obtaining significance letters
library(rcompanion)
#CLD_Dunn_dispr_culrivar = cldList(P.adj ~ Comparison, data=Dunn_dispr_culrivar$res);CLD_Dunn_dispr_culrivar 

dispr_cultivar=ggboxplot(data_fram_cultivar, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Cultivars (n=100)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_cultivar
#write_csv(data_fram_cultivar, file="data_fram_cultivar.csv")
```

#Dispersion attributed to the regions of production
```{r, echo=FALSE,warning=FALSE}
dispr <- vegan::betadisper(hellinger_dist_matrix, phyloseq::sample_data(ps_hellinger)$Production_site_2);dispr #calculate the dispersion
data_fram_Region <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_Region
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_Region)
#data:  Richness by Region
#Kruskal-Wallis chi-squared = 235.44, df = 11, p-value < 2.2e-16

# Dunn's Test 
Dunn_dispr_Region=dunn_test(data_fram_Region, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_Region
write_csv(Dunn_dispr_Region, "Dunn_dispr_Region.csv")
Dunn_dispr_Region <- dunnTest(data_fram_Region$Distance_to_centroid, data_fram_Region$Group, method = "bonferroni");Dunn_dispr_Region
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_Region = cldList(P.adj ~ Comparison, data=Dunn_dispr_Region$res);CLD_Dunn_dispr_Region 


dispr_Region=ggboxplot(data_fram_Region, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Regions of Production")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_Region


```


```{r,echo=FALSE,warning=FALSE}
#TMV_resistance
ps_hellinger_TMV_resistance <- subset_samples(ps_hellinger, !is.na(TMV_resistance));ps_hellinger_TMV_resistance

#Distance
hellinger_dist_matrix_TMV <- phyloseq::distance(ps_hellinger_TMV_resistance, method = "bray");hellinger_dist_matrix_TMV
dispr <- vegan::betadisper(hellinger_dist_matrix_TMV, phyloseq::sample_data(ps_hellinger_TMV_resistance)$TMV_resistance);dispr #calculate the dispersion
data_fram_TMV_resistance <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_TMV_resistance
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_TMV_resistance)
#data:  Richness by TMV_resistance
#Kruskal-Wallis chi-squared = 20.23, df = 2, p-value = 4.047e-05

# Dunn's Test 
Dunn_dispr_TMV_resistance=dunn_test(data_fram_TMV_resistance, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_TMV_resistance
write_csv(Dunn_dispr_TMV_resistance, "Dunn_dispr_TMV_resistance.csv")
Dunn_dispr_TMV_resistance <- dunnTest(data_fram_TMV_resistance$Distance_to_centroid, data_fram_TMV_resistance$Group, method = "bonferroni");Dunn_dispr_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_dispr_TMV_resistance$res);CLD_Dunn_dispr_TMV_resistance 

myco_TMV=c("#0B775E","#FB9A99","grey35");myco_TMV
dispr_TMV_resistancen=ggboxplot(data_fram_TMV_resistance, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "TMV_resistance")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_TMV_resistancen
#################################################################################################

```

#consider insect resistance
```{r,echo=FALSE, warning=FALSE}
ps_hellinger_Insect_resistance <- subset_samples(ps_hellinger, !is.na(Insect_resistance));ps_hellinger_Insect_resistance

#Generate distance matrix
hellinger_dist_matrix_Insect_resistance <- phyloseq::distance(ps_hellinger_Insect_resistance, method = "bray");hellinger_dist_matrix_Insect_resistance
dispr <- vegan::betadisper(hellinger_dist_matrix_Insect_resistance, phyloseq::sample_data(ps_hellinger_Insect_resistance)$Insect_resistance);dispr #calculate the dispersion
data_fram_Insect_resistance <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_Insect_resistance
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_Insect_resistance)
#data:  Richness by Insect_resistance
#Kruskal-Wallis chi-squared = 63.43, df = 5, p-value = 2.371e-12

# Dunn's Test 
Dunn_dispr_Insect_resistance=dunn_test(data_fram_Insect_resistance, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_Insect_resistance
write_csv(Dunn_dispr_Insect_resistance, "Dunn_dispr_Insect_resistance.csv")
Dunn_dispr_Insect_resistance <- dunnTest(data_fram_Insect_resistance$Distance_to_centroid, data_fram_Insect_resistance$Group, method = "bonferroni");Dunn_dispr_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_dispr_Insect_resistance$res);CLD_Dunn_dispr_Insect_resistance 

myco_insRes=c("yellow3","grey35","chartreuse4","#0B775E","#00AFBB", "#FB9A99");myco_insRes
dispr_Insect_resistancen=ggboxplot(data_fram_Insect_resistance, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Insect resistance")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5))  +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_Insect_resistancen
```

##Fruit_shape 
```{r,echo=FALSE, warning=FALSE}
ps_hellinger_Fruit_shape <- subset_samples(ps_hellinger, !is.na(Fruit_shape));ps_hellinger_Fruit_shape

#Generate distance matrix
hellinger_dist_matrix_Fruit_shape <- phyloseq::distance(ps_hellinger_Fruit_shape, method = "bray");hellinger_dist_matrix_Fruit_shape
dispr <- vegan::betadisper(hellinger_dist_matrix_Fruit_shape, phyloseq::sample_data(ps_hellinger_Fruit_shape)$Fruit_shape);dispr #calculate the dispersion
data_fram_Fruit_shape <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_Fruit_shape
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_Fruit_shape)
#data:  Richness by Fruit_shape
#Kruskal-Wallis chi-squared = 35.887, df = 2, p-value = 1.611e-08

# Dunn's Test 
Dunn_dispr_Fruit_shape=dunn_test(data_fram_Fruit_shape, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_Fruit_shape
write_csv(Dunn_dispr_Fruit_shape, "Dunn_dispr_Fruit_shape.csv")
Dunn_dispr_Fruit_shape <- dunnTest(data_fram_Fruit_shape$Distance_to_centroid, data_fram_Fruit_shape$Group, method = "bonferroni");Dunn_dispr_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_dispr_Fruit_shape$res);CLD_Dunn_dispr_Fruit_shape 
#figure
myco_shp=c("darkturquoise","chartreuse4","#FB9A99");myco_shp
dispr_Fruit_shapen=ggboxplot(data_fram_Fruit_shape, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Berry shape")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5))  +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_Fruit_shapen
```

##Fruit_color
```{r,echo=FALSE, warning=FALSE}
ps_hellinger_Fruit_color <- subset_samples(ps_hellinger, !is.na(Fruit_color));ps_hellinger_Fruit_color

#Generate distance matrix
hellinger_dist_matrix_Fruit_color <- phyloseq::distance(ps_hellinger_Fruit_color, method = "bray");hellinger_dist_matrix_Fruit_color
dispr <- vegan::betadisper(hellinger_dist_matrix_Fruit_color, phyloseq::sample_data(ps_hellinger_Fruit_color)$Fruit_color);dispr #calculate the dispersion
data_fram_Fruit_color <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_Fruit_color
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_Fruit_color)
#data:  Richness by Fruit_color
#Kruskal-Wallis chi-squared = 43.556, df = 4, p-value = 7.932e-09

# Dunn's Test 
Dunn_dispr_Fruit_color=dunn_test(data_fram_Fruit_color, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_Fruit_color
write_csv(Dunn_dispr_Fruit_color, "Dunn_dispr_Fruit_color.csv")
Dunn_dispr_Fruit_color <- dunnTest(data_fram_Fruit_color$Distance_to_centroid, data_fram_Fruit_color$Group, method = "bonferroni");Dunn_dispr_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_dispr_Fruit_color$res);CLD_Dunn_dispr_Fruit_color 
#figure
myco_col=c("grey35","#00AFBB", "yellow3","chartreuse4","#FB9A99");myco_col
dispr_Fruit_colorn=ggboxplot(data_fram_Fruit_color, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Berry color")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_Fruit_colorn
```

#dispersion between the yield group
```{r,echo=FALSE, warning=FALSE}
ps_hellinger_yield <- subset_samples(ps_hellinger, !is.na(yield));ps_hellinger_yield

#Generate distance matrix
hellinger_dist_matrix_yield <- phyloseq::distance(ps_hellinger_yield, method = "bray");hellinger_dist_matrix_yield
dispr <- vegan::betadisper(hellinger_dist_matrix_yield, phyloseq::sample_data(ps_hellinger_yield)$yield);dispr #calculate the dispersion
data_fram_yield <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_yield
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_yield)
#data:  Richness by yield
#Kruskal-Wallis chi-squared = 48.201, df = 4, p-value = 8.569e-10

# Dunn's Test 
Dunn_dispr_yield=dunn_test(data_fram_yield, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_yield
write_csv(Dunn_dispr_yield, "Dunn_dispr_yield.csv")
Dunn_dispr_yield <- dunnTest(data_fram_yield$Distance_to_centroid, data_fram_yield$Group, method = "bonferroni");Dunn_dispr_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_yield = cldList(P.adj ~ Comparison, data=Dunn_dispr_yield$res);CLD_Dunn_dispr_yield 

#Fig
myco_yield=c("chartreuse4","yellow3","#0B775E","#00AFBB","grey35");myco_yield
dispr_yieldn=ggboxplot(data_fram_yield, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Yield in Kilograms")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_yieldn
```

#ovaries
```{r,echo=FALSE, warning=FALSE}
ps_hellinger_ovaries <- subset_samples(ps_hellinger, !is.na(ovaries));ps_hellinger_ovaries

#Generate distance matrix
hellinger_dist_matrix_ovaries <- phyloseq::distance(ps_hellinger_ovaries, method = "bray");hellinger_dist_matrix_ovaries
dispr <- vegan::betadisper(hellinger_dist_matrix_ovaries, phyloseq::sample_data(ps_hellinger_ovaries)$ovaries);dispr #calculate the dispersion
data_fram_ovaries <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_ovaries
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_ovaries)
#data:  Richness by ovaries
#Kruskal-Wallis chi-squared = 5.9035, df = 2, p-value = 0.05225

# Dunn's Test 
Dunn_dispr_ovaries=dunn_test(data_fram_ovaries, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_ovaries
write_csv(Dunn_dispr_ovaries, "Dunn_dispr_ovaries.csv")
Dunn_dispr_ovaries <- dunnTest(data_fram_ovaries$Distance_to_centroid, data_fram_ovaries$Group, method = "bonferroni");Dunn_dispr_ovaries
#ns
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_ovaries = cldList(P.adj ~ Comparison, data=Dunn_dispr_ovaries$res);CLD_Dunn_dispr_ovaries 

myco_ovaries=c("#00AFBB","#FB9A99","chartreuse4");myco_ovaries
dispr_ovariesn=ggboxplot(data_fram_ovaries, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Number of Ovaries")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_ovariesn
```

#seedweight_1000
```{r,echo=FALSE, warning=FALSE}
ps_hellinger_seedweight_1000 <- subset_samples(ps_hellinger, !is.na(seedweight_1000));ps_hellinger_seedweight_1000

#Generate distance matrix
hellinger_dist_matrix_seedweight_1000 <- phyloseq::distance(ps_hellinger_seedweight_1000, method = "bray");hellinger_dist_matrix_seedweight_1000
dispr <- vegan::betadisper(hellinger_dist_matrix_seedweight_1000, phyloseq::sample_data(ps_hellinger_seedweight_1000)$seedweight_1000);dispr #calculate the dispersion
data_fram_seedweight_1000 <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_seedweight_1000
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_seedweight_1000)
#data:  Richness by seedweight_1000
#Kruskal-Wallis chi-squared = 36.276, df = 3, p-value = 6.548e-08

# Dunn's Test 
Dunn_dispr_seedweight_1000=dunn_test(data_fram_seedweight_1000, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_seedweight_1000
write_csv(Dunn_dispr_seedweight_1000, "Dunn_dispr_seedweight_1000.csv")
Dunn_dispr_seedweight_1000 <- dunnTest(data_fram_seedweight_1000$Distance_to_centroid, data_fram_seedweight_1000$Group, method = "bonferroni");Dunn_dispr_seedweight_1000

# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_dispr_seedweight_1000$res);CLD_Dunn_dispr_seedweight_1000 

myco_weight=c("#0B775E","#FB9A99","yellow3","grey35");myco_weight
dispr_seedweight_1000n=ggboxplot(data_fram_seedweight_1000, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Weight in grams/1000 seeds")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_seedweight_1000n
```

#Fruit_hori_diameter
```{r,echo=FALSE, warning=FALSE}
ps_hellinger_Fruit_hori_diameter <- subset_samples(ps_hellinger, !is.na(Fruit_hori_diameter));ps_hellinger_Fruit_hori_diameter

#Generate distance matrix
hellinger_dist_matrix_Fruit_hori_diameter <- phyloseq::distance(ps_hellinger_Fruit_hori_diameter, method = "bray");hellinger_dist_matrix_Fruit_hori_diameter

dispr <- vegan::betadisper(hellinger_dist_matrix_Fruit_hori_diameter, phyloseq::sample_data(ps_hellinger_Fruit_hori_diameter)$Fruit_hori_diameter);dispr #calculate the dispersion
data_fram_Fruit_hori_diameter <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_Fruit_hori_diameter
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_Fruit_hori_diameter)
#data:  Richness by Fruit_hori_diameter
#ruskal-Wallis chi-squared = 1.4474, df = 2, p-value = 0.485

# Dunn's Test 
Dunn_dispr_Fruit_hori_diameter=dunn_test(data_fram_Fruit_hori_diameter, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_Fruit_hori_diameter
write_csv(Dunn_dispr_Fruit_hori_diameter, "Dunn_dispr_Fruit_hori_diameter.csv")
Dunn_dispr_Fruit_hori_diameter <- dunnTest(data_fram_Fruit_hori_diameter$Distance_to_centroid, data_fram_Fruit_hori_diameter$Group, method = "bonferroni");Dunn_dispr_Fruit_hori_diameter
#ns
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_dispr_Fruit_hori_diameter$res);CLD_Dunn_dispr_Fruit_hori_diameter 

myco_horiz=c("chartreuse4","#00AFBB","#FB9A99");myco_horiz
dispr_Fruit_hori_diametern=ggboxplot(data_fram_Fruit_hori_diameter, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Fruit horizontal diameter/cm")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_Fruit_hori_diametern
```

##Fruit_long_diameter
```{r}
ps_hellinger_Fruit_long_diameter <- subset_samples(ps_hellinger, !is.na(Fruit_long_diameter));ps_hellinger_Fruit_long_diameter

#Generate distance matrix
hellinger_dist_matrix_Fruit_long_diameter <- phyloseq::distance(ps_hellinger_Fruit_long_diameter, method = "bray");hellinger_dist_matrix_Fruit_long_diameter

dispr <- vegan::betadisper(hellinger_dist_matrix_Fruit_long_diameter, phyloseq::sample_data(ps_hellinger_Fruit_long_diameter)$Fruit_long_diameter);dispr #calculate the dispersion
data_fram_Fruit_long_diameter <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_Fruit_long_diameter
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_Fruit_long_diameter)
#data:  Richness by Fruit_long_diameter
#Kruskal-Wallis chi-squared = 24.86, df = 2, p-value = 3.997e-06

# Dunn's Test 
Dunn_dispr_Fruit_long_diameter=dunn_test(data_fram_Fruit_long_diameter, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_Fruit_long_diameter
write_csv(Dunn_dispr_Fruit_long_diameter, "Dunn_dispr_Fruit_long_diameter.csv")
Dunn_dispr_Fruit_long_diameter <- dunnTest(data_fram_Fruit_long_diameter$Distance_to_centroid, data_fram_Fruit_long_diameter$Group, method = "bonferroni");Dunn_dispr_Fruit_long_diameter
#ns
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_dispr_Fruit_long_diameter$res);CLD_Dunn_dispr_Fruit_long_diameter 

myco_longdiam=c("chartreuse4","#00AFBB","#FB9A99" );myco_longdiam
dispr_Fruit_long_diametern=ggboxplot(data_fram_Fruit_long_diameter, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Fruit longitudinal diameter/cm")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_Fruit_long_diametern
```


#Fruit_taste
```{r}
ps_hellinger_Fruit_taste <- subset_samples(ps_hellinger, !is.na(Fruit_taste));ps_hellinger_Fruit_taste

#Generate distance matrix
hellinger_dist_matrix_Fruit_taste <- phyloseq::distance(ps_hellinger_Fruit_taste, method = "bray");hellinger_dist_matrix_Fruit_taste

#ADONIS test
vegan::adonis2(hellinger_dist_matrix_Fruit_taste ~ phyloseq::sample_data(ps_hellinger_Fruit_taste)$Fruit_taste, permutation=999)
#                                                             Df SumOfSqs      R2      F Pr(>F)    
#phyloseq::sample_data(ps_hellinger_Fruit_taste)$Fruit_taste    1     1.22 0.00427 4.7137  0.001 ***
#Residual                                                    1100   284.81 0.99573                  
#Total                                                       1101   286.03 1.00000  
dispr <- vegan::betadisper(hellinger_dist_matrix_Fruit_taste, phyloseq::sample_data(ps_hellinger_Fruit_taste)$Fruit_taste);dispr #calculate the dispersion
data_fram_Fruit_taste <- data.frame(Distance_to_centroid=dispr$distances,Group=dispr$group);data_fram_Fruit_taste
groups <- dispr$group

#stats
kruskal.test(Distance_to_centroid ~ Group, data = data_fram_Fruit_taste)
#data:  Richness by Fruit_taste
#Kruskal-Wallis chi-squared = 2.142, df = 1, p-value = 0.1433

# Dunn's Test 
Dunn_dispr_Fruit_taste=dunn_test(data_fram_Fruit_taste, Distance_to_centroid ~ Group, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_dispr_Fruit_taste
write_csv(Dunn_dispr_Fruit_taste, "Dunn_dispr_Fruit_taste.csv")
Dunn_dispr_Fruit_taste <- dunnTest(data_fram_Fruit_taste$Distance_to_centroid, data_fram_Fruit_taste$Group, method = "bonferroni");Dunn_dispr_Fruit_taste
#ns
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_dispr_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_dispr_Fruit_taste$res);CLD_Dunn_dispr_Fruit_taste 

myco_taste=c("#FB9A99","chartreuse4");myco_taste
dispr_Fruit_tasten=ggboxplot(data_fram_Fruit_taste, x = "Group", y = "Distance_to_centroid", fill= "Group", ylab = "Distance to centroid", xlab = "Berry taste")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_text(face="bold", colour="gray33", size=12),
        axis.text.y  = element_text( colour="gray33", size=12))+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));dispr_Fruit_tasten
##########################################

```

#Fig.S16 &17
```{r,echo=FALSE,warning=FALSE}
#Merging
library(patchwork)
tiff("Dist_from_centroid_2.tiff", units="in", width=12, height=10, res=300)
Dist_from_centroid_2.tiff=((dispr_yieldn+dispr_Fruit_long_diametern+dispr_Fruit_hori_diametern)/(dispr_Fruit_colorn+dispr_TMV_resistancen+dispr_ovariesn)/(dispr_Fruit_shapen+dispr_Fruit_tasten));Dist_from_centroid_2.tiff
ggsave("Dist_from_centroid_2.tiff", plot = Dist_from_centroid_2.tiff)
dev.off()

tiff("Dist_from_centroid.tiff", units="in", width=13, height=13, res=300)
Dist_from_centroid.tiff=(dispr_cultivar/(dispr_Region)/(dispr_Insect_resistancen+dispr_seedweight_1000n));Dist_from_centroid.tiff
ggsave("Dist_from_centroid.tiff", plot = Dist_from_centroid.tiff)
dev.off()
```

#Beta-diversity  & associated stats
#USING THE MicrobiotaProcess package
#https://r-statistics.co/Top50-Ggplot2-Visualizations-MasterList-R-Code.htm
#https://yulab-smu.top/MicrobiotaProcessWorkshop/articles/MicrobiotaProcessWorkshop.html

##Beta diversity (cultivars)
```{r,echo=FALSE, warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)

distme <- get_dist(ps2, distmethod ="bray", method="hellinger");distme
sampleda <- data.frame(sample_data(ps2), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(distme)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$English_Name_2 <- factor(sampleda$English_Name_2)
adores <- adonis2(distme ~ English_Name_2, 
                  data=sampleda, 
                  permutation=999, 
                  strata =sampleda$Production_site_2 );adores

#           Df SumOfSqs      R2     F Pr(>F)    
#Model      99   184.77 0.59353 16.18  0.001 ***
#Residual 1097   126.54 0.40647                 
#Total    1196   311.31 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
adores <- adonis2(distme ~ English_Name_2+Production_site_2,
                  data=sampleda, 
                  permutation=999);adores

#           Df SumOfSqs      R2     F Pr(>F)    
#Model      99   184.77 0.59353 16.18  0.001 ***
#Residual 1097   126.54 0.40647                 
#Total    1196   311.31 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adores=adonis2(distme ~ Production_site_2 / English_Name_2, 
               data = sampleda, 
               permutations = 999, 
               method = "bray");adores
#           Df SumOfSqs      R2     F Pr(>F)    
#Model      99   184.77 0.59353 16.18  0.001 ***
#Residual 1097   126.54 0.40647                 
#Total    1196   311.31 1.00000                 
#---
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
###################################################################################################
# "unifrac", "wunifrac", "manhattan", "euclidean", "canberra", "bray", "kulczynski" ...(vegdist, dist)
pcares <- get_pca(obj=ps2, method="hellinger");pcares# for PCA ordination to show sample distribution
pcoares <- get_pcoa(obj=ps2, distmethod="bray", method="hellinger");pcoares#computes the pcoa distance matrices

##Plot PCoA considering the cultivars
pcoaplot5 <- ggordpoint(obj=pcares, pc=c(1, 2), biplot=F, speciesannot=F,
                       factorNames=c("English_Name_2"), ellipse=F)+
  scale_fill_manual(values=mycola)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "G");pcoaplot5
#ggsave("pcaplot1.TIFF", plot = pcaplot1)
# pc = c(1, 3) to show the first & third principal components.

pcoaplot3 <- ggordpoint(obj=pcoares, biplot=F, speciesannot=F,
                        factorNames=c("English_Name_2"), ellipse=F) +
  scale_fill_manual(values=mycola)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=4))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "H");pcoaplot3
#ggsave("pcoaplot3.TIFF", plot = pcoaplot3);pcoaplot3

#Combine the figures
tiff("PCOA_cultitvars.tiff", units="in", width=8, height=4, res=300)
PCOA_cultitvars.tiff=pcoaplot5+pcoaplot3+plot_layout(ncol = 2);PCOA_cultitvars.tiff
ggsave("PCOA_cultitvars.tiff", plot = PCOA_cultitvars.tiff)
dev.off()
###################################################################################################
```

#Beta diversity Jiangsu
```{r,echo=FALSE,warning=FALSE}

library(MicrobiotaProcess)
ps2_2= subset_samples(ps2_2, Production_site_2 %in% c("Jiangsu"));ps2_2#[ 15421 taxa & 1102 samples ]
distme_ps2_2 <- get_dist(ps2_2, distmethod ="bray", method="hellinger");distme_ps2_2
sampleda_ps2_2 <- data.frame(sample_data(ps2_2), check.names=FALSE);sampleda_ps2_2
sampleda_ps2_2 <- sampleda_ps2_2[match(colnames(as.matrix(distme_ps2_2)),rownames(sampleda_ps2_2)),,drop=FALSE];sampleda_ps2_2
sampleda_ps2_2$English_Name_2 <- factor(sampleda_ps2_2$English_Name_2)
adores <- adonis2(distme_ps2_2 ~ English_Name_2, data=sampleda_ps2_2, permutation=999);adores#change from adonis to adonis2
#         Df SumOfSqs      R2      F Pr(>F)    
#Model     50   72.347 0.51929 12.078  0.001 ***
#Residual 559   66.971 0.48071                  
#Total    609  139.318 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##Fig.S14A
pcoares_ps2_2 <- get_pcoa(obj=ps2_2, distmethod="bray", method="hellinger");pcoares_ps2_2#pcoa
pcoaplot_ps2_2 <- ggordpoint(obj=pcoares_ps2_2, biplot=F, speciesannot=F,
                              factorNames=c("English_Name_2"), ellipse=F) +
  scale_fill_manual(values=mycola)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "right")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_ps2_2

tiff("pcoaplot_ps2_2.tiff", units="in", width=8, height=5, res=300)
pcoaplot_ps2_2.tiff=(pcoaplot_ps2_2);pcoaplot_ps2_2.tiff
ggsave("pcoaplot_ps2_2.tiff", plot = pcoaplot_ps2_2.tiff)
dev.off()
#compositional plots across the selected regions Jiangu

# Bar plots:
##Showing individual replicates. Top 500 taxa
toptax = names(sort(taxa_sums(ps2_2), decreasing=TRUE))[1:500];toptax
ps.toptax = transform_sample_counts(ps2_2, function(x) {x/sum(x)*100});ps.toptax
ps.toptax <- prune_taxa(toptax, ps.toptax);ps.toptax

#Class level
psmelt(ps.toptax)
gg1_jiangsu=ggplot(psmelt(ps.toptax), aes(x=reorder(Sample, English_Name), y=Abundance, fill=Class))+
  geom_bar(stat = "identity", position = "stack", width = 1) +
  #scale_fill_manual(values=c(ocean.dense(7), ocean.tempo(8)))+
  #scale_fill_manual(values=c("#25725d", "#439b5b", "#7bc369", "#abd69a", "#d8e8b0", "#cccda7", "#bfb77b", "#7e9998", "#92B2B2", "#83bbbf", "#61acc1", "#58a0c4", "#7087c3","#8973b4", "#512972"))+
  scale_fill_manual(values = mycola)+
  xlab("") + ylab("Relative abundance (%)") +
  facet_wrap(~reorder(Production_site_2, Sample), scales = "free_x", nrow = 1) + 
  ggtitle("Class, top 500 ASVs") +
  theme_pubclean() + theme(legend.position="bottom", axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=0)) +
theme(legend.text = element_text(face = c(rep("italic", 5), rep("plain", 5))))+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow = 3))+
  theme(text=element_text(size=8,  family="Arial"))+ labs(tag = "");gg1_jiangsu
```


#Fig.S14B
```{r}
library(ggpubr)
tiff("gg1_jiangsu.tiff", units="in", width=9, height=5, res=300)
gg1_jiangsu.tiff=(gg1_jiangsu);gg1_jiangsu.tiff
ggsave("gg1_jiangsu.tiff", plot = gg1_jiangsu.tiff)
dev.off()
```


#consider the region of production a a factor
```{r,echo=FALSE, warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)
pcares <- get_pca(obj=ps2, method="hellinger");pcares# for PCA ordination to show sample distribution
pcoares <- get_pcoa(obj=ps2, distmethod="bray", method="hellinger");pcoares#computes the pcoa distance matrices
distme <- get_dist(ps2, distmethod ="bray", method="hellinger");distme
sampleda <- data.frame(sample_data(ps2), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(distme)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Production_site_2 <- factor(sampleda$Production_site_2)
adores <- adonis2(distme ~ Production_site_2, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                     Df SumOfSqs      R2      F Pr(>F)    
#Production_site_2   11   35.537 0.11415 13.882  0.001 ***
#Residual          1185  275.776 0.88585                  
#Total             1196  311.313 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1                

#VisuAlizing the result (#PCA_pcoa_countries of origin)
pcaplot1 <- ggordpoint(obj=pcares, pc=c(1, 2), biplot=F, speciesannot=F,
                       factorNames=c("Production_site_2"), ellipse=F)+
  scale_fill_manual(values=mycola)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "E");pcaplot1
#ggsave("pcaplot1.TIFF", plot = pcaplot1)
# pc = c(1, 3) to show the first & third principal components.

pcoaplot1 <- ggordpoint(obj=pcoares, biplot=F, speciesannot=F,
                        factorNames=c("Production_site_2"), ellipse=F) +
  scale_fill_manual(values=mycola)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=14,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "F");pcoaplot1

#Fig.S13E&F
tiff("PCA_PCOA_Region.tiff", units="in", width=8, height=4, res=300)
PCA_PCOA_Region.tiff=pcaplot1+pcoaplot1+plot_layout(ncol = 2);PCA_PCOA_Region.tiff
ggsave("PCA_PCOA_Region.tiff", plot = PCA_PCOA_Region.tiff)
dev.off()
```


#Insect_resistance
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)
Bacteria_Insect_resistance <- subset_samples(ps2, !is.na(Insect_resistance));Bacteria_Insect_resistance
distme_InsectRes <- get_dist(Bacteria_Insect_resistance, distmethod ="bray", method="hellinger");distme_InsectRes
sampleda_InsectRes <- data.frame(sample_data(Bacteria_Insect_resistance), check.names=FALSE);sampleda_InsectRes
sampleda_InsectRes <- sampleda_InsectRes[match(colnames(as.matrix(distme_InsectRes)),rownames(sampleda_InsectRes)),,drop=FALSE];sampleda_InsectRes
sampleda_InsectRes$Insect_resistance <- factor(sampleda_InsectRes$Insect_resistance)
adores <- adonis2(distme_InsectRes ~ Insect_resistance, data=sampleda_InsectRes, permutation=999);adores#change from adonis to adonis2
#                  Df SumOfSqs      R2      F Pr(>F)    
#Insect_resistance   5   10.811 0.07079 9.0201  0.001 ***
#Residual          592  141.911 0.92921                  
#Total             597  152.722 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Figure
myco_insRes=c("yellow3","grey35","chartreuse4","#0B775E","#00AFBB", "#FB9A99");myco_insRes
pcoares_InsectRes <- get_pcoa(obj=Bacteria_Insect_resistance, distmethod="bray", method="hellinger");pcoares_InsectRes#pcoa
pcoaplot_InsectRes <- ggordpoint(obj=pcoares_InsectRes, biplot=F, speciesannot=F,
                              factorNames=c("Insect_resistance"), ellipse=F) +
  scale_fill_manual(values=myco_insRes)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_InsectRes
```
#seedweight in grams/1000 seeds
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)
Bacteria_seedweight_1000 <- subset_samples(ps2, !is.na(seedweight_1000));Bacteria_seedweight_1000#
distme_seedweight_1000 <- get_dist(Bacteria_seedweight_1000, distmethod ="bray", method="hellinger");distme_seedweight_1000
sampleda_seedweight_1000 <- data.frame(sample_data(Bacteria_seedweight_1000), check.names=FALSE);sampleda_seedweight_1000
sampleda_seedweight_1000 <- sampleda_seedweight_1000[match(colnames(as.matrix(distme_seedweight_1000)),rownames(sampleda_seedweight_1000)),,drop=FALSE];sampleda_seedweight_1000
sampleda_seedweight_1000$seedweight_1000 <- factor(sampleda_seedweight_1000$seedweight_1000)
adores <- adonis2(distme_seedweight_1000 ~ seedweight_1000, data=sampleda_seedweight_1000, permutation=999);adores#change from adonis to adonis2
#                 Df SumOfSqs      R2      F Pr(>F)    
#seedweight_1000   3    6.822 0.03147 9.1536  0.001 ***
#Residual        845  209.922 0.96853                  
#Total           848  216.744 1.00000

#Figure
myco_weight=c("#0B775E","#FB9A99","yellow3","grey35");myco_weight
pcoares_seedweight_1000 <- get_pcoa(obj=Bacteria_seedweight_1000, distmethod="bray", method="hellinger");pcoares_seedweight_1000#pcoa
pcoaplot_seedweight_1000 <- ggordpoint(obj=pcoares_seedweight_1000, biplot=F, speciesannot=F,
                                       factorNames=c("seedweight_1000"), ellipse=F) +
  scale_fill_manual(values=myco_weight)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_seedweight_1000#
```

#yield
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)
Bacteria_yield <- subset_samples(ps2, !is.na(yield));Bacteria_yield
distme_yield <- get_dist(Bacteria_yield, distmethod ="bray", method="hellinger");distme_yield
sampleda_yield <- data.frame(sample_data(Bacteria_yield), check.names=FALSE);sampleda_yield
sampleda_yield <- sampleda_yield[match(colnames(as.matrix(distme_yield)),rownames(sampleda_yield)),,drop=FALSE];sampleda_yield
sampleda_yield$yield <- factor(sampleda_yield$yield)
adores <- adonis2(distme_yield ~ yield, data=sampleda_yield, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)    
#yield       4    8.256 0.03093 8.2746  0.001 ***
#Residual 1037  258.680 0.96907                  
#Total    1041  266.936 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
##############################################################################################################################################################################################################
myco_yield=c("chartreuse4","yellow3","#0B775E","#00AFBB","grey35");myco_yield
pcoares_yield <- get_pcoa(obj=Bacteria_yield, distmethod="bray", method="hellinger");pcoares_yield#pcoa
pcoaplot_yield <- ggordpoint(obj=pcoares_yield, biplot=F, speciesannot=F,
                             factorNames=c("yield"), ellipse=F) +
  scale_fill_manual(values=myco_yield)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_yield

```

#Berry Longitudinal Diameter
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)
Bacteria_Fruit_long_diameter <- subset_samples(ps2, !is.na(Fruit_long_diameter));Bacteria_Fruit_long_diameter
distme_Fruit_long_diameter <- get_dist(Bacteria_Fruit_long_diameter, distmethod ="bray", method="hellinger");distme_Fruit_long_diameter
sampleda_Fruit_long_diameter <- data.frame(sample_data(Bacteria_Fruit_long_diameter), check.names=FALSE);sampleda_Fruit_long_diameter
sampleda_Fruit_long_diameter <- sampleda_Fruit_long_diameter[match(colnames(as.matrix(distme_Fruit_long_diameter)),rownames(sampleda_Fruit_long_diameter)),,drop=FALSE];sampleda_Fruit_long_diameter
sampleda_Fruit_long_diameter$Fruit_long_diameter <- factor(sampleda_Fruit_long_diameter$Fruit_long_diameter)
adores <- adonis2(distme_Fruit_long_diameter ~ Fruit_long_diameter, data=sampleda_Fruit_long_diameter, permutation=999);adores#change from adonis to adonis2
#                      Df SumOfSqs      R2      F Pr(>F)    
#Fruit_long_diameter    2    5.583 0.02014 11.086  0.001 ***
#Residual            1079  271.687 0.97986                  
#Total               1081  277.270 1.00000                   
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
##############################################################################################################################################################################################################
myco_longdiam=c("chartreuse4","#00AFBB","#FB9A99" );myco_longdiam
pcoares_Fruit_long_diameter <- get_pcoa(obj=Bacteria_Fruit_long_diameter, distmethod="bray", method="hellinger");pcoares_Fruit_long_diameter#pcoa
pcoaplot_Fruit_long_diameter <- ggordpoint(obj=pcoares_Fruit_long_diameter, biplot=F, speciesannot=F,
                                           factorNames=c("Fruit_long_diameter"), ellipse=F) +
  scale_fill_manual(values=myco_longdiam)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_Fruit_long_diameter

```
#Fruit_hori_diameter
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
Bacteria_Fruit_hori_diameter <- subset_samples(ps2, !is.na(Fruit_hori_diameter));Bacteria_Fruit_hori_diameter
distme_Fruit_hori_diameter <- get_dist(Bacteria_Fruit_hori_diameter, distmethod ="bray", method="hellinger");distme_Fruit_hori_diameter
sampleda_Fruit_hori_diameter <- data.frame(sample_data(Bacteria_Fruit_hori_diameter), check.names=FALSE);sampleda_Fruit_hori_diameter
sampleda_Fruit_hori_diameter <- sampleda_Fruit_hori_diameter[match(colnames(as.matrix(distme_Fruit_hori_diameter)),rownames(sampleda_Fruit_hori_diameter)),,drop=FALSE];sampleda_Fruit_hori_diameter
sampleda_Fruit_hori_diameter$Fruit_hori_diameter <- factor(sampleda_Fruit_hori_diameter$Fruit_hori_diameter)
adores <- adonis2(distme_Fruit_hori_diameter ~ Fruit_hori_diameter, data=sampleda_Fruit_hori_diameter, permutation=999);adores#change from adonis to adonis2
#                       Df SumOfSqs     R2      F Pr(>F)    
#Fruit_hori_diameter    2    5.101 0.0174 9.6611  0.001 ***
#Residual            1091  288.020 0.9826                  
#Total               1093  293.121 1.0000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
##############################################################################################################################################################################################################
myco_horiz=c("chartreuse4","#00AFBB","#FB9A99");myco_horiz
pcoares_Fruit_hori_diameter <- get_pcoa(obj=Bacteria_Fruit_hori_diameter, distmethod="bray", method="hellinger");pcoares_Fruit_hori_diameter#pcoa
pcoaplot_Fruit_hori_diameter <- ggordpoint(obj=pcoares_Fruit_hori_diameter, biplot=F, speciesannot=F,
                                           factorNames=c("Fruit_hori_diameter"), ellipse=F) +
  scale_fill_manual(values=myco_horiz)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_Fruit_hori_diameter

```
#Fruit color
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
#Fruit color
tiff("PCOA_Parameters.tiff", units="in", width=20, height=20, res=300)
Bacteria_Fruit_color <- subset_samples(ps2, !is.na(Fruit_color));Bacteria_Fruit_color
distme_Fcol <- get_dist(Bacteria_Fruit_color, distme_Fcolthod ="bray", method="hellinger");distme_Fcol
sampleda_Fcol <- data.frame(sample_data(Bacteria_Fruit_color), check.names=FALSE);sampleda_Fcol
sampleda_Fcol <- sampleda_Fcol[match(colnames(as.matrix(distme_Fcol)),rownames(sampleda_Fcol)),,drop=FALSE];sampleda_Fcol
sampleda_Fcol$Fruit_color <- factor(sampleda_Fcol$Fruit_color)
adores <- adonis2(distme_Fcol ~ Fruit_color, data=sampleda_Fcol, permutation=999);adores#change from adonis to adonis2
#             Df SumOfSqs      R2      F Pr(>F)    
#Fruit_color    4    15.05 0.02084 6.2792  0.001 ***
#Residual    1180   707.05 0.97916                  
#Total       1184   722.10 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
##############################################################################################################################################################################################################
myco_col=c("grey35","#00AFBB", "yellow3","chartreuse4","#FB9A99");myco_col
pcoares_Fcol <- get_pcoa(obj=Bacteria_Fruit_color, distmethod="bray", method="hellinger");pcoares_Fcol#pcoa
pcoaplot_Fcol <- ggordpoint(obj=pcoares_Fcol, biplot=F, speciesannot=F,
                        factorNames=c("Fruit_color"), ellipse=F) +
  scale_fill_manual(values=mycola)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_Fcol

```

##TMV resisitance (beta diversity stats & figure)
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)
Bacteria_TMV_resistance <- subset_samples(ps2, !is.na(TMV_resistance));Bacteria_TMV_resistance
distme_TmvRes <- get_dist(Bacteria_TMV_resistance, distmethod ="bray", method="hellinger");distme_TmvRes
sampleda_TmvRes <- data.frame(sample_data(Bacteria_TMV_resistance), check.names=FALSE);sampleda_TmvRes
sampleda_TmvRes <- sampleda_TmvRes[match(colnames(as.matrix(distme_TmvRes)),rownames(sampleda_TmvRes)),,drop=FALSE];sampleda_TmvRes
sampleda_TmvRes$TMV_resistance <- factor(sampleda_TmvRes$TMV_resistance)
adores <- adonis2(distme_TmvRes ~ TMV_resistance, data=sampleda_TmvRes, permutation=999);adores#change from adonis to adonis2
#                 Df SumOfSqs      R2      F Pr(>F)    
#TMV_resistance   2    2.327 0.01441 4.5263  0.001 ***
#Residual       619  159.134 0.98559                  
#Total          621  161.461 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
##############################################################################################################################################################################################################
myco_TMV=c("#0B775E","#FB9A99","grey35");myco_TMV
pcoares_TmvRes <- get_pcoa(obj=Bacteria_TMV_resistance, distmethod="bray", method="hellinger");pcoares_TmvRes#pcoa
pcoaplot_TmvRes <- ggordpoint(obj=pcoares_TmvRes, biplot=F, speciesannot=F,
                              factorNames=c("TMV_resistance"), ellipse=F) +
  scale_fill_manual(values=myco_TMV)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_TmvRes

```
#NO_ovaries
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)

Bacteria_ovaries <- subset_samples(ps2, !is.na(ovaries));Bacteria_ovaries
distme_ovaries <- get_dist(Bacteria_ovaries, distmethod ="bray", method="hellinger");distme_ovaries
sampleda_ovaries <- data.frame(sample_data(Bacteria_ovaries), check.names=FALSE);sampleda_ovaries
sampleda_ovaries <- sampleda_ovaries[match(colnames(as.matrix(distme_ovaries)),rownames(sampleda_ovaries)),,drop=FALSE];sampleda_ovaries
sampleda_ovaries$ovaries <- factor(sampleda_ovaries$ovaries)
adores <- adonis2(distme_ovaries ~ ovaries, data=sampleda_ovaries, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)    
#ovaries     2    5.045 0.01694 9.5556  0.001 ***
#Residual 1109  292.753 0.98306                  
#Total    1111  297.798 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
##############################################################################################################################################################################################################
myco_ovaries=c("#00AFBB","#FB9A99","chartreuse4");myco_ovaries
pcoares_ovaries <- get_pcoa(obj=Bacteria_ovaries, distmethod="bray", method="hellinger");pcoares_ovaries#pcoa
pcoaplot_ovaries <- ggordpoint(obj=pcoares_ovaries, biplot=F, speciesannot=F,
                               factorNames=c("ovaries"), ellipse=F) +
  scale_fill_manual(values=myco_ovaries)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "")+  
  geom_mark_hull(alpha = 0.2) ;pcoaplot_ovaries

```
#Fruitshape
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)

Bacteria_Fruit_shape <- subset_samples(ps2, !is.na(Fruit_shape));Bacteria_Fruit_shape
distme_Fshape <- get_dist(Bacteria_Fruit_shape, distmethod ="bray", method="hellinger");distme_Fshape
sampleda_Fshape <- data.frame(sample_data(Bacteria_Fruit_shape), check.names=FALSE);sampleda_Fshape
sampleda_Fshape <- sampleda_Fshape[match(colnames(as.matrix(distme_Fshape)),rownames(sampleda_Fshape)),,drop=FALSE];sampleda_Fshape
sampleda_Fshape$Fruit_shape <- factor(sampleda_Fshape$Fruit_shape)
adores <- adonis2(distme_Fshape ~ Fruit_shape, data=sampleda_Fshape, permutation=999);adores#change from adonis to adonis2
#               Df SumOfSqs     R2      F Pr(>F)    
#Fruit_shape    2     4.37 0.0136 8.1494  0.001 ***
#Residual    1182   317.21 0.9864                  
#Total       1184   321.58 1.0000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


myco_shp=c("darkturquoise","chartreuse4","#FB9A99");myco_shp
pcoares_Fshape <- get_pcoa(obj=Bacteria_Fruit_shape, distmethod="bray", method="hellinger");pcoares_Fshape#pcoa
pcoaplot_Fshape <- ggordpoint(obj=pcoares_Fshape, biplot=F, speciesannot=F,
                              factorNames=c("Fruit_shape"), ellipse=F) +
  scale_fill_manual(values=myco_shp)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_Fshape
#
```

#Fruit_taste
```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")
library(MicrobiotaProcess)

Bacteria_Fruit_taste <- subset_samples(ps2, Fruit_taste %in% c("Sweet", "Sour"));Bacteria_Fruit_taste#[ 15588 taxa & 1102 samples ]
distme_Fruit_taste <- get_dist(Bacteria_Fruit_taste, distmethod ="bray", method="hellinger");distme_Fruit_taste
sampleda_Fruit_taste <- data.frame(sample_data(Bacteria_Fruit_taste), check.names=FALSE);sampleda_Fruit_taste
sampleda_Fruit_taste <- sampleda_Fruit_taste[match(colnames(as.matrix(distme_Fruit_taste)),rownames(sampleda_Fruit_taste)),,drop=FALSE];sampleda_Fruit_taste
sampleda_Fruit_taste$Fruit_taste <- factor(sampleda_Fruit_taste$Fruit_taste)
adores <- adonis2(distme_Fruit_taste ~ Fruit_taste, data=sampleda_Fruit_taste, permutation=999);adores#change from adonis to adonis2
#              Df SumOfSqs      R2      F Pr(>F)    
#Fruit_taste    1     1.22 0.00427 4.7137  0.001 ***
#Residual    1100   284.81 0.99573                  
#Total       1101   286.03 1.00000                   
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
##############################################################################################################################################################################################################
myco_taste=c("#FB9A99","chartreuse4");myco_taste
pcoares_Fruit_taste <- get_pcoa(obj=Bacteria_Fruit_taste, distmethod="bray", method="hellinger");pcoares_Fruit_taste#pcoa
pcoaplot_Fruit_taste <- ggordpoint(obj=pcoares_Fruit_taste, biplot=F, speciesannot=F,
                                   factorNames=c("Fruit_taste"), ellipse=F) +
  scale_color_viridis_b()+
  #scale_fill_manual(values=myco_taste)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, nrow=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+
  theme(text=element_text(size=12,  family="sans"))+ labs(tag = "");pcoaplot_Fruit_taste

```

#Fig.3 & Fig.S15
```{r,echo=FALSE,warning=FALSE}
library(patchwork)
tiff("PCOA_Parameters.tiff", units="in", width=10, height=13, res=300)
PCOA_Parameters.tiff=pcoaplot_InsectRes+pcoaplot_seedweight_1000+pcoaplot_yield+pcoaplot_Fruit_long_diameter+pcoaplot_Fruit_hori_diameter+pcoaplot_Fcol+plot_layout(ncol = 2);PCOA_Parameters.tiff
ggsave("PCOA_Parameters.tiff", plot = PCOA_Parameters.tiff)
dev.off()

tiff("PCOA_Parameters1.tiff", units="in", width=8, height=8, res=300)
PCOA_Parameters1.tiff=pcoaplot_TmvRes+pcoaplot_ovaries+pcoaplot_Fshape+pcoaplot_Fruit_taste+plot_layout(ncol = 2);PCOA_Parameters1.tiff
ggsave("PCOA_Parameters1.tiff", plot = PCOA_Parameters1.tiff)
dev.off()
```

#Bacterial compositional plots for the differnt plant traits
#https://yulab-smu.top/MicrobiotaProcessWorkshop/articles/MicrobiotaProcessWorkshop.html

```{r,echo=FALSE,warning=FALSE}
library(ggplot2)
library(MicrobiotaProcess)
#citation("MicrobiotaProcess")

load("wd/ps2.RData")

classtaxa <- get_taxadf(obj=ps2, taxlevel=3);classtaxa
phylataxa <- get_taxadf(obj=ps2, taxlevel=2);phylataxa
ordertaxa <- get_taxadf(obj=ps2, taxlevel=4);ordertaxa
Famtaxa <- get_taxadf(obj=ps2, taxlevel=5);Famtaxa
#Gentaxa <- get_taxadf(obj=ps2, taxlevel=6);Gentaxa

#No.ovaries
ffam_ovaries <- ggbartax(obj=Famtaxa, facetNames="ovaries", plotgroup=TRUE, topn=30) +
  xlab("No. of Ovaries") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_ovaries

#Fruit_shape
ffam_Fruit_shape <- ggbartax(obj=Famtaxa, facetNames="Fruit_shape", plotgroup=TRUE, topn=30) +
  xlab("Berry Shape") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_Fruit_shape

#Fruit_taste
ffam_Fruit_taste <- ggbartax(obj=Famtaxa, facetNames="Fruit_taste", plotgroup=TRUE, topn=30) +
  xlab("Berry Taste") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_Fruit_taste


#Fruit_color
ffam_Fruit_color <- ggbartax(obj=Famtaxa, facetNames="Fruit_color", plotgroup=TRUE, topn=30) +
  xlab("Berry Color") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_Fruit_color


#Yield
ffam_yield <- ggbartax(obj=Famtaxa, facetNames="yield", plotgroup=TRUE, topn=30) +
  xlab("Yield (kg/sq.meter)") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_yield


#seedweight_1000
ffam_seedweight_1000 <- ggbartax(obj=Famtaxa, facetNames="seedweight_1000", plotgroup=TRUE, topn=30) +
  xlab("weight of 1000 seeds/g") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_seedweight_1000


#Fruit_hori_diameter
ffam_Fruit_hori_diameter <- ggbartax(obj=Famtaxa, facetNames="Fruit_hori_diameter", plotgroup=TRUE, topn=30) +
  xlab("Hor. Diam./cm") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_Fruit_hori_diameter


#Fruit_long_diameter
ffam_Fruit_long_diameter <- ggbartax(obj=Famtaxa, facetNames="Fruit_long_diameter", plotgroup=TRUE, topn=30) +
  xlab("Long. Diam./cm") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_Fruit_long_diameter


#TMV_resistance
ffam_TMV_resistance <- ggbartax(obj=Famtaxa, facetNames="TMV_resistance", plotgroup=TRUE, topn=30) +
  xlab("TMV Res.") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=5))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_TMV_resistance


#Insect_resistance
ffam_Insect_resistance <- ggbartax(obj=Famtaxa, facetNames="Insect_resistance", plotgroup=TRUE, topn=30) +
  xlab("Insect Res.") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 90)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=4))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=12),
        axis.text.y  = element_text(colour="black", size=12))+ labs(tag = "")+
  theme(legend.position = "")+ 
  labs(fill='Family')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+ coord_flip();ffam_Insect_resistance

#Fig.S18

tiff("Barplot_Parameters.tiff", units="in", width=15, height=11, res=300)
Barplot_Parameters.tiff=ffam_seedweight_1000+ffam_yield+ffam_Fruit_long_diameter+ffam_Fruit_hori_diameter+ffam_Fruit_color+ffam_TMV_resistance+ffam_ovaries+ffam_Fruit_shape+ffam_Fruit_taste+plot_layout(ncol = 3);Barplot_Parameters.tiff
ggsave("Barplot_Parameters.tiff", plot = Barplot_Parameters.tiff)
dev.off()
```

#Alpha diversity measures
#Normalization & alpha diversity (based on the Rhea package; Lagkouvardos et al., 2017)

#' Version 2.0
#' This script was last modified on 20/01/2020
#' Script Task: Normalize OTU-tables
#' Author: Ilias Lagkouvardos
#' Contributions by: Thomas Clavel, S&ra Reitmeier
#'
#' Normalize abundance values of the input OTU table
#' Calculate relative abundances for all OTUs based on normalized values
#' Calculate rarefaction curves to help estimate sufficiency of sequencing depth for each sample
#'
#' Input: Please enter following parameters
#' 1. Set the path to the directory where the file is stored
#' 2. Write the name of the OTU-table of interest in quotes
#'
#' The script generates five tab-delimited files & one pdf file
#' 1. Normalized counts with taxonomy information
#' 2. Normalized counts without taxonomy information
#' 3. Normalized relative abundances with taxonomy information
#' 4. Normalized relative abundances without taxonomy information
#' 5. Rarefaction curves for all samples & the most undersampled ones (default 5 cases) as PDF
#' 6. Slope of the Rarefaction curve expressed as the number of species per 100 reads
#'
#' Concept:
#' The default method followed is normalization via division by the sum of sequences in a given sample
#' & multiplication by the minimum sum across all samples. It is used instead of the classic rarefaction approach
#' to avoid unnecessary variation due to the r&om subsampling & loss of information due to rounding.
#' The option of r&om subsampling is still available for normalization if deemed necessary by users (see Set parameters section below).
#' Rarefaction curves are showing species richness with respect to sequencing depth (number of reads).
#' The depth of sequencing may be considered too low for those samples with a rarefaction curve that does not reach a plateau at the available maximum number of reads.
#' This indicates that additional, less abundant species are probably in the sample
#' but were not detected at the available depth of sequencing.
#' The terminal slope of the curve for each sample is documented in the tab-delimited file
#' as the number of species added in richness by the last 100 reads

#' Note:
#' Files are stored in the current folder
#' If a file is needed for downstream analysis, it is also automatically added to the appropriate next folder
#' under the condition that the original folder structure of Rhea is maintained.
#' If the primary evaluation of sufficient sequencing depth led to subsequent removal of samples from the analysis,
#' the normalization script should be re-run with the new OTU table.

```{r,echo=FALSE,warning=FALSE}

##################################################################################
######             Set parameters in this section manually                  ######
##################################################################################

#' Please set the directory of the script as the working folder (e.g D:/studyname/NGS-Data/Rhea/normalize/)
#' Note: the path is denoted by forward slash "/"
setwd("wd/Rhea_package/1.Normalization_2.Alpha-Diversity")    #<--- CHANGE ACCORDINGLY

#' Please give the file name of the original OTU-table with taxonomic classification
file_name <- "Table.txt"                   #<--- CHANGE ACCORDINGLY

#' Please select the normalisation method
#' 0 = No r&om subsampling, no rounding
#' 1 = R&om subsampling with rounding
method <- 1                                   #<--- CHANGE ACCORDINGLY

#' Pease select the normalization level used
#' 0 = Minimum sampling size
#' 1 = Fixed value (e.g. 1000)
level <- 1                                   #<--- CHANGE ACCORDINGLY

#' Please choose the value at which all samples will be normalized. (Only used if level selected is 1)
normCutoff <- 500

#' Please choose the number of samples with the steepest rarefaction curves to be selectively plotted
#' The default number of samples presented separately is 5
labelCutoff <- 12                             #<--- CHANGE ACCORDINGLY

######                  NO CHANGES ARE NEEDED BELOW THIS LINE               ######

##################################################################################
######                             Main Script                              ######
##################################################################################

###################       Load all required libraries     ########################

# Check if required packages are already installed, & install if missing
packages <- c("GUniFrac","vegan")

# Function to check whether the package is installed
InsPack <- function(pack)
{
  if ((pack %in% installed.packages()) == FALSE) {
    install.packages(pack,repos = "http://cloud.r-project.org/")
  }
}

# Applying the installation on the list of packages
lapply(packages, InsPack)

# Make the libraries
lib <- lapply(packages, require, character.only = TRUE)

# Check if it was possible to install all required libraries
flag <- all(as.logical(lib))

###################       Read all required input files      ####################

# Load the tab-delimited file containing the values to be be checked (rownames in the first column)
otu_table <-  read.table (file_name,
                          check.names = FALSE,
                          header = TRUE,
                          dec = ".",
                          sep = "\t",
                          row.names = 1,
                          comment.char = "")

# Making sure tha taxonomy column gets lower-case
col_names <- colnames(otu_table)
tax_ind <- which(sapply(tolower(col_names),
                        function(x) "taxonomy" %in% x, USE.NAMES = FALSE))
if (length(tax_ind) != 0) {
  col_names[tax_ind] <- tolower(col_names[tax_ind])
  colnames(otu_table) <- col_names
}
rm(col_names)
rm(tax_ind)

# Clean table from empty lines
otu_table <- otu_table[!apply(is.na(otu_table) | otu_table=="",1,all),]

####################       Normalize OTU Table          ###################


# Save taxonomy information in vector
taxonomy <- as.vector(otu_table$taxonomy)

# Delete column with taxonomy information in dataframe
otu_table$taxonomy <- NULL



if (level == 0) {
  # Calculate the minimum sum of all columns/samples
  min_sum <- min(colSums(otu_table))
} else {
  # The minimum size is set to a fixed reference level
  min_sum <- normCutoff
}


if (method == 0) {
  # Divide each value by the sum of the sample & multiply by the minimal sample sum
  norm_otu_table <- t(min_sum * t(otu_table) / colSums(otu_table))
} else {
  # Rarefy the OTU table to an equal sequencing depth
  norm_otu_table <- Rarefy(t(otu_table),depth = min_sum)
  norm_otu_table <- t(as.data.frame(norm_otu_table$otu.tab.rff))
}

# Calculate relative abundances for all OTUs over all samples
# Divide each value by the sum of the sample & multiply by 100
rel_otu_table <- t(100 * t(otu_table) / colSums(otu_table))


# Re-insert the taxonomy information in normalized counts table
norm_otu_table_tax <- cbind(norm_otu_table,taxonomy)

# Re-insert the taxonomy information in relative abundance table
rel_otu_table_tax <- cbind(rel_otu_table,taxonomy)

################################################################################
# Generate a two-sided pdf with a rarefaction curve for all samples & a curve
pdf(file = "RarefactionCurve.pdf")

# Plot the rarefaction curve for all samples
rarefactionCurve <- rarecurve(data.frame(t(otu_table)),
                              step = 20,
                              col = "black",
                              lty = "solid",
                              label = F,
                              xlab = "Number of Reads",
                              ylab = "Number of Species",
                              main = "Rarefaction Curves of All Samples")

# Generate empty vectors for the analysis of the rarefaction curve
slope = vector()
SampleID = vector()
angle <- c()

# Iterate through all samples
for (i in seq_along(rarefactionCurve)) {
  # If the sequencing depth is greater than 100, the slope of the line that passes between the last & last-100 count is calculated
  richness <- ifelse(length(rarefactionCurve[[i]]) > 6, 
                     (rarefactionCurve[[i]][length(rarefactionCurve[[i]])] - rarefactionCurve[[i]][length(rarefactionCurve[[i]])-5])/(attr(rarefactionCurve[[i]], "Subsample")[length(rarefactionCurve[[i]])]-attr(rarefactionCurve[[i]], "Subsample")[length(rarefactionCurve[[i]])-5]) , 500)
  angle[i] <- ifelse(richness!=500, atan(richness)*180/pi, NA)
  slope <- c(slope,richness)
  SampleID <- c(SampleID,as.character(names(otu_table)[i]))
}

# Generate the output table for rarefaction curve
curvedf <- cbind(SampleID, slope, angle)
ordered_vector <- order(as.numeric(curvedf[,2]), decreasing = TRUE)
curvedf <- curvedf[order(as.numeric(curvedf[,2]), decreasing = TRUE),]

# Generates a graph with all samples
# Underestimated cases are shown in red
for (i in 1:labelCutoff) {
  N <- attr(rarefactionCurve[[ordered_vector[i]]], "Subsample")
  lines(N, rarefactionCurve[[ordered_vector[i]]],col="red")
}

# Determine the plotting width & height
Nmax <- sapply(rarefactionCurve[ordered_vector[1:labelCutoff]], function(x) max(attr(x, "Subsample")))
Smax <- sapply(rarefactionCurve[ordered_vector[1:labelCutoff]], max)

# Creates an empty plot for rarefaction curves of underestimated cases
plot(c(1, max(Nmax)), c(1, max(Smax)), xlab = "Number of Reads",
     ylab = "Number of Species", type = "n", main=paste(labelCutoff,"- most undersampled cases"))

for (i in 1:labelCutoff) {
  N <- attr(rarefactionCurve[[ordered_vector[i]]], "Subsample")
  lines(N, rarefactionCurve[[ordered_vector[i]]],col="red")
  text(max(attr(rarefactionCurve[[ordered_vector[i]]],"Subsample")), max(rarefactionCurve[[ordered_vector[i]]]), curvedf[i,1],cex=0.6)
}

dev.off()

#################################################################################
######                        Write Output Files                           ######
#################################################################################

# Write the normalized table in a file & copy in directories alpha-diversity & beta-diversity if existing
write.table(norm_otu_table, "OTUs_Table-norm.tab", sep = "\t",col.names = NA, quote = FALSE)
suppressWarnings (try(write.table(norm_otu_table,"wd/Rhea_package/1.Normalization_2.Alpha-Diversity/OTUs_Table-norm.tab", sep = "\t",col.names = NA, quote = FALSE), silent =TRUE))

# Write the rarefaction table
write.table(curvedf, "RarefactionCurve.tab", sep ="\t", quote = FALSE, row.names = FALSE)

# Write the normalized table with taxonomy in a file
write.table(norm_otu_table_tax, "OTUs_Table-norm-tax.tab", sep = "\t",col.names = NA, quote = FALSE)

# Write the rarefaction table
write.table(curvedf, "RarefactionCurve.tab", sep ="\t", quote = FALSE, row.names = FALSE)

# Error message
if(!flag) { stop("
                 It was not possible to install all required R libraries properly.
                 Please check the installation of all required libraries manually.\n
                 Required libaries: GUniFrac")
}

#################################################################################
######                           End of Script                             ######
#################################################################################

#Compute alpha diversity using the output from the above normalization (OTUs_Table-norm.tab)

#' Version 2.0
#' Last modified on 20/01/2020
#' Script Task: Calculate alpha-diversity
#' Author: Ilias Lagkouvardos
#' Contributions by: Thomas Clavel, S&ra Reitmeier
#'
#' For meaningful comparisons of species richness across samples, 
#' use of normalized sequence counts is expected.
#' For normalized richness calculation, only OTUs that are above 0.5 normalized counts are considered.
#' For comparisons across studies the notions of "Normalized" & "Effective" richness are being used.
#' Effective Richness is the number of OTUs occurring above the relative abundance of 0.25% in a given sample.
#' This threshold was experimentally determined using Mock communities in vitro & in vivo & 
#' is efficient in removing spurious taxa from amplicon sequencing data.
#' Normalized Richness is the number of OTUs above 0.5 counts in a sample when normalized to a fixed depth (1000)
#' 
#' Note:
#' Files are stored in the current folder 
#' If a file is needed for downstream analysis, it is also automatically added to the appropriate next folder
#' under the condition that the original folder structure of Rhea is maintained.

##################################################################################
######             Set parameters in this section manually                  ######
##################################################################################

#' Please set the directory of the script as working folder (e.g D:/MyStudy/NGS/Rhea/alpha-diversity)
#' Note: the path is denoted by forward slash "/"
setwd("wd/Rhea/2.Alpha-Diversity/Diversity_500_rarefaction")    #<--- CHANGE ACCORDINGLY

#' Please give the file name of the normalized OTU-table (without taxonomic classification)
file_name <- "OTUs_Table-norm.tab"                #<--- CHANGE ACCORDINGLY

#' The abundance filtering cutoff 
eff.cutoff <- 0.0025 # this is the default value for Effective Richness (0.25%)

#' The normalized depth cutoff
norm.cutoff <- 500 # this is the default value for St&ard Richness (1000)


######                  NO CHANGES ARE NEEDED BELOW THIS LINE               ######

##################################################################################
######                        Diversity Functions                           ###### 
##################################################################################

# Calculate the species richness in a sample
Species.richness <- function(x)
{
  # Count only the OTUs that are present >0.5 normalized counts (normalization produces real values for counts)
  count=sum(x[x>0.5]^0)
  return(count)
}

# Calculate the Effective species richness in each individual sample
Eff.Species.richness <- function(x)
{
  # Count only the OTUs that are present more than the set proportion
  total=sum(x)
  count=sum(x[x/total>eff.cutoff]^0)
  return(count)
}

# Calculate the Normalized species richness in each individual sample
Norm.Species.richness <- function(x)
{
  # Count only the OTUs that are present >0.5 normalized counts (normalization produces real values for counts)
  # Given a fixed Normalization reads depth
  total=sum(x)
  count=sum(x[norm.cutoff*x/total>0.5]^0)
  return(count)
}


# Calculate the Shannon diversity index
Shannon.entropy <- function(x)
{
  total=sum(x)
  se=-sum(x[x>0]/total*log(x[x>0]/total))
  return(se)
}

# Calculate the effective number of species for Shannon
Shannon.effective <- function(x)
{
  total=sum(x)
  se=round(exp(-sum(x[x>0]/total*log(x[x>0]/total))),digits =2)
  return(se)
}

# Calculate the Simpson diversity index
Simpson.concentration <- function(x)
{
  total=sum(x)
  si=sum((x[x>0]/total)^2)
  return(si)
}

# Calculate the effective number of species for Simpson
Simpson.effective <- function(x)
{
  total=sum(x)
  si=round(1/sum((x[x>0]/total)^2),digits =2)
  return(si)
}

##################################################################################
######                             Main Script                              ###### 
##################################################################################

# Read a normalized OTU-table without taxonomy  
otu_table <- read.table ("wd/Rhea_package/1.Normalization_2.Alpha-Diversity/OTUs_Table-norm.tab", 
                       check.names = FALSE, 
                       header=TRUE, 
                       dec=".", 
                       sep = "\t",
                       row.names = 1);otu_table

# Clean table from empty lines
otu_table <- otu_table[!apply(is.na(otu_table) | otu_table=="",1,all),];otu_table

# Order & transpose OTU-table
my_otu_table <- otu_table[,order(names(otu_table))];my_otu_table
my_otu_table <-data.frame(t(my_otu_table));my_otu_table

# Apply diversity functions to table
otus_div_stats<-data.frame(my_otu_table[,]);otus_div_stats
otus_div_stats$Richness<-apply(my_otu_table,1,Species.richness)
otus_div_stats$Normalized.Richness<-apply(my_otu_table,1,Norm.Species.richness)
otus_div_stats$Effective.Richness<-apply(my_otu_table,1,Eff.Species.richness)
otus_div_stats$Shannon.Index<-apply(my_otu_table,1,Shannon.entropy)
otus_div_stats$Shannon.Effective<-apply(my_otu_table,1,Shannon.effective)
otus_div_stats$Simpson.Index<-apply(my_otu_table,1,Simpson.concentration)
otus_div_stats$Simpson.Effective<-apply(my_otu_table,1,Simpson.effective)
otus_div_stats$Evenness <- otus_div_stats$Shannon.Index/log(otus_div_stats$Richness,2)

setwd("wd/Rhea_package/1.Normalization_2.Alpha-Diversity")
# Write the results in a file & copy in directory "Serial-Group-Comparisons" if existing
write.table(otus_div_stats, "alpha-diversity.tab", sep="\t", col.names=NA, quote=FALSE)
suppressWarnings (try(write.table(otus_div_stats[c(1,2,3,5,7)], "../Rhea_package/1.Normalization_2.Alpha-Diversity/alpha-diversity.tab", sep = "\t",col.names = NA, quote = FALSE), silent =TRUE))

##################################################################################
######                          End of Script                               ###### 
##################################################################################
```

```{r,echo=FALSE,warning=FALSE}
load("wd/ps2.RData")

#calculate community phylogenetic diversity & merge it to the figure above
summary(sample_sums(ps2))#[ 15421 taxa & 1197 samples ]
Bacteria_subsamp <- rarefy_even_depth(ps2, sample.size=500 , rngseed=5163, replace=FALSE, trimOTUs=TRUE);Bacteria_subsamp#[ 1128 samples by 20 sample variables ]
#Lost Samples after rarefaction: 69 samples

#pd=estimate_pd(Bacteria_subsamp);pd
#write.csv(pd, file="pd.csv")

```

#Statistics for alpha diversity
##Import the rarefied dataset & compute the statistics
#Load additional libraries
```{r,echo=FALSE,warning=FALSE}
setwd("wd")

library(microViz)
library(ggtext)
library(microViz)
library(ggraph)
library(DT)
library(corncob)
library(ggplot2)
library(patchwork)
library(ggplot2) # 3.2.1
library(dplyr)
library(lubridate)
library(cowplot) # 1.0.0
library(egg) # 0.4.5
library(microbiome)
library(knitr)
library(tidyplots)
library(rcompanion)
library(multcompView)
library(tidyverse)
library(ggpubr)
library(dunn.test)
library(FSA)
#install.packages("explore")
library(explore)

#Diversity was computed for using the functions mentioned in Rhea package considering ASVs with abundance >0.25%
#Load the alpha diversity data (rarefied at 500 depth)#alpha-diversity_500.tab
#Combine the pd on to  dataframe for other alpha diversity at 500 sampling depth (alpha-diversity_500.tab)
alpha_500_PD <- read.table ("alpha-diversity_500.tab.txt", check.names = FALSE, header = TRUE, dec = ".", sep = "\t", row.names = 1, comment.char = "");alpha_500_PD
alpha_500_PD$yield <- cut(alpha_500_PD$`Yield_(kg/sq_meter)`, breaks = seq(0,150,20), include.lowest = TRUE)
alpha_500_PD$yield
alpha_500_PD$Fruit_long_diameter=cut(alpha_500_PD$Fruit_longitudinal_diamete, breaks = seq(0,10,3), include.lowest = TRUE)
alpha_500_PD$Fruit_long_diameter
alpha_500_PD$Fruit_hori_diameter=cut(alpha_500_PD$Fruit_horizontal_diameter, breaks = seq(0,10,3), include.lowest = TRUE)
alpha_500_PD$Fruit_hori_diameter
alpha_500_PD$seedweight_1000=cut(alpha_500_PD$`1000_seeds_weight_(units)`, breaks = seq(0,5,1), include.lowest = TRUE)
alpha_500_PD$seedweight_1000
alpha_500_PD$ovaries =cut(alpha_500_PD$Number_of_ovary, breaks = seq(0,12,4), include.lowest = TRUE)
alpha_500_PD$ovaries
#write_csv(alpha_500_PD,"alpha_500_PD.csv")
#subset for the different parameters
alpha_500_PD_Fruit_taste=alpha_500_PD[!is.na(alpha_500_PD$Fruit_taste),];alpha_500_PD_Fruit_taste
alpha_500_PD_Fruit_color=alpha_500_PD[!is.na(alpha_500_PD$Fruit_color),];alpha_500_PD_Fruit_color
alpha_500_PD_Fruit_shape=alpha_500_PD[!is.na(alpha_500_PD$Fruit_shape),];alpha_500_PD_Fruit_shape
alpha_500_PD_TMV_resistance=alpha_500_PD[!is.na(alpha_500_PD$TMV_resistance),];alpha_500_PD_TMV_resistance
alpha_500_PD_Insect_resistance=alpha_500_PD[!is.na(alpha_500_PD$Insect_resistance),];alpha_500_PD_Insect_resistance
alpha_500_PD_yield=alpha_500_PD[!is.na(alpha_500_PD$yield),];alpha_500_PD_yield
alpha_500_PD_seedweight_1000=alpha_500_PD[!is.na(alpha_500_PD$seedweight_1000),];alpha_500_PD_seedweight_1000
alpha_500_PD_ovaries=alpha_500_PD[!is.na(alpha_500_PD$ovaries),];alpha_500_PD_ovaries
alpha_500_PD_ovaries=alpha_500_PD[!is.na(alpha_500_PD$ovaries),];alpha_500_PD_ovaries
alpha_500_PD_Fruit_hori_diameter=alpha_500_PD[!is.na(alpha_500_PD$Fruit_hori_diameter),];alpha_500_PD_Fruit_hori_diameter
alpha_500_PD_Fruit_long_diameter=alpha_500_PD[!is.na(alpha_500_PD$Fruit_long_diameter),];alpha_500_PD_Fruit_long_diameter

attach(alpha_500_PD)
########################################################################################################################
#https://www.rdocumentation.org/packages/rstatix/versions/0.7.2
#citation("rstatix ")

#explore(alpha_500_PD)
# report of all variables
#alpha_500_PD %>% report(output_file = "report.html", output_dir = tempdir())
#alpha_500_PD %>% get_summary_stats(type = "common")
##########################################################################################################################
x=alpha_500_PD %>%
  group_by(English_Name_2, Production_site_2) %>%
  get_summary_stats(Shannon_Index, type = "common");x
write_csv(x,"x.csv")


alpha_500_PD %>%
    get_summary_stats(Shannon_Index, show = c("mean", "sd", "median", "iqr"))

alpha_500_PD$Shannon_Index
```

#consider rhe richness index
```{r,echo=FALSE,warning=FALSE}

# RICHNESS
kruskal.test(Richness ~ English_Name_2, data = alpha_500_PD)

kruskal.test(Richness ~ Production_site_2, data = alpha_500_PD)

kruskal.test(Richness ~ Fruit_taste, data = alpha_500_PD_Fruit_taste)

kruskal.test(Richness ~ Fruit_color, data = alpha_500_PD_Fruit_color)

kruskal.test(Richness ~ Fruit_shape, data = alpha_500_PD_Fruit_shape)

kruskal.test(Richness ~ TMV_resistance, data = alpha_500_PD_TMV_resistance)

kruskal.test(Richness ~ Insect_resistance, data = alpha_500_PD_Insect_resistance)

kruskal.test(Richness ~ seedweight_1000, data = alpha_500_PD_seedweight_1000)

kruskal.test(Richness ~ ovaries, data = alpha_500_PD_ovaries)

#yield
kruskal.test(Richness ~ yield, data = alpha_500_PD_yield)

#Fruit_hori_diameter
kruskal.test(Richness ~ Fruit_hori_diameter, data = alpha_500_PD_Fruit_hori_diameter)

#Fruit_long_diameter
kruskal.test(Richness ~ Fruit_long_diameter, data = alpha_500_PD_Fruit_long_diameter)

#POST-HOC ANALYSIS WITH DUNN'S TEST

#Region of production
Dunn_tomato_richness_Production_site_2=dunn_test(alpha_500_PD, Richness ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_richness_Production_site_2
write_csv(Dunn_tomato_richness_Production_site_2, "Dunn_tomato_richness_Production_site_2.csv")
Dunn_tomato_richness_Production_site_2 <- dunnTest(alpha_500_PD$Richness, alpha_500_PD$Production_site_2, method = "bonferroni");Dunn_tomato_richness_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_richness_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_richness_Production_site_2$res);CLD_Dunn_tomato_richness_Production_site_2 

#cultivars
Dunn_tomato_richness_English_Name_2=dunn_test(alpha_500_PD, Richness ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_richness_English_Name_2
write_csv(Dunn_tomato_richness_English_Name_2, "Dunn_tomato_richness_English_Name_2.csv")
Dunn_tomato_richness_English_Name_2 <- dunnTest(alpha_500_PD$Richness, alpha_500_PD$English_Name, method = "bonferroni");Dunn_tomato_richness_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_richness_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_richness_English_Name_2$res);CLD_Dunn_tomato_richness_English_Name_2 

#Taste
Dunn_tomato_Richness_Fruit_taste=dunn_test(alpha_500_PD_Fruit_taste, Richness ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_taste
write_csv(Dunn_tomato_Richness_Fruit_taste, "Dunn_tomato_Richness_Fruit_taste.csv")
Dunn_tomato_Richness_Fruit_taste <- dunnTest(alpha_500_PD_Fruit_taste$Richness, alpha_500_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Richness_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_taste$res);CLD_Dunn_tomato_Richness_Fruit_taste 

#COLOR
Dunn_tomato_Richness_Fruit_color=dunn_test(alpha_500_PD_Fruit_color, Richness ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_color
write_csv(Dunn_tomato_Richness_Fruit_color, "Dunn_tomato_Richness_Fruit_color.csv")
Dunn_tomato_Richness_Fruit_color <- dunnTest(alpha_500_PD_Fruit_color$Richness, alpha_500_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Richness_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_color$res);CLD_Dunn_tomato_Richness_Fruit_color 

#insect resistance
Dunn_tomato_Richness_Insect_resistance=dunn_test(alpha_500_PD_Insect_resistance, Richness ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Insect_resistance
write_csv(Dunn_tomato_Richness_Insect_resistance, "Dunn_tomato_Richness_Insect_resistance.csv")
Dunn_tomato_Richness_Insect_resistance <- dunnTest(alpha_500_PD_Insect_resistance$Richness, alpha_500_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Richness_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Insect_resistance$res);CLD_Dunn_tomato_Richness_Insect_resistance 

#TMV RESISTANCE
Dunn_tomato_Richness_TMV_resistance=dunn_test(alpha_500_PD_TMV_resistance, Richness ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_TMV_resistance
write_csv(Dunn_tomato_Richness_TMV_resistance, "Dunn_tomato_Richness_TMV_resistance.csv")
Dunn_tomato_Richness_TMV_resistance <- dunnTest(alpha_500_PD_TMV_resistance$Richness, alpha_500_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Richness_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_TMV_resistance$res);CLD_Dunn_tomato_Richness_TMV_resistance 

#seed weight
Dunn_tomato_Richness_seedweight_1000=dunn_test(alpha_500_PD_seedweight_1000, Richness ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_seedweight_1000
write_csv(Dunn_tomato_Richness_seedweight_1000, "Dunn_tomato_Richness_seedweight_1000.csv")
Dunn_tomato_Richness_seedweight_1000 <- dunnTest(alpha_500_PD_seedweight_1000$Richness, alpha_500_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Richness_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_seedweight_1000$res);CLD_Dunn_tomato_Richness_seedweight_1000 

#seed weight
Dunn_tomato_Richness_Fruit_shape=dunn_test(alpha_500_PD_Fruit_shape, Richness ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_shape
write_csv(Dunn_tomato_Richness_Fruit_shape, "Dunn_tomato_Richness_Fruit_shape.csv")
Dunn_tomato_Richness_Fruit_shape <- dunnTest(alpha_500_PD_Fruit_shape$Richness, alpha_500_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Richness_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_shape$res);CLD_Dunn_tomato_Richness_Fruit_shape 

#ovaries

Dunn_tomato_Richness_ovaries=dunn_test(alpha_500_PD_ovaries, Richness ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_ovaries
write_csv(Dunn_tomato_Richness_ovaries, "Dunn_tomato_Richness_ovaries.csv")
Dunn_tomato_Richness_ovaries <- dunnTest(alpha_500_PD_ovaries$Richness, alpha_500_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Richness_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_ovaries$res);CLD_Dunn_tomato_Richness_ovaries 

#yield
Dunn_tomato_Richness_yield=dunn_test(alpha_500_PD_yield, Richness ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_yield
write_csv(Dunn_tomato_Richness_yield, "Dunn_tomato_Richness_yield.csv")
Dunn_tomato_Richness_yield <- dunnTest(alpha_500_PD_yield$Richness, alpha_500_PD_yield$yield, method = "bonferroni");Dunn_tomato_Richness_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_yield$res);CLD_Dunn_tomato_Richness_yield 

#Fruit_hori_diameter
Dunn_tomato_Richness_Fruit_hori_diameter=dunn_test(alpha_500_PD_Fruit_hori_diameter, Richness ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_hori_diameter
write_csv(Dunn_tomato_Richness_Fruit_hori_diameter, "Dunn_tomato_Richness_Fruit_hori_diameter.csv")
Dunn_tomato_Richness_Fruit_hori_diameter <- dunnTest(alpha_500_PD_Fruit_hori_diameter$Richness, alpha_500_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Richness_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_hori_diameter$res);CLD_Dunn_tomato_Richness_Fruit_hori_diameter 

#Fruit_long_diameter
Dunn_tomato_Richness_Fruit_long_diameter=dunn_test(alpha_500_PD_Fruit_long_diameter, Richness ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Richness_Fruit_long_diameter
write_csv(Dunn_tomato_Richness_Fruit_long_diameter, "Dunn_tomato_Richness_Fruit_long_diameter.csv")
Dunn_tomato_Richness_Fruit_long_diameter <- dunnTest(alpha_500_PD_Fruit_long_diameter$Richness, alpha_500_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Richness_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Richness_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Richness_Fruit_long_diameter$res);CLD_Dunn_tomato_Richness_Fruit_long_diameter 
```
 
 
#EFFECTIVE RICHNESS
```{r,echo=FALSE,warning=FALSE}
setwd("wd")

kruskal.test(Effective_Richness ~ English_Name_2, data = alpha_500_PD)

kruskal.test(Effective_Richness ~ Production_site_2, data = alpha_500_PD)

kruskal.test(Effective_Richness ~ Fruit_taste, data = alpha_500_PD_Fruit_taste)

kruskal.test(Effective_Richness ~ Fruit_color, data = alpha_500_PD_Fruit_color)

kruskal.test(Effective_Richness ~ Fruit_shape, data = alpha_500_PD_Fruit_shape)

kruskal.test(Effective_Richness ~ TMV_resistance, data = alpha_500_PD_TMV_resistance)

kruskal.test(Effective_Richness ~ Insect_resistance, data = alpha_500_PD_Insect_resistance)

kruskal.test(Effective_Richness ~ seedweight_1000, data = alpha_500_PD_seedweight_1000)

kruskal.test(Effective_Richness ~ ovaries, data = alpha_500_PD_ovaries)

kruskal.test(Effective_Richness ~ yield, data = alpha_500_PD_yield)

kruskal.test(Effective_Richness ~ Fruit_hori_diameter, data = alpha_500_PD_Fruit_hori_diameter)

kruskal.test(Effective_Richness ~ Fruit_long_diameter, data = alpha_500_PD_Fruit_long_diameter)

#Post-hoc tests (Dunn's test)

#Region of production)
Dunn_tomato_Effective_Richness_Production_site_2=dunn_test(alpha_500_PD, Effective_Richness ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Production_site_2
write_csv(Dunn_tomato_Effective_Richness_Production_site_2, "Dunn_tomato_Effective_Richness_Production_site_2.csv")
Dunn_tomato_Effective_Richness_Production_site_2 <- dunnTest(alpha_500_PD$Effective_Richness, alpha_500_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Effective_Richness_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Production_site_2$res);CLD_Dunn_tomato_Effective_Richness_Production_site_2 

#Cultivar
Dunn_tomato_Effective_Richness_English_Name_2=dunn_test(alpha_500_PD, Effective_Richness ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_English_Name_2
write_csv(Dunn_tomato_Effective_Richness_English_Name_2, "Dunn_tomato_Effective_Richness_English_Name_2.csv")
Dunn_tomato_Effective_Richness_English_Name_2 <- dunnTest(alpha_500_PD$Effective_Richness, alpha_500_PD$English_Name, method = "bonferroni");Dunn_tomato_Effective_Richness_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_English_Name_2$res);CLD_Dunn_tomato_Effective_Richness_English_Name_2 

#taste
Dunn_tomato_Effective_Richness_Fruit_taste=dunn_test(alpha_500_PD_Fruit_taste, Effective_Richness ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_taste
write_csv(Dunn_tomato_Effective_Richness_Fruit_taste, "Dunn_tomato_Effective_Richness_Fruit_taste.csv")
Dunn_tomato_Effective_Richness_Fruit_taste <- dunnTest(alpha_500_PD_Fruit_taste$Effective_Richness, alpha_500_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_taste$res);CLD_Dunn_tomato_Effective_Richness_Fruit_taste 

#color
Dunn_tomato_Effective_Richness_Fruit_color=dunn_test(alpha_500_PD_Fruit_color, Effective_Richness ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_color
write_csv(Dunn_tomato_Effective_Richness_Fruit_color, "Dunn_tomato_Effective_Richness_Fruit_color.csv")
Dunn_tomato_Effective_Richness_Fruit_color <- dunnTest(alpha_500_PD_Fruit_color$Effective_Richness, alpha_500_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_color$res);CLD_Dunn_tomato_Effective_Richness_Fruit_color 

#insect resistance
Dunn_tomato_Effective_Richness_Insect_resistance=dunn_test(alpha_500_PD_Insect_resistance, Effective_Richness ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Insect_resistance
write_csv(Dunn_tomato_Effective_Richness_Insect_resistance, "Dunn_tomato_Effective_Richness_Insect_resistance.csv")
Dunn_tomato_Effective_Richness_Insect_resistance <- dunnTest(alpha_500_PD_Insect_resistance$Effective_Richness, alpha_500_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Effective_Richness_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Insect_resistance$res);CLD_Dunn_tomato_Effective_Richness_Insect_resistance 

#TMV RESISTANCE
Dunn_tomato_Effective_Richness_TMV_resistance=dunn_test(alpha_500_PD_TMV_resistance, Effective_Richness ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_TMV_resistance
write_csv(Dunn_tomato_Effective_Richness_TMV_resistance, "Dunn_tomato_Effective_Richness_TMV_resistance.csv")
Dunn_tomato_Effective_Richness_TMV_resistance <- dunnTest(alpha_500_PD_TMV_resistance$Effective_Richness, alpha_500_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Effective_Richness_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_TMV_resistance$res);CLD_Dunn_tomato_Effective_Richness_TMV_resistance 

#seed weight
Dunn_tomato_Effective_Richness_seedweight_1000=dunn_test(alpha_500_PD_seedweight_1000, Effective_Richness ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_seedweight_1000
write_csv(Dunn_tomato_Effective_Richness_seedweight_1000, "Dunn_tomato_Effective_Richness_seedweight_1000.csv")
Dunn_tomato_Effective_Richness_seedweight_1000 <- dunnTest(alpha_500_PD_seedweight_1000$Effective_Richness, alpha_500_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Effective_Richness_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_seedweight_1000$res);CLD_Dunn_tomato_Effective_Richness_seedweight_1000 

#seed SHAPE
Dunn_tomato_Effective_Richness_Fruit_shape=dunn_test(alpha_500_PD_Fruit_shape, Effective_Richness ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_shape
write_csv(Dunn_tomato_Effective_Richness_Fruit_shape, "Dunn_tomato_Effective_Richness_Fruit_shape.csv")
Dunn_tomato_Effective_Richness_Fruit_shape <- dunnTest(alpha_500_PD_Fruit_shape$Effective_Richness, alpha_500_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_shape$res);CLD_Dunn_tomato_Effective_Richness_Fruit_shape 

#ovaries
Dunn_tomato_Effective_Richness_ovaries=dunn_test(alpha_500_PD_ovaries, Effective_Richness ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_ovaries
write_csv(Dunn_tomato_Effective_Richness_ovaries, "Dunn_tomato_Effective_Richness_ovaries.csv")
Dunn_tomato_Effective_Richness_ovaries <- dunnTest(alpha_500_PD_ovaries$Effective_Richness, alpha_500_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Effective_Richness_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_ovaries$res);CLD_Dunn_tomato_Effective_Richness_ovaries 

#yield
Dunn_tomato_Effective_Richness_yield=dunn_test(alpha_500_PD_yield, Effective_Richness ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_yield
write_csv(Dunn_tomato_Effective_Richness_yield, "Dunn_tomato_Effective_Richness_yield.csv")
Dunn_tomato_Effective_Richness_yield <- dunnTest(alpha_500_PD_yield$Effective_Richness, alpha_500_PD_yield$yield, method = "bonferroni");Dunn_tomato_Effective_Richness_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_yield$res);CLD_Dunn_tomato_Effective_Richness_yield 

#Fruit_hori_diameter
Dunn_tomato_Effective_Richness_Fruit_hori_diameter=dunn_test(alpha_500_PD_Fruit_hori_diameter, Effective_Richness ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_hori_diameter
write_csv(Dunn_tomato_Effective_Richness_Fruit_hori_diameter, "Dunn_tomato_Effective_Richness_Fruit_hori_diameter.csv")
Dunn_tomato_Effective_Richness_Fruit_hori_diameter <- dunnTest(alpha_500_PD_Fruit_hori_diameter$Effective_Richness, alpha_500_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_hori_diameter$res);CLD_Dunn_tomato_Effective_Richness_Fruit_hori_diameter 

#Fruit_long_diameter
Dunn_tomato_Effective_Richness_Fruit_long_diameter=dunn_test(alpha_500_PD_Fruit_long_diameter, Effective_Richness ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Effective_Richness_Fruit_long_diameter
write_csv(Dunn_tomato_Effective_Richness_Fruit_long_diameter, "Dunn_tomato_Effective_Richness_Fruit_long_diameter.csv")
Dunn_tomato_Effective_Richness_Fruit_long_diameter <- dunnTest(alpha_500_PD_Fruit_long_diameter$Effective_Richness, alpha_500_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Effective_Richness_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Effective_Richness_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Effective_Richness_Fruit_long_diameter$res);CLD_Dunn_tomato_Effective_Richness_Fruit_long_diameter 
###########################################################################################

```
 
 
#SHANNON INDEX
```{r,echo=FALSE,warning=FALSE}
setwd("wd")
kruskal.test(Shannon_Index ~ English_Name_2, data = alpha_500_PD)

kruskal.test(Shannon_Index ~ Production_site_2, data = alpha_500_PD)

kruskal.test(Shannon_Index ~ Fruit_taste, data = alpha_500_PD_Fruit_taste)

kruskal.test(Shannon_Index ~ Fruit_color, data = alpha_500_PD_Fruit_color)

kruskal.test(Shannon_Index ~ Fruit_shape, data = alpha_500_PD_Fruit_shape)

kruskal.test(Shannon_Index ~ TMV_resistance, data = alpha_500_PD_TMV_resistance)

kruskal.test(Shannon_Index ~ Insect_resistance, data = alpha_500_PD_Insect_resistance)

kruskal.test(Shannon_Index ~ seedweight_1000, data = alpha_500_PD_seedweight_1000)

kruskal.test(Shannon_Index ~ ovaries, data = alpha_500_PD_ovaries)

kruskal.test(Shannon_Index ~ yield, data = alpha_500_PD_yield)

kruskal.test(Shannon_Index ~ Fruit_hori_diameter, data = alpha_500_PD_Fruit_hori_diameter)

kruskal.test(Shannon_Index ~ Fruit_long_diameter, data = alpha_500_PD_Fruit_long_diameter)

#Post-hoc (Dunn's Test)

#Region of seed production
Dunn_tomato_Shannon_Index_Production_site_2=dunn_test(alpha_500_PD, Shannon_Index ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Production_site_2
write_csv(Dunn_tomato_Shannon_Index_Production_site_2, "Dunn_tomato_Shannon_Index_Production_site_2.csv")
Dunn_tomato_Shannon_Index_Production_site_2 <- dunnTest(alpha_500_PD$Shannon_Index, alpha_500_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Shannon_Index_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Production_site_2$res);CLD_Dunn_tomato_Shannon_Index_Production_site_2 

#cultivar
Dunn_tomato_Shannon_Index_English_Name_2=dunn_test(alpha_500_PD, Shannon_Index ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_English_Name_2
write_csv(Dunn_tomato_Shannon_Index_English_Name_2, "Dunn_tomato_Shannon_Index_English_Name_2.csv")
Dunn_tomato_Shannon_Index_English_Name_2 <- dunnTest(alpha_500_PD$Shannon_Index, alpha_500_PD$English_Name, method = "bonferroni");Dunn_tomato_Shannon_Index_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_English_Name_2$res);CLD_Dunn_tomato_Shannon_Index_English_Name_2 

#Taste
Dunn_tomato_Shannon_Index_Fruit_taste=dunn_test(alpha_500_PD_Fruit_taste, Shannon_Index ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_taste
write_csv(Dunn_tomato_Shannon_Index_Fruit_taste, "Dunn_tomato_Shannon_Index_Fruit_taste.csv")
Dunn_tomato_Shannon_Index_Fruit_taste <- dunnTest(alpha_500_PD_Fruit_taste$Shannon_Index, alpha_500_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_taste$res);CLD_Dunn_tomato_Shannon_Index_Fruit_taste 


#color
Dunn_tomato_Shannon_Index_Fruit_color=dunn_test(alpha_500_PD_Fruit_color, Shannon_Index ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_color
write_csv(Dunn_tomato_Shannon_Index_Fruit_color, "Dunn_tomato_Shannon_Index_Fruit_color.csv")
Dunn_tomato_Shannon_Index_Fruit_color <- dunnTest(alpha_500_PD_Fruit_color$Shannon_Index, alpha_500_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_color$res);CLD_Dunn_tomato_Shannon_Index_Fruit_color 

#insect resistance
Dunn_tomato_Shannon_Index_Insect_resistance=dunn_test(alpha_500_PD_Insect_resistance, Shannon_Index ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Insect_resistance
write_csv(Dunn_tomato_Shannon_Index_Insect_resistance, "Dunn_tomato_Shannon_Index_Insect_resistance.csv")
Dunn_tomato_Shannon_Index_Insect_resistance <- dunnTest(alpha_500_PD_Insect_resistance$Shannon_Index, alpha_500_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Shannon_Index_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Insect_resistance$res);CLD_Dunn_tomato_Shannon_Index_Insect_resistance 

#TMV RESISTANCE
Dunn_tomato_Shannon_Index_TMV_resistance=dunn_test(alpha_500_PD_TMV_resistance, Shannon_Index ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_TMV_resistance
write_csv(Dunn_tomato_Shannon_Index_TMV_resistance, "Dunn_tomato_Shannon_Index_TMV_resistance.csv")
Dunn_tomato_Shannon_Index_TMV_resistance <- dunnTest(alpha_500_PD_TMV_resistance$Shannon_Index, alpha_500_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Shannon_Index_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_TMV_resistance$res);CLD_Dunn_tomato_Shannon_Index_TMV_resistance 

#seed weight
Dunn_tomato_Shannon_Index_seedweight_1000=dunn_test(alpha_500_PD_seedweight_1000, Shannon_Index ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_seedweight_1000
write_csv(Dunn_tomato_Shannon_Index_seedweight_1000, "Dunn_tomato_Shannon_Index_seedweight_1000.csv")
Dunn_tomato_Shannon_Index_seedweight_1000 <- dunnTest(alpha_500_PD_seedweight_1000$Shannon_Index, alpha_500_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Shannon_Index_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_seedweight_1000$res);CLD_Dunn_tomato_Shannon_Index_seedweight_1000 


#seed SHAPE
Dunn_tomato_Shannon_Index_Fruit_shape=dunn_test(alpha_500_PD_Fruit_shape, Shannon_Index ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_shape
write_csv(Dunn_tomato_Shannon_Index_Fruit_shape, "Dunn_tomato_Shannon_Index_Fruit_shape.csv")
Dunn_tomato_Shannon_Index_Fruit_shape <- dunnTest(alpha_500_PD_Fruit_shape$Shannon_Index, alpha_500_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_shape$res);CLD_Dunn_tomato_Shannon_Index_Fruit_shape 

#ovaries
Dunn_tomato_Shannon_Index_ovaries=dunn_test(alpha_500_PD_ovaries, Shannon_Index ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_ovaries
write_csv(Dunn_tomato_Shannon_Index_ovaries, "Dunn_tomato_Shannon_Index_ovaries.csv")
Dunn_tomato_Shannon_Index_ovaries <- dunnTest(alpha_500_PD_ovaries$Shannon_Index, alpha_500_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Shannon_Index_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_ovaries$res);CLD_Dunn_tomato_Shannon_Index_ovaries 

#yield
Dunn_tomato_Shannon_Index_yield=dunn_test(alpha_500_PD_yield, Shannon_Index ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_yield
write_csv(Dunn_tomato_Shannon_Index_yield, "Dunn_tomato_Shannon_Index_yield.csv")
Dunn_tomato_Shannon_Index_yield <- dunnTest(alpha_500_PD_yield$Shannon_Index, alpha_500_PD_yield$yield, method = "bonferroni");Dunn_tomato_Shannon_Index_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_yield$res);CLD_Dunn_tomato_Shannon_Index_yield 

#Fruit_hori_diameter
Dunn_tomato_Shannon_Index_Fruit_hori_diameter=dunn_test(alpha_500_PD_Fruit_hori_diameter, Shannon_Index ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_hori_diameter
write_csv(Dunn_tomato_Shannon_Index_Fruit_hori_diameter, "Dunn_tomato_Shannon_Index_Fruit_hori_diameter.csv")
Dunn_tomato_Shannon_Index_Fruit_hori_diameter <- dunnTest(alpha_500_PD_Fruit_hori_diameter$Shannon_Index, alpha_500_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_hori_diameter$res);CLD_Dunn_tomato_Shannon_Index_Fruit_hori_diameter 

#Fruit_long_diameter
Dunn_tomato_Shannon_Index_Fruit_long_diameter=dunn_test(alpha_500_PD_Fruit_long_diameter, Shannon_Index ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Index_Fruit_long_diameter
write_csv(Dunn_tomato_Shannon_Index_Fruit_long_diameter, "Dunn_tomato_Shannon_Index_Fruit_long_diameter.csv")
Dunn_tomato_Shannon_Index_Fruit_long_diameter <- dunnTest(alpha_500_PD_Fruit_long_diameter$Shannon_Index, alpha_500_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Shannon_Index_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Index_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Index_Fruit_long_diameter$res);CLD_Dunn_tomato_Shannon_Index_Fruit_long_diameter 
```
 

#Effective Shannon-diversity index
 
```{r,echo=FALSE,warning=FALSE}
setwd("wd")

kruskal.test(Shannon_Effective ~ English_Name_2, data = alpha_500_PD)

kruskal.test(Shannon_Effective ~ Production_site_2, data = alpha_500_PD)

kruskal.test(Shannon_Effective ~ Fruit_taste, data = alpha_500_PD_Fruit_taste)

kruskal.test(Shannon_Effective ~ Fruit_color, data = alpha_500_PD_Fruit_color)

kruskal.test(Shannon_Effective ~ Fruit_shape, data = alpha_500_PD_Fruit_shape)

kruskal.test(Shannon_Effective ~ TMV_resistance, data = alpha_500_PD_TMV_resistance)

kruskal.test(Shannon_Effective ~ Insect_resistance, data = alpha_500_PD_Insect_resistance)

kruskal.test(Shannon_Effective ~ seedweight_1000, data = alpha_500_PD_seedweight_1000)

kruskal.test(Shannon_Effective ~ ovaries, data = alpha_500_PD_ovaries)

kruskal.test(Shannon_Effective ~ yield, data = alpha_500_PD_yield)

kruskal.test(Shannon_Effective ~ Fruit_hori_diameter, data = alpha_500_PD_Fruit_hori_diameter)

kruskal.test(Shannon_Effective ~ Fruit_long_diameter, data = alpha_500_PD_Fruit_long_diameter)

#Region of tomato seed production
# Post-hoc (Dunn's Test)
Dunn_tomato_Shannon_Effective_Production_site_2=dunn_test(alpha_500_PD, Shannon_Effective ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Production_site_2
write_csv(Dunn_tomato_Shannon_Effective_Production_site_2, "Dunn_tomato_Shannon_Effective_Production_site_2.csv")
Dunn_tomato_Shannon_Effective_Production_site_2 <- dunnTest(alpha_500_PD$Shannon_Effective, alpha_500_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Shannon_Effective_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Production_site_2$res);CLD_Dunn_tomato_Shannon_Effective_Production_site_2 

#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_Shannon_Effective_English_Name_2=dunn_test(alpha_500_PD, Shannon_Effective ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_English_Name_2
write_csv(Dunn_tomato_Shannon_Effective_English_Name_2, "Dunn_tomato_Shannon_Effective_English_Name_2.csv")
Dunn_tomato_Shannon_Effective_English_Name_2 <- dunnTest(alpha_500_PD$Shannon_Effective, alpha_500_PD$English_Name, method = "bonferroni");Dunn_tomato_Shannon_Effective_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_English_Name_2$res);CLD_Dunn_tomato_Shannon_Effective_English_Name_2 

#Taste
Dunn_tomato_Shannon_Effective_Fruit_taste=dunn_test(alpha_500_PD_Fruit_taste, Shannon_Effective ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_taste
write_csv(Dunn_tomato_Shannon_Effective_Fruit_taste, "Dunn_tomato_Shannon_Effective_Fruit_taste.csv")
Dunn_tomato_Shannon_Effective_Fruit_taste <- dunnTest(alpha_500_PD_Fruit_taste$Shannon_Effective, alpha_500_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_taste$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_taste 

#Color
Dunn_tomato_Shannon_Effective_Fruit_color=dunn_test(alpha_500_PD_Fruit_color, Shannon_Effective ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_color
write_csv(Dunn_tomato_Shannon_Effective_Fruit_color, "Dunn_tomato_Shannon_Effective_Fruit_color.csv")
Dunn_tomato_Shannon_Effective_Fruit_color <- dunnTest(alpha_500_PD_Fruit_color$Shannon_Effective, alpha_500_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_color$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_color 

#insect resistance
Dunn_tomato_Shannon_Effective_Insect_resistance=dunn_test(alpha_500_PD_Insect_resistance, Shannon_Effective ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Insect_resistance
write_csv(Dunn_tomato_Shannon_Effective_Insect_resistance, "Dunn_tomato_Shannon_Effective_Insect_resistance.csv")
Dunn_tomato_Shannon_Effective_Insect_resistance <- dunnTest(alpha_500_PD_Insect_resistance$Shannon_Effective, alpha_500_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Shannon_Effective_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Insect_resistance$res);CLD_Dunn_tomato_Shannon_Effective_Insect_resistance 

#TMV RESISTANCE
Dunn_tomato_Shannon_Effective_TMV_resistance=dunn_test(alpha_500_PD_TMV_resistance, Shannon_Effective ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_TMV_resistance
write_csv(Dunn_tomato_Shannon_Effective_TMV_resistance, "Dunn_tomato_Shannon_Effective_TMV_resistance.csv")
Dunn_tomato_Shannon_Effective_TMV_resistance <- dunnTest(alpha_500_PD_TMV_resistance$Shannon_Effective, alpha_500_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Shannon_Effective_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_TMV_resistance$res);CLD_Dunn_tomato_Shannon_Effective_TMV_resistance 

#seed weight
Dunn_tomato_Shannon_Effective_seedweight_1000=dunn_test(alpha_500_PD_seedweight_1000, Shannon_Effective ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_seedweight_1000
write_csv(Dunn_tomato_Shannon_Effective_seedweight_1000, "Dunn_tomato_Shannon_Effective_seedweight_1000.csv")
Dunn_tomato_Shannon_Effective_seedweight_1000 <- dunnTest(alpha_500_PD_seedweight_1000$Shannon_Effective, alpha_500_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Shannon_Effective_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_seedweight_1000$res);CLD_Dunn_tomato_Shannon_Effective_seedweight_1000 

#seed SHAPE
Dunn_tomato_Shannon_Effective_Fruit_shape=dunn_test(alpha_500_PD_Fruit_shape, Shannon_Effective ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_shape
write_csv(Dunn_tomato_Shannon_Effective_Fruit_shape, "Dunn_tomato_Shannon_Effective_Fruit_shape.csv")
Dunn_tomato_Shannon_Effective_Fruit_shape <- dunnTest(alpha_500_PD_Fruit_shape$Shannon_Effective, alpha_500_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_shape$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_shape 

#ovaries
Dunn_tomato_Shannon_Effective_ovaries=dunn_test(alpha_500_PD_ovaries, Shannon_Effective ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_ovaries
write_csv(Dunn_tomato_Shannon_Effective_ovaries, "Dunn_tomato_Shannon_Effective_ovaries.csv")
Dunn_tomato_Shannon_Effective_ovaries <- dunnTest(alpha_500_PD_ovaries$Shannon_Effective, alpha_500_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Shannon_Effective_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_ovaries$res);CLD_Dunn_tomato_Shannon_Effective_ovaries 

#yield
Dunn_tomato_Shannon_Effective_yield=dunn_test(alpha_500_PD_yield, Shannon_Effective ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_yield
write_csv(Dunn_tomato_Shannon_Effective_yield, "Dunn_tomato_Shannon_Effective_yield.csv")
Dunn_tomato_Shannon_Effective_yield <- dunnTest(alpha_500_PD_yield$Shannon_Effective, alpha_500_PD_yield$yield, method = "bonferroni");Dunn_tomato_Shannon_Effective_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_yield$res);CLD_Dunn_tomato_Shannon_Effective_yield

#Fruit_hori_diameter
Dunn_tomato_Shannon_Effective_Fruit_hori_diameter=dunn_test(alpha_500_PD_Fruit_hori_diameter, Shannon_Effective ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_hori_diameter
write_csv(Dunn_tomato_Shannon_Effective_Fruit_hori_diameter, "Dunn_tomato_Shannon_Effective_Fruit_hori_diameter.csv")
Dunn_tomato_Shannon_Effective_Fruit_hori_diameter <- dunnTest(alpha_500_PD_Fruit_hori_diameter$Shannon_Effective, alpha_500_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_hori_diameter$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_hori_diameter

#Fruit_long_diameter
Dunn_tomato_Shannon_Effective_Fruit_long_diameter=dunn_test(alpha_500_PD_Fruit_long_diameter, Shannon_Effective ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Shannon_Effective_Fruit_long_diameter
write_csv(Dunn_tomato_Shannon_Effective_Fruit_long_diameter, "Dunn_tomato_Shannon_Effective_Fruit_long_diameter.csv")
Dunn_tomato_Shannon_Effective_Fruit_long_diameter <- dunnTest(alpha_500_PD_Fruit_long_diameter$Shannon_Effective, alpha_500_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Shannon_Effective_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Shannon_Effective_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Shannon_Effective_Fruit_long_diameter$res);CLD_Dunn_tomato_Shannon_Effective_Fruit_long_diameter
```
 
 
#PHYLOGENETIC DIVERSITY
```{r,echo=FALSE,warning=FALSE}
setwd("wd")

kruskal.test(PD ~ English_Name_2, data = alpha_500_PD)

kruskal.test(PD ~ Production_site_2, data = alpha_500_PD)

kruskal.test(PD ~ Fruit_taste, data = alpha_500_PD_Fruit_taste)

kruskal.test(PD ~ Fruit_color, data = alpha_500_PD_Fruit_color)

kruskal.test(PD ~ Fruit_shape, data = alpha_500_PD_Fruit_shape)

kruskal.test(PD ~ TMV_resistance, data = alpha_500_PD_TMV_resistance)

kruskal.test(PD ~ Insect_resistance, data = alpha_500_PD_Insect_resistance)

kruskal.test(PD ~ seedweight_1000, data = alpha_500_PD_seedweight_1000)

kruskal.test(PD ~ ovaries, data = alpha_500_PD_ovaries)

kruskal.test(PD ~ yield, data = alpha_500_PD_yield)

kruskal.test(PD ~ Fruit_hori_diameter, data = alpha_500_PD_Fruit_hori_diameter)

kruskal.test(PD ~ Fruit_long_diameter, data = alpha_500_PD_Fruit_long_diameter)

# Post-hoc Dunn's Test 
#Region of tomato production

Dunn_tomato_PD_Production_site_2=dunn_test(alpha_500_PD, PD ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Production_site_2
write_csv(Dunn_tomato_PD_Production_site_2, "Dunn_tomato_PD_Production_site_2.csv")
Dunn_tomato_PD_Production_site_2 <- dunnTest(alpha_500_PD$PD, alpha_500_PD$Production_site_2, method = "bonferroni");Dunn_tomato_PD_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Production_site_2$res);CLD_Dunn_tomato_PD_Production_site_2 

#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_PD_English_Name_2=dunn_test(alpha_500_PD, PD ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_English_Name_2
write_csv(Dunn_tomato_PD_English_Name_2, "Dunn_tomato_PD_English_Name_2.csv")
Dunn_tomato_PD_English_Name_2 <- dunnTest(alpha_500_PD$PD, alpha_500_PD$English_Name, method = "bonferroni");Dunn_tomato_PD_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_English_Name_2$res);CLD_Dunn_tomato_PD_English_Name_2 

#Taste
Dunn_tomato_PD_Fruit_taste=dunn_test(alpha_500_PD_Fruit_taste, PD ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_taste
write_csv(Dunn_tomato_PD_Fruit_taste, "Dunn_tomato_PD_Fruit_taste.csv")
Dunn_tomato_PD_Fruit_taste <- dunnTest(alpha_500_PD_Fruit_taste$PD, alpha_500_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_PD_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_taste$res);CLD_Dunn_tomato_PD_Fruit_taste 

#color
Dunn_tomato_PD_Fruit_color=dunn_test(alpha_500_PD_Fruit_color, PD ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_color
write_csv(Dunn_tomato_PD_Fruit_color, "Dunn_tomato_PD_Fruit_color.csv")
Dunn_tomato_PD_Fruit_color <- dunnTest(alpha_500_PD_Fruit_color$PD, alpha_500_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_PD_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_color$res);CLD_Dunn_tomato_PD_Fruit_color 

#insect resistance
Dunn_tomato_PD_Insect_resistance=dunn_test(alpha_500_PD_Insect_resistance, PD ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Insect_resistance
write_csv(Dunn_tomato_PD_Insect_resistance, "Dunn_tomato_PD_Insect_resistance.csv")
Dunn_tomato_PD_Insect_resistance <- dunnTest(alpha_500_PD_Insect_resistance$PD, alpha_500_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_PD_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Insect_resistance$res);CLD_Dunn_tomato_PD_Insect_resistance 

#TMV RESISTANCE
Dunn_tomato_PD_TMV_resistance=dunn_test(alpha_500_PD_TMV_resistance, PD ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_TMV_resistance
write_csv(Dunn_tomato_PD_TMV_resistance, "Dunn_tomato_PD_TMV_resistance.csv")
Dunn_tomato_PD_TMV_resistance <- dunnTest(alpha_500_PD_TMV_resistance$PD, alpha_500_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_PD_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_TMV_resistance$res);CLD_Dunn_tomato_PD_TMV_resistance 

#seed weight
Dunn_tomato_PD_seedweight_1000=dunn_test(alpha_500_PD_seedweight_1000, PD ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_seedweight_1000
write_csv(Dunn_tomato_PD_seedweight_1000, "Dunn_tomato_PD_seedweight_1000.csv")
Dunn_tomato_PD_seedweight_1000 <- dunnTest(alpha_500_PD_seedweight_1000$PD, alpha_500_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_PD_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_seedweight_1000$res);CLD_Dunn_tomato_PD_seedweight_1000 

#seed SHAPE
Dunn_tomato_PD_Fruit_shape=dunn_test(alpha_500_PD_Fruit_shape, PD ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_shape
write_csv(Dunn_tomato_PD_Fruit_shape, "Dunn_tomato_PD_Fruit_shape.csv")
Dunn_tomato_PD_Fruit_shape <- dunnTest(alpha_500_PD_Fruit_shape$PD, alpha_500_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_PD_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_shape$res);CLD_Dunn_tomato_PD_Fruit_shape 

#ovaries
Dunn_tomato_PD_ovaries=dunn_test(alpha_500_PD_ovaries, PD ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_ovaries
write_csv(Dunn_tomato_PD_ovaries, "Dunn_tomato_PD_ovaries.csv")
Dunn_tomato_PD_ovaries <- dunnTest(alpha_500_PD_ovaries$PD, alpha_500_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_PD_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_ovaries$res);CLD_Dunn_tomato_PD_ovaries 

#yield
Dunn_tomato_PD_yield=dunn_test(alpha_500_PD_yield, PD ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_yield
write_csv(Dunn_tomato_PD_yield, "Dunn_tomato_PD_yield.csv")
Dunn_tomato_PD_yield <- dunnTest(alpha_500_PD_yield$PD, alpha_500_PD_yield$yield, method = "bonferroni");Dunn_tomato_PD_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_yield$res);CLD_Dunn_tomato_PD_yield 

#Fruit_hori_diameter
Dunn_tomato_PD_Fruit_hori_diameter=dunn_test(alpha_500_PD_Fruit_hori_diameter, PD ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_hori_diameter
write_csv(Dunn_tomato_PD_Fruit_hori_diameter, "Dunn_tomato_PD_Fruit_hori_diameter.csv")
Dunn_tomato_PD_Fruit_hori_diameter <- dunnTest(alpha_500_PD_Fruit_hori_diameter$PD, alpha_500_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_PD_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_hori_diameter$res);CLD_Dunn_tomato_PD_Fruit_hori_diameter 

#Fruit_long_diameter
Dunn_tomato_PD_Fruit_long_diameter=dunn_test(alpha_500_PD_Fruit_long_diameter, PD ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_PD_Fruit_long_diameter
write_csv(Dunn_tomato_PD_Fruit_long_diameter, "Dunn_tomato_PD_Fruit_long_diameter.csv")
Dunn_tomato_PD_Fruit_long_diameter <- dunnTest(alpha_500_PD_Fruit_long_diameter$PD, alpha_500_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_PD_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_PD_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_PD_Fruit_long_diameter$res);CLD_Dunn_tomato_PD_Fruit_long_diameter 
###########################################################################################################################
```


#Simpson_Index
```{r,echo=FALSE,warning=F}
setwd("wd")


kruskal.test(Simpson_Index ~ English_Name_2, data = alpha_500_PD)

kruskal.test(Simpson_Index ~ Production_site_2, data = alpha_500_PD)


kruskal.test(Simpson_Index ~ Fruit_taste, data = alpha_500_PD_Fruit_taste)

kruskal.test(Simpson_Index ~ Fruit_color, data = alpha_500_PD_Fruit_color)

kruskal.test(Simpson_Index ~ Fruit_shape, data = alpha_500_PD_Fruit_shape)

kruskal.test(Simpson_Index ~ TMV_resistance, data = alpha_500_PD_TMV_resistance)

kruskal.test(Simpson_Index ~ Insect_resistance, data = alpha_500_PD_Insect_resistance)

kruskal.test(Simpson_Index ~ seedweight_1000, data = alpha_500_PD_seedweight_1000)

kruskal.test(Simpson_Index ~ ovaries, data = alpha_500_PD_ovaries)

kruskal.test(Simpson_Index ~ yield, data = alpha_500_PD_yield)

kruskal.test(Simpson_Index ~ Fruit_hori_diameter, data = alpha_500_PD_Fruit_hori_diameter)

kruskal.test(Simpson_Index ~ Fruit_long_diameter, data = alpha_500_PD_Fruit_long_diameter)

#Region of tomato production
# Dunn's Test 
Dunn_tomato_Simpson_Index_Production_site_2=dunn_test(alpha_500_PD, Simpson_Index ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Production_site_2
write_csv(Dunn_tomato_Simpson_Index_Production_site_2, "Dunn_tomato_Simpson_Index_Production_site_2.csv")
Dunn_tomato_Simpson_Index_Production_site_2 <- dunnTest(alpha_500_PD$Simpson_Index, alpha_500_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Simpson_Index_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Production_site_2$res);CLD_Dunn_tomato_Simpson_Index_Production_site_2 

#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_Simpson_Index_English_Name_2=dunn_test(alpha_500_PD, Simpson_Index ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_English_Name_2
write_csv(Dunn_tomato_Simpson_Index_English_Name_2, "Dunn_tomato_Simpson_Index_English_Name_2.csv")
Dunn_tomato_Simpson_Index_English_Name_2 <- dunnTest(alpha_500_PD$Simpson_Index, alpha_500_PD$English_Name, method = "bonferroni");Dunn_tomato_Simpson_Index_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_English_Name_2$res);CLD_Dunn_tomato_Simpson_Index_English_Name_2 

#Taste
Dunn_tomato_Simpson_Index_Fruit_taste=dunn_test(alpha_500_PD_Fruit_taste, Simpson_Index ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_taste
write_csv(Dunn_tomato_Simpson_Index_Fruit_taste, "Dunn_tomato_Simpson_Index_Fruit_taste.csv")
Dunn_tomato_Simpson_Index_Fruit_taste <- dunnTest(alpha_500_PD_Fruit_taste$Simpson_Index, alpha_500_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_taste$res);CLD_Dunn_tomato_Simpson_Index_Fruit_taste 

Dunn_tomato_Simpson_Index_Fruit_color=dunn_test(alpha_500_PD_Fruit_color, Simpson_Index ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_color
write_csv(Dunn_tomato_Simpson_Index_Fruit_color, "Dunn_tomato_Simpson_Index_Fruit_color.csv")
Dunn_tomato_Simpson_Index_Fruit_color <- dunnTest(alpha_500_PD_Fruit_color$Simpson_Index, alpha_500_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_color
# letters of significance 
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_color$res);CLD_Dunn_tomato_Simpson_Index_Fruit_color 

#insect resistance
Dunn_tomato_Simpson_Index_Insect_resistance=dunn_test(alpha_500_PD_Insect_resistance, Simpson_Index ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Insect_resistance
write_csv(Dunn_tomato_Simpson_Index_Insect_resistance, "Dunn_tomato_Simpson_Index_Insect_resistance.csv")
Dunn_tomato_Simpson_Index_Insect_resistance <- dunnTest(alpha_500_PD_Insect_resistance$Simpson_Index, alpha_500_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Simpson_Index_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Insect_resistance$res);CLD_Dunn_tomato_Simpson_Index_Insect_resistance 

#TMV RESISTANCE
Dunn_tomato_Simpson_Index_TMV_resistance=dunn_test(alpha_500_PD_TMV_resistance, Simpson_Index ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_TMV_resistance
write_csv(Dunn_tomato_Simpson_Index_TMV_resistance, "Dunn_tomato_Simpson_Index_TMV_resistance.csv")
Dunn_tomato_Simpson_Index_TMV_resistance <- dunnTest(alpha_500_PD_TMV_resistance$Simpson_Index, alpha_500_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Simpson_Index_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_TMV_resistance$res);CLD_Dunn_tomato_Simpson_Index_TMV_resistance 

#seed weight
Dunn_tomato_Simpson_Index_seedweight_1000=dunn_test(alpha_500_PD_seedweight_1000, Simpson_Index ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_seedweight_1000
write_csv(Dunn_tomato_Simpson_Index_seedweight_1000, "Dunn_tomato_Simpson_Index_seedweight_1000.csv")
Dunn_tomato_Simpson_Index_seedweight_1000 <- dunnTest(alpha_500_PD_seedweight_1000$Simpson_Index, alpha_500_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Simpson_Index_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_seedweight_1000$res);CLD_Dunn_tomato_Simpson_Index_seedweight_1000 

#seed SHAPE
Dunn_tomato_Simpson_Index_Fruit_shape=dunn_test(alpha_500_PD_Fruit_shape, Simpson_Index ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_shape
write_csv(Dunn_tomato_Simpson_Index_Fruit_shape, "Dunn_tomato_Simpson_Index_Fruit_shape.csv")
Dunn_tomato_Simpson_Index_Fruit_shape <- dunnTest(alpha_500_PD_Fruit_shape$Simpson_Index, alpha_500_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_shape$res);CLD_Dunn_tomato_Simpson_Index_Fruit_shape 

#ovaries
Dunn_tomato_Simpson_Index_ovaries=dunn_test(alpha_500_PD_ovaries, Simpson_Index ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_ovaries
write_csv(Dunn_tomato_Simpson_Index_ovaries, "Dunn_tomato_Simpson_Index_ovaries.csv")
Dunn_tomato_Simpson_Index_ovaries <- dunnTest(alpha_500_PD_ovaries$Simpson_Index, alpha_500_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Simpson_Index_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_ovaries$res);CLD_Dunn_tomato_Simpson_Index_ovaries 

#yield
Dunn_tomato_Simpson_Index_yield=dunn_test(alpha_500_PD_yield, Simpson_Index ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_yield
write_csv(Dunn_tomato_Simpson_Index_yield, "Dunn_tomato_Simpson_Index_yield.csv")
Dunn_tomato_Simpson_Index_yield <- dunnTest(alpha_500_PD_yield$Simpson_Index, alpha_500_PD_yield$yield, method = "bonferroni");Dunn_tomato_Simpson_Index_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_yield$res);CLD_Dunn_tomato_Simpson_Index_yield 

#Fruit_hori_diameter
Dunn_tomato_Simpson_Index_Fruit_hori_diameter=dunn_test(alpha_500_PD_Fruit_hori_diameter, Simpson_Index ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_hori_diameter
write_csv(Dunn_tomato_Simpson_Index_Fruit_hori_diameter, "Dunn_tomato_Simpson_Index_Fruit_hori_diameter.csv")
Dunn_tomato_Simpson_Index_Fruit_hori_diameter <- dunnTest(alpha_500_PD_Fruit_hori_diameter$Simpson_Index, alpha_500_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_hori_diameter$res);CLD_Dunn_tomato_Simpson_Index_Fruit_hori_diameter 

#Fruit_long_diameter
Dunn_tomato_Simpson_Index_Fruit_long_diameter=dunn_test(alpha_500_PD_Fruit_long_diameter, Simpson_Index ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Index_Fruit_long_diameter
write_csv(Dunn_tomato_Simpson_Index_Fruit_long_diameter, "Dunn_tomato_Simpson_Index_Fruit_long_diameter.csv")
Dunn_tomato_Simpson_Index_Fruit_long_diameter <- dunnTest(alpha_500_PD_Fruit_long_diameter$Simpson_Index, alpha_500_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Simpson_Index_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Index_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Index_Fruit_long_diameter$res);CLD_Dunn_tomato_Simpson_Index_Fruit_long_diameter 
```

#Effective simpson index
```{r,echo=FALSE,warning=FALSE}
########################################################################################################
setwd("wd")

kruskal.test(Simpson_Effective ~English_Name_2, data = alpha_500_PD)

kruskal.test(Simpson_Effective ~ Production_site_2, data = alpha_500_PD)

kruskal.test(Simpson_Effective ~ Fruit_taste, data = alpha_500_PD_Fruit_taste)

kruskal.test(Simpson_Effective ~ Fruit_color, data = alpha_500_PD_Fruit_color)

kruskal.test(Simpson_Effective ~ Fruit_shape, data = alpha_500_PD_Fruit_shape)

kruskal.test(Simpson_Effective ~ TMV_resistance, data = alpha_500_PD_TMV_resistance)

kruskal.test(Simpson_Effective ~ Insect_resistance, data = alpha_500_PD_Insect_resistance)

kruskal.test(Simpson_Effective ~ seedweight_1000, data = alpha_500_PD_seedweight_1000)

kruskal.test(Simpson_Effective ~ ovaries, data = alpha_500_PD_ovaries)

kruskal.test(Simpson_Effective ~ yield, data = alpha_500_PD_yield)

kruskal.test(Simpson_Effective ~ Fruit_hori_diameter, data = alpha_500_PD_Fruit_hori_diameter)

kruskal.test(Simpson_Effective ~ Fruit_long_diameter, data = alpha_500_PD_Fruit_long_diameter)

#Region of production
# Dunn's Test 
Dunn_tomato_Simpson_Effective_Production_site_2=dunn_test(alpha_500_PD, Simpson_Effective ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Production_site_2
write_csv(Dunn_tomato_Simpson_Effective_Production_site_2, "Dunn_tomato_Simpson_Effective_Production_site_2.csv")
Dunn_tomato_Simpson_Effective_Production_site_2 <- dunnTest(alpha_500_PD$Simpson_Effective, alpha_500_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Simpson_Effective_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Production_site_2$res);CLD_Dunn_tomato_Simpson_Effective_Production_site_2 

#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_Simpson_Effective_English_Name_2=dunn_test(alpha_500_PD, Simpson_Effective ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_English_Name_2
write_csv(Dunn_tomato_Simpson_Effective_English_Name_2, "Dunn_tomato_Simpson_Effective_English_Name_2.csv")
Dunn_tomato_Simpson_Effective_English_Name_2 <- dunnTest(alpha_500_PD$Simpson_Effective, alpha_500_PD$English_Name, method = "bonferroni");Dunn_tomato_Simpson_Effective_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_English_Name_2$res);CLD_Dunn_tomato_Simpson_Effective_English_Name_2 

#Taste
Dunn_tomato_Simpson_Effective_Fruit_taste=dunn_test(alpha_500_PD_Fruit_taste, Simpson_Effective ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_taste
write_csv(Dunn_tomato_Simpson_Effective_Fruit_taste, "Dunn_tomato_Simpson_Effective_Fruit_taste.csv")
Dunn_tomato_Simpson_Effective_Fruit_taste <- dunnTest(alpha_500_PD_Fruit_taste$Simpson_Effective, alpha_500_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_taste$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_taste 

#Color
Dunn_tomato_Simpson_Effective_Fruit_color=dunn_test(alpha_500_PD_Fruit_color, Simpson_Effective ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_color
write_csv(Dunn_tomato_Simpson_Effective_Fruit_color, "Dunn_tomato_Simpson_Effective_Fruit_color.csv")
Dunn_tomato_Simpson_Effective_Fruit_color <- dunnTest(alpha_500_PD_Fruit_color$Simpson_Effective, alpha_500_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_color$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_color 

#insect resistance
Dunn_tomato_Simpson_Effective_Insect_resistance=dunn_test(alpha_500_PD_Insect_resistance, Simpson_Effective ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Insect_resistance
write_csv(Dunn_tomato_Simpson_Effective_Insect_resistance, "Dunn_tomato_Simpson_Effective_Insect_resistance.csv")
Dunn_tomato_Simpson_Effective_Insect_resistance <- dunnTest(alpha_500_PD_Insect_resistance$Simpson_Effective, alpha_500_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Simpson_Effective_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Insect_resistance$res);CLD_Dunn_tomato_Simpson_Effective_Insect_resistance 

#TMV RESISTANCE
Dunn_tomato_Simpson_Effective_TMV_resistance=dunn_test(alpha_500_PD_TMV_resistance, Simpson_Effective ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_TMV_resistance
write_csv(Dunn_tomato_Simpson_Effective_TMV_resistance, "Dunn_tomato_Simpson_Effective_TMV_resistance.csv")
Dunn_tomato_Simpson_Effective_TMV_resistance <- dunnTest(alpha_500_PD_TMV_resistance$Simpson_Effective, alpha_500_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Simpson_Effective_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_TMV_resistance$res);CLD_Dunn_tomato_Simpson_Effective_TMV_resistance 

#seed weight
Dunn_tomato_Simpson_Effective_seedweight_1000=dunn_test(alpha_500_PD_seedweight_1000, Simpson_Effective ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_seedweight_1000
write_csv(Dunn_tomato_Simpson_Effective_seedweight_1000, "Dunn_tomato_Simpson_Effective_seedweight_1000.csv")
Dunn_tomato_Simpson_Effective_seedweight_1000 <- dunnTest(alpha_500_PD_seedweight_1000$Simpson_Effective, alpha_500_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Simpson_Effective_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_seedweight_1000$res);CLD_Dunn_tomato_Simpson_Effective_seedweight_1000 

#seed SHAPE
Dunn_tomato_Simpson_Effective_Fruit_shape=dunn_test(alpha_500_PD_Fruit_shape, Simpson_Effective ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_shape
write_csv(Dunn_tomato_Simpson_Effective_Fruit_shape, "Dunn_tomato_Simpson_Effective_Fruit_shape.csv")
Dunn_tomato_Simpson_Effective_Fruit_shape <- dunnTest(alpha_500_PD_Fruit_shape$Simpson_Effective, alpha_500_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_shape$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_shape

#ovaries
Dunn_tomato_Simpson_Effective_ovaries=dunn_test(alpha_500_PD_ovaries, Simpson_Effective ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_ovaries
write_csv(Dunn_tomato_Simpson_Effective_ovaries, "Dunn_tomato_Simpson_Effective_ovaries.csv")
Dunn_tomato_Simpson_Effective_ovaries <- dunnTest(alpha_500_PD_ovaries$Simpson_Effective, alpha_500_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Simpson_Effective_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_ovaries$res);CLD_Dunn_tomato_Simpson_Effective_ovaries

#yield
Dunn_tomato_Simpson_Effective_yield=dunn_test(alpha_500_PD_yield, Simpson_Effective ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_yield
write_csv(Dunn_tomato_Simpson_Effective_yield, "Dunn_tomato_Simpson_Effective_yield.csv")
Dunn_tomato_Simpson_Effective_yield <- dunnTest(alpha_500_PD_yield$Simpson_Effective, alpha_500_PD_yield$yield, method = "bonferroni");Dunn_tomato_Simpson_Effective_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_yield$res);CLD_Dunn_tomato_Simpson_Effective_yield

#Fruit_hori_diameter
Dunn_tomato_Simpson_Effective_Fruit_hori_diameter=dunn_test(alpha_500_PD_Fruit_hori_diameter, Simpson_Effective ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_hori_diameter
write_csv(Dunn_tomato_Simpson_Effective_Fruit_hori_diameter, "Dunn_tomato_Simpson_Effective_Fruit_hori_diameter.csv")
Dunn_tomato_Simpson_Effective_Fruit_hori_diameter <- dunnTest(alpha_500_PD_Fruit_hori_diameter$Simpson_Effective, alpha_500_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_hori_diameter$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_hori_diameter

#Fruit_long_diameter
Dunn_tomato_Simpson_Effective_Fruit_long_diameter=dunn_test(alpha_500_PD_Fruit_long_diameter, Simpson_Effective ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Simpson_Effective_Fruit_long_diameter
write_csv(Dunn_tomato_Simpson_Effective_Fruit_long_diameter, "Dunn_tomato_Simpson_Effective_Fruit_long_diameter.csv")
Dunn_tomato_Simpson_Effective_Fruit_long_diameter <- dunnTest(alpha_500_PD_Fruit_long_diameter$Simpson_Effective, alpha_500_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Simpson_Effective_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Simpson_Effective_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Simpson_Effective_Fruit_long_diameter$res);CLD_Dunn_tomato_Simpson_Effective_Fruit_long_diameter
```


#EVENNESS
```{r,echo=FALSE,warning=FALSE}
setwd("wd")

kruskal.test(Evenness ~ English_Name_2, data = alpha_500_PD)

kruskal.test(Evenness ~ Production_site_2, data = alpha_500_PD)

kruskal.test(Evenness ~ Fruit_taste, data = alpha_500_PD_Fruit_taste)

kruskal.test(Evenness ~ Fruit_color, data = alpha_500_PD_Fruit_color)

kruskal.test(Evenness ~ Fruit_shape, data = alpha_500_PD_Fruit_shape)

kruskal.test(Evenness ~ TMV_resistance, data = alpha_500_PD_TMV_resistance)

kruskal.test(Evenness ~ Insect_resistance, data = alpha_500_PD_Insect_resistance)

kruskal.test(Evenness ~ seedweight_1000, data = alpha_500_PD_seedweight_1000)

kruskal.test(Evenness ~ ovaries, data = alpha_500_PD_ovaries)

kruskal.test(Evenness ~ yield, data = alpha_500_PD_yield)

kruskal.test(Evenness ~ Fruit_hori_diameter, data = alpha_500_PD_Fruit_hori_diameter)

kruskal.test(Evenness ~ Fruit_long_diameter, data = alpha_500_PD_Fruit_long_diameter)

#Post-hoc (Dunn's tests)

# Region of tomato seed production
Dunn_tomato_Evenness_Production_site_2=dunn_test(alpha_500_PD, Evenness ~ Production_site_2, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Production_site_2
write_csv(Dunn_tomato_Evenness_Production_site_2, "Dunn_tomato_Evenness_Production_site_2.csv")
Dunn_tomato_Evenness_Production_site_2 <- dunnTest(alpha_500_PD$Evenness, alpha_500_PD$Production_site_2, method = "bonferroni");Dunn_tomato_Evenness_Production_site_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Production_site_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Production_site_2$res);CLD_Dunn_tomato_Evenness_Production_site_2 

#Cultivar
#*****The use of BH is rather less stringent; however an alternative to bonferroni is holm
Dunn_tomato_Evenness_English_Name_2=dunn_test(alpha_500_PD, Evenness ~ English_Name, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_English_Name_2
write_csv(Dunn_tomato_Evenness_English_Name_2, "Dunn_tomato_Evenness_English_Name_2.csv")
Dunn_tomato_Evenness_English_Name_2 <- dunnTest(alpha_500_PD$Evenness, alpha_500_PD$English_Name, method = "bonferroni");Dunn_tomato_Evenness_English_Name_2
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_English_Name_2 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_English_Name_2$res);CLD_Dunn_tomato_Evenness_English_Name_2 

#Taste
Dunn_tomato_Evenness_Fruit_taste=dunn_test(alpha_500_PD_Fruit_taste, Evenness ~ Fruit_taste, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_taste
write_csv(Dunn_tomato_Evenness_Fruit_taste, "Dunn_tomato_Evenness_Fruit_taste.csv")
Dunn_tomato_Evenness_Fruit_taste <- dunnTest(alpha_500_PD_Fruit_taste$Evenness, alpha_500_PD_Fruit_taste$Fruit_taste, method = "bonferroni");Dunn_tomato_Evenness_Fruit_taste
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_taste = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_taste$res);CLD_Dunn_tomato_Evenness_Fruit_taste 

#color
Dunn_tomato_Evenness_Fruit_color=dunn_test(alpha_500_PD_Fruit_color, Evenness ~ Fruit_color, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_color
write_csv(Dunn_tomato_Evenness_Fruit_color, "Dunn_tomato_Evenness_Fruit_color.csv")
Dunn_tomato_Evenness_Fruit_color <- dunnTest(alpha_500_PD_Fruit_color$Evenness, alpha_500_PD_Fruit_color$Fruit_color, method = "bonferroni");Dunn_tomato_Evenness_Fruit_color
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_color = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_color$res);CLD_Dunn_tomato_Evenness_Fruit_color 

#insect resistance
Dunn_tomato_Evenness_Insect_resistance=dunn_test(alpha_500_PD_Insect_resistance, Evenness ~ Insect_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Insect_resistance
write_csv(Dunn_tomato_Evenness_Insect_resistance, "Dunn_tomato_Evenness_Insect_resistance.csv")
Dunn_tomato_Evenness_Insect_resistance <- dunnTest(alpha_500_PD_Insect_resistance$Evenness, alpha_500_PD_Insect_resistance$Insect_resistance, method = "bonferroni");Dunn_tomato_Evenness_Insect_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Insect_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Insect_resistance$res);CLD_Dunn_tomato_Evenness_Insect_resistance 

#TMV RESISTANCE
Dunn_tomato_Evenness_TMV_resistance=dunn_test(alpha_500_PD_TMV_resistance, Evenness ~ TMV_resistance, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_TMV_resistance
write_csv(Dunn_tomato_Evenness_TMV_resistance, "Dunn_tomato_Evenness_TMV_resistance.csv")
Dunn_tomato_Evenness_TMV_resistance <- dunnTest(alpha_500_PD_TMV_resistance$Evenness, alpha_500_PD_TMV_resistance$TMV_resistance, method = "bonferroni");Dunn_tomato_Evenness_TMV_resistance
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_TMV_resistance = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_TMV_resistance$res);CLD_Dunn_tomato_Evenness_TMV_resistance 

#seed weight
Dunn_tomato_Evenness_seedweight_1000=dunn_test(alpha_500_PD_seedweight_1000, Evenness ~ seedweight_1000, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_seedweight_1000
write_csv(Dunn_tomato_Evenness_seedweight_1000, "Dunn_tomato_Evenness_seedweight_1000.csv")
Dunn_tomato_Evenness_seedweight_1000 <- dunnTest(alpha_500_PD_seedweight_1000$Evenness, alpha_500_PD_seedweight_1000$seedweight_1000, method = "bonferroni");Dunn_tomato_Evenness_seedweight_1000
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_seedweight_1000 = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_seedweight_1000$res);CLD_Dunn_tomato_Evenness_seedweight_1000 

#seed SHAPE
Dunn_tomato_Evenness_Fruit_shape=dunn_test(alpha_500_PD_Fruit_shape, Evenness ~ Fruit_shape, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_shape
write_csv(Dunn_tomato_Evenness_Fruit_shape, "Dunn_tomato_Evenness_Fruit_shape.csv")
Dunn_tomato_Evenness_Fruit_shape <- dunnTest(alpha_500_PD_Fruit_shape$Evenness, alpha_500_PD_Fruit_shape$Fruit_shape, method = "bonferroni");Dunn_tomato_Evenness_Fruit_shape
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_shape = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_shape$res);CLD_Dunn_tomato_Evenness_Fruit_shape 

#ovaries
Dunn_tomato_Evenness_ovaries=dunn_test(alpha_500_PD_ovaries, Evenness ~ ovaries, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_ovaries
write_csv(Dunn_tomato_Evenness_ovaries, "Dunn_tomato_Evenness_ovaries.csv")
Dunn_tomato_Evenness_ovaries <- dunnTest(alpha_500_PD_ovaries$Evenness, alpha_500_PD_ovaries$ovaries, method = "bonferroni");Dunn_tomato_Evenness_ovaries
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_ovaries = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_ovaries$res);CLD_Dunn_tomato_Evenness_ovaries 

#yield
Dunn_tomato_Evenness_yield=dunn_test(alpha_500_PD_yield, Evenness ~ yield, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_yield
write_csv(Dunn_tomato_Evenness_yield, "Dunn_tomato_Evenness_yield.csv")
Dunn_tomato_Evenness_yield <- dunnTest(alpha_500_PD_yield$Evenness, alpha_500_PD_yield$yield, method = "bonferroni");Dunn_tomato_Evenness_yield
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_yield = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_yield$res);CLD_Dunn_tomato_Evenness_yield 

#Fruit_hori_diameter
Dunn_tomato_Evenness_Fruit_hori_diameter=dunn_test(alpha_500_PD_Fruit_hori_diameter, Evenness ~ Fruit_hori_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_hori_diameter
write_csv(Dunn_tomato_Evenness_Fruit_hori_diameter, "Dunn_tomato_Evenness_Fruit_hori_diameter.csv")
Dunn_tomato_Evenness_Fruit_hori_diameter <- dunnTest(alpha_500_PD_Fruit_hori_diameter$Evenness, alpha_500_PD_Fruit_hori_diameter$Fruit_hori_diameter, method = "bonferroni");Dunn_tomato_Evenness_Fruit_hori_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_hori_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_hori_diameter$res);CLD_Dunn_tomato_Evenness_Fruit_hori_diameter 

#Fruit_long_diameter
Dunn_tomato_Evenness_Fruit_long_diameter=dunn_test(alpha_500_PD_Fruit_long_diameter, Evenness ~ Fruit_long_diameter, p.adjust.method = "bonferroni", detailed = FALSE);Dunn_tomato_Evenness_Fruit_long_diameter
write_csv(Dunn_tomato_Evenness_Fruit_long_diameter, "Dunn_tomato_Evenness_Fruit_long_diameter.csv")
Dunn_tomato_Evenness_Fruit_long_diameter <- dunnTest(alpha_500_PD_Fruit_long_diameter$Evenness, alpha_500_PD_Fruit_long_diameter$Fruit_long_diameter, method = "bonferroni");Dunn_tomato_Evenness_Fruit_long_diameter
# Obtaining significance letters
library(rcompanion)
CLD_Dunn_tomato_Evenness_Fruit_long_diameter = cldList(P.adj ~ Comparison, data=Dunn_tomato_Evenness_Fruit_long_diameter$res);CLD_Dunn_tomato_Evenness_Fruit_long_diameter 
```



#Alpha diversity  graphics  
```{r,echo=FALSE,warning=FALSE}
setwd("wd")
#1:richness_shannon_pd_Simpson_eveness.tiff
plot_richness=ggboxplot(alpha_500_PD, x = "English_Name_2", y = "Richness", fill= "English_Name_2",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=18))+
  theme(text=element_text(size=18,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness


#SHANON DIVERSITY
plot_Shannon=ggboxplot(alpha_500_PD, x = "English_Name_2", y = "Shannon_Index", fill= "English_Name_2", ylab = "Shannon Index", xlab = "Tomato cultivars")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_text( colour="gray33", size=18),
        axis.text.y  = element_text( colour="gray33", size=18))+
  theme(text=element_text(size=18,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon
#ggsave("plot_Shannon.TIFF", plot = plot_Shannon)


#Phylogenetic diversity
plot_PD=ggboxplot(alpha_500_PD, x = "English_Name_2", y = "PD", fill= "English_Name_2", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=18))+
  theme(text=element_text(size=18,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD
#ggsave("plot_PD.TIFF", plot = plot_PD)

#Figure 2A & B
#tiff("richness_shannon_pd_Simpson_eveness.tiff", units="in", width=20, height=18, res=300)
tiff("richness_shannon_pd_Simpson_eveness.tiff", units="in", width=20, height=10, res=300)
richness_shannon_pd_Simpson_eveness.tiff=(plot_richness/plot_Shannon);richness_shannon_pd_Simpson_eveness.tiff
ggsave("richness_shannon_pd_Simpson_eveness.tiff", plot = richness_shannon_pd_Simpson_eveness.tiff)
dev.off()

 
plot_Simpson=ggboxplot(alpha_500_PD, x = "English_Name_2", y = "Simpson_Index", fill= "English_Name_2", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson
#ggsave("plot_Simpson.TIFF", plot = plot_Simpson)

#EVENNESS
 
plot_evenness=ggboxplot(alpha_500_PD, x = "English_Name_2", y = "Evenness", fill= "English_Name_2", ylab = "Evenness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=18),
        axis.title.y = element_text(face="bold", colour="gray33", size=18),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness

#Effective_alpha_diversity
tiff("Effective_alpha_diversity.tiff", units="in", width=20, height=15, res=300)
plot_effective_richness=ggboxplot(alpha_500_PD, x = "English_Name_2", y = "Effective_Richness", fill= "English_Name_2",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=16),
        axis.title.y = element_text(face="bold", colour="gray33", size=16),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness


plot_effective_Shannon=ggboxplot(alpha_500_PD, x = "English_Name_2", y = "Shannon_Effective", fill= "English_Name_2",  ylab = "Eff. Shannon div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('gray', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=16),
        axis.title.y = element_text(face="bold", colour="gray33", size=16),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon


plot_effective_Simpson=ggboxplot(alpha_500_PD, x = "English_Name_2", y = "Simpson_Effective", fill= "English_Name_2",  ylab = "Eff. Simpson div.", xlab = "Cultivar (n=100)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('gray', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=16),
        axis.title.y = element_text(face="bold", colour="gray33", size=16),
        axis.text.x = element_text( colour="gray33", size=16),
        axis.text.y  = element_text( colour="gray33", size=16))+
  theme(text=element_text(size=16,  family="sans"))+ labs(tag = "")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson

----
#Fig.S21
----
tiff("pd_Simpson_eveness_Effective_alpha_diversity.tiff", units="in", width=19, height=19, res=300)
pd_Simpson_eveness_Effective_alpha_diversity.tiff=(p2/plot_PD/plot_Simpson/plot_evenness/plot_effective_richness/plot_effective_Shannon/plot_effective_Simpson);pd_Simpson_eveness_Effective_alpha_diversity.tiff
ggsave("pd_Simpson_eveness_Effective_alpha_diversity.tiff", plot = pd_Simpson_eveness_Effective_alpha_diversity.tiff)
dev.off()

###########################################################################################################################
#Production site
plot_richness_prod_site=ggboxplot(alpha_500_PD, x = "Production_site_2", y = "Richness", fill= "Production_site_2",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_prod_site

#Effective_alpha_diversity
plot_effective_richness_prod_site=ggboxplot(alpha_500_PD, x = "Production_site_2", y = "Effective_Richness", fill= "Production_site_2",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_prod_site

#SHANON DIVERSITY
plot_Shannon_prod_site=ggboxplot(alpha_500_PD, x = "Production_site_2", y = "Shannon_Index", fill= "Production_site_2", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_prod_site


plot_effective_Shannon_prod_site=ggboxplot(alpha_500_PD, x = "Production_site_2", y = "Shannon_Effective", fill= "Production_site_2",  ylab = "Eff. Shannon div.", xlab = "Region of seed of production")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('gray', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_text( colour="gray33", size=17),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_prod_site

#Phylogenetic diversity
plot_PD_prod_site=ggboxplot(alpha_500_PD, x = "Production_site_2", y = "PD", fill= "Production_site_2", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_prod_site


plot_Simpson_prod_site=ggboxplot(alpha_500_PD, x = "Production_site_2", y = "Simpson_Index", fill= "Production_site_2", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +
  #scale_color_gradientn(colours = c('grey', 'pink'))+
 scale_color_viridis_b() +
   theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_prod_site


plot_effective_Simpson_prod_site=ggboxplot(alpha_500_PD, x = "Production_site_2", y = "Simpson_Effective", fill= "Production_site_2",  ylab = "Effe. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('gray', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_prod_site


#EVENNESS
plot_evenness_prod_site=ggboxplot(alpha_500_PD, x = "Production_site_2", y = "Evenness", fill= "Production_site_2", ylab = "Evenness", xlab = "Region of seed of production")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_color_gradientn(colours = c('grey', 'pink'))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=17),
        axis.title.y = element_text(face="bold", colour="gray33", size=17),
        axis.text.x = element_text( colour="gray33", size=17),
        axis.text.y  = element_text( colour="gray33", size=17))+
  theme(text=element_text(size=17,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_prod_site
#ggsave("plot_evenness.TIFF", plot = plot_evenness)

----
#Fig.S2
----
tiff("richness_shannon_pd_Simpson_eveness_production_site.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_priduction_site.tiff=(plot_richness_prod_site+plot_effective_richness_prod_site+plot_Shannon_prod_site+plot_effective_Shannon_prod_site+plot_PD_prod_site+plot_Simpson_prod_site+plot_effective_Simpson_prod_site+plot_evenness_prod_site+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_priduction_site.tiff
ggsave("richness_shannon_pd_Simpson_eveness_priduction_site.tiff", plot = richness_shannon_pd_Simpson_eveness_priduction_site.tiff)
dev.off()

###############################################################################################

#Dataframe
alpha_500_PD_Fruit_color=alpha_500_PD[!is.na(alpha_500_PD$Fruit_color),];alpha_500_PD_Fruit_color


myco_col=c("grey35","#00AFBB", "yellow3","chartreuse4","#FB9A99");myco_col
#FRUIT COLOR FRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLORFRUIT COLOR

#alpha_500_PD_Fruit_color
plot_richness_Fruit_color=ggboxplot(alpha_500_PD_Fruit_color, x = "Fruit_color", y = "Richness", fill= "Fruit_color",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_Fruit_color

#Effective_alpha_diversity
plot_effective_richness_Fruit_color=ggboxplot(alpha_500_PD_Fruit_color, x = "Fruit_color", y = "Effective_Richness", fill= "Fruit_color",  ylab = "Eff.Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_Fruit_color

#SHANON DIVERSITY
plot_Shannon_Fruit_color=ggboxplot(alpha_500_PD_Fruit_color, x = "Fruit_color", y = "Shannon_Index", fill= "Fruit_color", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_Fruit_color


plot_effective_Shannon_Fruit_color=ggboxplot(alpha_500_PD_Fruit_color, x = "Fruit_color", y = "Shannon_Effective", fill= "Fruit_color",  ylab = "Eff. Shannon div.", xlab = "Berry Color")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_Fruit_color

#Phylogenetic diversity
plot_PD_Fruit_color=ggboxplot(alpha_500_PD_Fruit_color, x = "Fruit_color", y = "PD", fill= "Fruit_color", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_Fruit_color

plot_Simpson_Fruit_color=ggboxplot(alpha_500_PD_Fruit_color, x = "Fruit_color", y = "Simpson_Index", fill= "Fruit_color", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_Fruit_color

plot_effective_Simpson_Fruit_color=ggboxplot(alpha_500_PD_Fruit_color, x = "Fruit_color", y = "Simpson_Effective", fill= "Fruit_color",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_Fruit_color

#EVENNESS
plot_evenness_Fruit_color=ggboxplot(alpha_500_PD_Fruit_color, x = "Fruit_color", y = "Evenness", fill= "Fruit_color", ylab = "Evenness", xlab = "Berry Color")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_col)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_Fruit_color

----
#Fig.S7
----
tiff("richness_shannon_pd_Simpson_eveness_Fruit_color.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_color.tiff=(plot_richness_Fruit_color+plot_effective_richness_Fruit_color+plot_Shannon_Fruit_color+plot_effective_Shannon_Fruit_color+plot_PD_Fruit_color+plot_Simpson_Fruit_color+plot_effective_Simpson_Fruit_color+plot_evenness_Fruit_color+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_color.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_color.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_color.tiff)
dev.off()

###############################################################################################
alpha_500_PD_Fruit_taste=alpha_500_PD[!is.na(alpha_500_PD$Fruit_taste),];alpha_500_PD_Fruit_taste
myco_taste=c("#FB9A99","chartreuse4");myco_taste
############################
plot_richness_Fruit_taste=ggboxplot(alpha_500_PD_Fruit_taste, x = "Fruit_taste", y = "Richness", fill= "Fruit_taste",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_Fruit_taste


#Effective_alpha_diversity
plot_effective_richness_Fruit_taste=ggboxplot(alpha_500_PD_Fruit_taste, x = "Fruit_taste", y = "Effective_Richness", fill= "Fruit_taste",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_Fruit_taste

#SHANON DIVERSITY
plot_Shannon_Fruit_taste=ggboxplot(alpha_500_PD_Fruit_taste, x = "Fruit_taste", y = "Shannon_Index", fill= "Fruit_taste", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_Fruit_taste


plot_effective_Shannon_Fruit_taste=ggboxplot(alpha_500_PD_Fruit_taste, x = "Fruit_taste", y = "Shannon_Effective", fill= "Fruit_taste",  ylab = "Eff. Shannon div.", xlab = "Berry taste")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_Fruit_taste

#Phylogenetic diversity
plot_PD_Fruit_taste=ggboxplot(alpha_500_PD_Fruit_taste, x = "Fruit_taste", y = "PD", fill= "Fruit_taste", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_Fruit_taste


plot_Simpson_Fruit_taste=ggboxplot(alpha_500_PD_Fruit_taste, x = "Fruit_taste", y = "Simpson_Index", fill= "Fruit_taste", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_Fruit_taste

plot_effective_Simpson_Fruit_taste=ggboxplot(alpha_500_PD_Fruit_taste, x = "Fruit_taste", y = "Simpson_Effective", fill= "Fruit_taste",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_Fruit_taste

#EVENNESS
 
plot_evenness_Fruit_taste=ggboxplot(alpha_500_PD_Fruit_taste, x = "Fruit_taste", y = "Evenness", fill= "Fruit_taste", ylab = "Evenness", xlab = "Berry taste")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_taste)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_Fruit_taste

----
#Fig.S8
----
tiff("richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff=(plot_richness_Fruit_taste+plot_effective_richness_Fruit_taste+plot_Shannon_Fruit_taste+plot_effective_Shannon_Fruit_taste+plot_PD_Fruit_taste+plot_Simpson_Fruit_taste+plot_effective_Simpson_Fruit_taste+plot_evenness_Fruit_taste+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_taste.tiff)
dev.off()

###############################################################################################
alpha_500_PD_Fruit_shape=alpha_500_PD[!is.na(alpha_500_PD$Fruit_shape),];alpha_500_PD_Fruit_shape
myco_shp=c("darkturquoise","chartreuse4","#FB9A99");myco_shp
#########################
alpha_500_PD_Fruit_shape=alpha_500_PD[!is.na(alpha_500_PD$Fruit_shape),];alpha_500_PD_Fruit_shape
tiff("richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff", units="in", width=20, height=20, res=300)
plot_richness_Fruit_shape=ggboxplot(alpha_500_PD_Fruit_shape, x = "Fruit_shape", y = "Richness", fill= "Fruit_shape",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_Fruit_shape

#Effective_alpha_diversity
plot_effective_richness_Fruit_shape=ggboxplot(alpha_500_PD_Fruit_shape, x = "Fruit_shape", y = "Effective_Richness", fill= "Fruit_shape",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_Fruit_shape

#SHANON DIVERSITY
plot_Shannon_Fruit_shape=ggboxplot(alpha_500_PD_Fruit_shape, x = "Fruit_shape", y = "Shannon_Index", fill= "Fruit_shape", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_Fruit_shape


plot_effective_Shannon_Fruit_shape=ggboxplot(alpha_500_PD_Fruit_shape, x = "Fruit_shape", y = "Shannon_Effective", fill= "Fruit_shape",  ylab = "Eff. Shannon div.", xlab = "Berry shape")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_Fruit_shape

#Phylogenetic diversity
plot_PD_Fruit_shape=ggboxplot(alpha_500_PD_Fruit_shape, x = "Fruit_shape", y = "PD", fill= "Fruit_shape", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_Fruit_shape


plot_Simpson_Fruit_shape=ggboxplot(alpha_500_PD_Fruit_shape, x = "Fruit_shape", y = "Simpson_Index", fill= "Fruit_shape", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_Fruit_shape


plot_effective_Simpson_Fruit_shape=ggboxplot(alpha_500_PD_Fruit_shape, x = "Fruit_shape", y = "Simpson_Effective", fill= "Fruit_shape",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_Fruit_shape

#EVENNESS
plot_evenness_Fruit_shape=ggboxplot(alpha_500_PD_Fruit_shape, x = "Fruit_shape", y = "Evenness", fill= "Fruit_shape", ylab = "Evenness", xlab = "Berry shape")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_shp)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_Fruit_shape

----
#Fig.S6
----
tiff("richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff=(plot_richness_Fruit_shape+plot_effective_richness_Fruit_shape+plot_Shannon_Fruit_shape+plot_effective_Shannon_Fruit_shape+plot_PD_Fruit_shape+plot_Simpson_Fruit_shape+plot_effective_Simpson_Fruit_shape+plot_evenness_Fruit_shape+ plot_layout(ncol= 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_shape.tiff)
dev.off()

###############################################################################################
alpha_500_PD_TMV_resistance=alpha_500_PD[!is.na(alpha_500_PD$TMV_resistance),];alpha_500_PD_TMV_resistance
myco_TMV=c("#0B775E","#FB9A99","grey35");myco_TMV
#############################
#############################
plot_richness_TMV_resistance=ggboxplot(alpha_500_PD_TMV_resistance, x = "TMV_resistance", y = "Richness", fill= "TMV_resistance",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_TMV_resistance

#Effective_alpha_diversity
plot_effective_richness_TMV_resistance=ggboxplot(alpha_500_PD_TMV_resistance, x = "TMV_resistance", y = "Effective_Richness", fill= "TMV_resistance",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_TMV_resistance

#SHANON DIVERSITY
plot_Shannon_TMV_resistance=ggboxplot(alpha_500_PD_TMV_resistance, x = "TMV_resistance", y = "Shannon_Index", fill= "TMV_resistance", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_TMV_resistance


plot_effective_Shannon_TMV_resistance=ggboxplot(alpha_500_PD_TMV_resistance, x = "TMV_resistance", y = "Shannon_Effective", fill= "TMV_resistance",  ylab = "Eff. Shannon div.", xlab = "TMV resistance")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_TMV_resistance

#Phylogenetic diversity
plot_PD_TMV_resistance=ggboxplot(alpha_500_PD_TMV_resistance, x = "TMV_resistance", y = "PD", fill= "TMV_resistance", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_TMV_resistance


plot_Simpson_TMV_resistance=ggboxplot(alpha_500_PD_TMV_resistance, x = "TMV_resistance", y = "Simpson_Index", fill= "TMV_resistance", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_TMV_resistance

plot_effective_Simpson_TMV_resistance=ggboxplot(alpha_500_PD_TMV_resistance, x = "TMV_resistance", y = "Simpson_Effective", fill= "TMV_resistance",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x =element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_TMV_resistance

#EVENNESS
plot_evenness_TMV_resistance=ggboxplot(alpha_500_PD_TMV_resistance, x = "TMV_resistance", y = "Evenness", fill= "TMV_resistance", ylab = "Evenness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_TMV)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_TMV_resistance

tiff("richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff=(plot_richness_TMV_resistance+plot_effective_richness_TMV_resistance+plot_Shannon_TMV_resistance+plot_effective_Shannon_TMV_resistance+plot_PD_TMV_resistance+plot_Simpson_TMV_resistance+plot_effective_Simpson_TMV_resistance+plot_evenness_TMV_resistance+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff
ggsave("richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff", plot = richness_shannon_pd_Simpson_eveness_TMV_resistance.tiff)
dev.off()
###########################################################################################################################
alpha_500_PD_Insect_resistance=alpha_500_PD[!is.na(alpha_500_PD$Insect_resistance),];alpha_500_PD_Insect_resistance
myco_insRes=c("yellow3","grey35","chartreuse4","#0B775E","#00AFBB", "#FB9A99");myco_insRes
##############################################################################################################################################################################################################################################
plot_richness_Insect_resistance=ggboxplot(alpha_500_PD_Insect_resistance, x = "Insect_resistance_2", y = "Richness", fill= "Insect_resistance_2",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_Insect_resistance


#Effective_alpha_diversity
plot_effective_richness_Insect_resistance=ggboxplot(alpha_500_PD_Insect_resistance, x = "Insect_resistance_2", y = "Effective_Richness", fill= "Insect_resistance_2",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_Insect_resistance

#SHANON DIVERSITY
plot_Shannon_Insect_resistance=ggboxplot(alpha_500_PD_Insect_resistance, x = "Insect_resistance_2", y = "Shannon_Index", fill= "Insect_resistance_2", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_Insect_resistance


plot_effective_Shannon_Insect_resistance=ggboxplot(alpha_500_PD_Insect_resistance, x = "Insect_resistance_2", y = "Shannon_Effective", fill= "Insect_resistance_2",  ylab = "Eff. Shannon div.", xlab = "Insect resistance")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_Insect_resistance

#Phylogenetic diversity
plot_PD_Insect_resistance=ggboxplot(alpha_500_PD_Insect_resistance, x = "Insect_resistance_2", y = "PD", fill= "Insect_resistance_2", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_Insect_resistance


plot_Simpson_Insect_resistance=ggboxplot(alpha_500_PD_Insect_resistance, x = "Insect_resistance_2", y = "Simpson_Index", fill= "Insect_resistance_2", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_Insect_resistance

plot_effective_Simpson_Insect_resistance=ggboxplot(alpha_500_PD_Insect_resistance, x = "Insect_resistance_2", y = "Simpson_Effective", fill= "Insect_resistance_2",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_Insect_resistance

#EVENNESS
plot_evenness_Insect_resistance=ggboxplot(alpha_500_PD_Insect_resistance, x = "Insect_resistance_2", y = "Evenness", fill= "Insect_resistance_2", ylab = "Evenness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_insRes)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_Insect_resistance

----
#Fig.S3
----
tiff("richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff=(plot_richness_Insect_resistance+plot_effective_richness_Insect_resistance+plot_Shannon_Insect_resistance+plot_effective_Shannon_Insect_resistance+plot_PD_Insect_resistance+plot_Simpson_Insect_resistance+plot_effective_Simpson_Insect_resistance+plot_evenness_Insect_resistance+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff", plot = richness_shannon_pd_Simpson_eveness_Insect_resistance.tiff)
dev.off()

###############################################################################################
alpha_500_PD_seedweight_1000=alpha_500_PD[!is.na(alpha_500_PD$seedweight_1000),];alpha_500_PD_seedweight_1000
myco_weight=c("#0B775E","#FB9A99","yellow3","grey35");myco_weight
###################
plot_richness_seedweight_1000=ggboxplot(alpha_500_PD_seedweight_1000, x = "seedweight_1000", y = "Richness", fill= "seedweight_1000",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_seedweight_1000


#Effective_alpha_diversity
plot_effective_richness_seedweight_1000=ggboxplot(alpha_500_PD_seedweight_1000, x = "seedweight_1000", y = "Effective_Richness", fill= "seedweight_1000",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_seedweight_1000

#SHANON DIVERSITY
plot_Shannon_seedweight_1000=ggboxplot(alpha_500_PD_seedweight_1000, x = "seedweight_1000", y = "Shannon_Index", fill= "seedweight_1000", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_seedweight_1000

plot_effective_Shannon_seedweight_1000=ggboxplot(alpha_500_PD_seedweight_1000, x = "seedweight_1000", y = "Shannon_Effective", fill= "seedweight_1000",  ylab = "Eff. Shannon div.", xlab = "Weight of 1000 seeds (grams)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_seedweight_1000

#Phylogenetic diversity
plot_PD_seedweight_1000=ggboxplot(alpha_500_PD_seedweight_1000, x = "seedweight_1000", y = "PD", fill= "seedweight_1000", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_seedweight_1000

plot_Simpson_seedweight_1000=ggboxplot(alpha_500_PD_seedweight_1000, x = "seedweight_1000", y = "Simpson_Index", fill= "seedweight_1000", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_seedweight_1000

plot_effective_Simpson_seedweight_1000=ggboxplot(alpha_500_PD_seedweight_1000, x = "seedweight_1000", y = "Simpson_Effective", fill= "seedweight_1000",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_seedweight_1000

#EVENNESS
plot_evenness_seedweight_1000=ggboxplot(alpha_500_PD_seedweight_1000, x = "seedweight_1000", y = "Evenness", fill= "seedweight_1000", ylab = "Evenness", xlab = "Weight of 1000 seeds (grams)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_weight)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_seedweight_1000

----
#Fig.S10
----
tiff("richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff=(plot_richness_seedweight_1000+plot_effective_richness_seedweight_1000+plot_Shannon_seedweight_1000+plot_effective_Shannon_seedweight_1000+plot_PD_seedweight_1000+plot_Simpson_seedweight_1000+plot_effective_Simpson_seedweight_1000+plot_evenness_seedweight_1000+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff
ggsave("richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff", plot = richness_shannon_pd_Simpson_eveness_seedweight_1000.tiff)
dev.off()

##########################
##########################
alpha_500_PD_yield=alpha_500_PD[!is.na(alpha_500_PD$yield),];alpha_500_PD_yield
myco_yield=c("chartreuse4","yellow3","#0B775E","#00AFBB","grey35");myco_yield
#########################################################################################################################
plot_richness_yield=ggboxplot(alpha_500_PD_yield, x = "yield", y = "Richness", fill= "yield",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_yield


#Effective_alpha_diversity
plot_effective_richness_yield=ggboxplot(alpha_500_PD_yield, x = "yield", y = "Effective_Richness", fill= "yield",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_yield

#SHANON DIVERSITY
plot_Shannon_yield=ggboxplot(alpha_500_PD_yield, x = "yield", y = "Shannon_Index", fill= "yield", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_yield


plot_effective_Shannon_yield=ggboxplot(alpha_500_PD_yield, x = "yield", y = "Shannon_Effective", fill= "yield",  ylab = "Eff. Shannon div.", xlab = "Yield (Kg/Sq.Km)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_yield

#Phylogenetic diversity
plot_PD_yield=ggboxplot(alpha_500_PD_yield, x = "yield", y = "PD", fill= "yield", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_yield

plot_Simpson_yield=ggboxplot(alpha_500_PD_yield, x = "yield", y = "Simpson_Index", fill= "yield", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_yield

plot_effective_Simpson_yield=ggboxplot(alpha_500_PD_yield, x = "yield", y = "Simpson_Effective", fill= "yield",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_yield

#EVENNESS
plot_evenness_yield=ggboxplot(alpha_500_PD_yield, x = "yield", y = "Evenness", fill= "yield", ylab = "Evenness", xlab = "Yield (Kg/Sq.Km)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_yield)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_yield

----
#Fig.S9
----
tiff("richness_shannon_pd_Simpson_eveness_yield.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_yield.tiff=(plot_richness_yield+plot_effective_richness_yield+plot_Shannon_yield+plot_effective_Shannon_yield+plot_PD_yield+plot_Simpson_yield+plot_effective_Simpson_yield+plot_evenness_yield+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_yield.tiff
ggsave("richness_shannon_pd_Simpson_eveness_yield.tiff", plot = richness_shannon_pd_Simpson_eveness_yield.tiff)
dev.off()

###############################################################################################alpha_500_PD_ovaries=alpha_500_PD[!is.na(alpha_500_PD$ovaries),];alpha_500_PD_ovaries
myco_ovaries=c("#00AFBB","#FB9A99","chartreuse4");myco_ovaries
##########################
###########################
plot_richness_ovaries=ggboxplot(alpha_500_PD_ovaries, x = "ovaries", y = "Richness", fill= "ovaries",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_ovaries
#ggsave("plot_richness.TIFF", plot = plot_richness)
#+facet_wrap(~ovaries);
#Chao1_simpson.tiff=(plot6 /plot7 );Chao1_simpson.tiff
#ggsave("Chao1_simpson.tiff", plot = Chao1_simpson.tiff)

#Effective_alpha_diversity
plot_effective_richness_ovaries=ggboxplot(alpha_500_PD_ovaries, x = "ovaries", y = "Effective_Richness", fill= "ovaries",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_ovaries
#ggsave("plot_effective_richness.TIFF", plot = plot_effective_richness)

#SHANON DIVERSITY
plot_Shannon_ovaries=ggboxplot(alpha_500_PD_ovaries, x = "ovaries", y = "Shannon_Index", fill= "ovaries", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_ovaries
#ggsave("plot_Shannon.TIFF", plot = plot_Shannon)


plot_effective_Shannon_ovaries=ggboxplot(alpha_500_PD_ovaries, x = "ovaries", y = "Shannon_Effective", fill= "ovaries",  ylab = "Eff. Shannon div.", xlab = "Number of ovaries")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_ovaries
#ggsave("plot_effective_Shannon.TIFF", plot = plot_effective_Shannon)

#Phylogenetic diversity
plot_PD_ovaries=ggboxplot(alpha_500_PD_ovaries, x = "ovaries", y = "PD", fill= "ovaries", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_ovaries
#ggsave("plot_PD.TIFF", plot = plot_PD)


plot_Simpson_ovaries=ggboxplot(alpha_500_PD_ovaries, x = "ovaries", y = "Simpson_Index", fill= "ovaries", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_ovaries
#ggsave("plot_Simpson.TIFF", plot = plot_Simpson)

plot_effective_Simpson_ovaries=ggboxplot(alpha_500_PD_ovaries, x = "ovaries", y = "Simpson_Effective", fill= "ovaries",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_ovaries
#ggsave("plot_effective_Simpson.TIFF", plot = plot_effective_Simpson)

#EVENNESS
 
plot_evenness_ovaries=ggboxplot(alpha_500_PD_ovaries, x = "ovaries", y = "Evenness", fill= "ovaries", ylab = "Evenness", xlab = "Number of ovaries")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_ovaries)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_ovaries
#ggsave("plot_evenness.TIFF", plot = plot_evenness)

----
#Fig.S5
----
tiff("richness_shannon_pd_Simpson_eveness_ovaries.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_ovaries.tiff=(plot_richness_ovaries+plot_effective_richness_ovaries+plot_Shannon_ovaries+plot_effective_Shannon_ovaries+plot_PD_ovaries+plot_Simpson_ovaries+plot_effective_Simpson_ovaries+plot_evenness_ovaries+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_ovaries.tiff
ggsave("richness_shannon_pd_Simpson_eveness_ovaries.tiff", plot = richness_shannon_pd_Simpson_eveness_ovaries.tiff)
dev.off()
###############################################################################################
alpha_500_PD_Fruit_hori_diameter=alpha_500_PD[!is.na(alpha_500_PD$Fruit_hori_diameter),];alpha_500_PD_Fruit_hori_diameter
myco_horiz=c("chartreuse4","#00AFBB","#FB9A99");myco_horiz
##########################
###########################
plot_richness_Fruit_hori_diameter=ggboxplot(alpha_500_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Richness", fill= "Fruit_hori_diameter",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_Fruit_hori_diameter
#ggsave("plot_richness.TIFF", plot = plot_richness)
#+facet_wrap(~Fruit_hori_diameter);
#Chao1_simpson.tiff=(plot6 /plot7 );Chao1_simpson.tiff
#ggsave("Chao1_simpson.tiff", plot = Chao1_simpson.tiff)

#Effective_alpha_diversity
plot_effective_richness_Fruit_hori_diameter=ggboxplot(alpha_500_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Effective_Richness", fill= "Fruit_hori_diameter",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_Fruit_hori_diameter
#ggsave("plot_effective_richness.TIFF", plot = plot_effective_richness)

#SHANON DIVERSITY
plot_Shannon_Fruit_hori_diameter=ggboxplot(alpha_500_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Shannon_Index", fill= "Fruit_hori_diameter", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_Fruit_hori_diameter
#ggsave("plot_Shannon.TIFF", plot = plot_Shannon)


plot_effective_Shannon_Fruit_hori_diameter=ggboxplot(alpha_500_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Shannon_Effective", fill= "Fruit_hori_diameter",  ylab = "Eff. Shannon div.", xlab = "Berry horizontal diameter (cm)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_Fruit_hori_diameter
#ggsave("plot_effective_Shannon.TIFF", plot = plot_effective_Shannon)

#Phylogenetic diversity
plot_PD_Fruit_hori_diameter=ggboxplot(alpha_500_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "PD", fill= "Fruit_hori_diameter", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_Fruit_hori_diameter
#ggsave("plot_PD.TIFF", plot = plot_PD)


plot_Simpson_Fruit_hori_diameter=ggboxplot(alpha_500_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Simpson_Index", fill= "Fruit_hori_diameter", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_Fruit_hori_diameter
#ggsave("plot_Simpson.TIFF", plot = plot_Simpson)

plot_effective_Simpson_Fruit_hori_diameter=ggboxplot(alpha_500_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Simpson_Effective", fill= "Fruit_hori_diameter",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_Fruit_hori_diameter
#ggsave("plot_effective_Simpson.TIFF", plot = plot_effective_Simpson)

#EVENNESS
 
plot_evenness_Fruit_hori_diameter=ggboxplot(alpha_500_PD_Fruit_hori_diameter, x = "Fruit_hori_diameter", y = "Evenness", fill= "Fruit_hori_diameter", ylab = "Evenness", xlab = "Berry horizontal diameter (cm)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_horiz)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_Fruit_hori_diameter
#ggsave("plot_evenness.TIFF", plot = plot_evenness)

----
#Fig.S11
----
tiff("richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff=(plot_richness_Fruit_hori_diameter+plot_effective_richness_Fruit_hori_diameter+plot_Shannon_Fruit_hori_diameter+plot_effective_Shannon_Fruit_hori_diameter+plot_PD_Fruit_hori_diameter+plot_Simpson_Fruit_hori_diameter+plot_effective_Simpson_Fruit_hori_diameter+plot_evenness_Fruit_hori_diameter+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_hori_diameter.tiff)
dev.off()
###############################################################################################

alpha_500_PD_Fruit_long_diameter=alpha_500_PD[!is.na(alpha_500_PD$Fruit_long_diameter),];alpha_500_PD_Fruit_long_diameter
myco_longdiam=c("chartreuse4","#00AFBB","#FB9A99" );myco_longdiam
##########################

plot_richness_Fruit_long_diameter=ggboxplot(alpha_500_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Richness", fill= "Fruit_long_diameter",  ylab = "Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") +geom_jitter(width = 0.3,alpha=0.2)+
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "A")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_richness_Fruit_long_diameter
#ggsave("plot_richness.TIFF", plot = plot_richness)
#+facet_wrap(~Fruit_long_diameter);
#Chao1_simpson.tiff=(plot6 /plot7 );Chao1_simpson.tiff
#ggsave("Chao1_simpson.tiff", plot = Chao1_simpson.tiff)

#Effective_alpha_diversity
plot_effective_richness_Fruit_long_diameter=ggboxplot(alpha_500_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Effective_Richness", fill= "Fruit_long_diameter",  ylab = "Eff. Richness", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "B")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_richness_Fruit_long_diameter
#ggsave("plot_effective_richness.TIFF", plot = plot_effective_richness)

#SHANON DIVERSITY
plot_Shannon_Fruit_long_diameter=ggboxplot(alpha_500_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Shannon_Index", fill= "Fruit_long_diameter", ylab = "Shannon Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "C")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Shannon_Fruit_long_diameter
#ggsave("plot_Shannon.TIFF", plot = plot_Shannon)


plot_effective_Shannon_Fruit_long_diameter=ggboxplot(alpha_500_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Shannon_Effective", fill= "Fruit_long_diameter",  ylab = "Eff. Shannon div.", xlab = "Berry longitudinal diameter (cm)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "D")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Shannon_Fruit_long_diameter
#ggsave("plot_effective_Shannon.TIFF", plot = plot_effective_Shannon)

#Phylogenetic diversity
plot_PD_Fruit_long_diameter=ggboxplot(alpha_500_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "PD", fill= "Fruit_long_diameter", ylab = "PD", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "E")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_PD_Fruit_long_diameter
#ggsave("plot_PD.TIFF", plot = plot_PD)


plot_Simpson_Fruit_long_diameter=ggboxplot(alpha_500_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Simpson_Index", fill= "Fruit_long_diameter", ylab = "Simpson Index", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "F")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_Simpson_Fruit_long_diameter
#ggsave("plot_Simpson.TIFF", plot = plot_Simpson)

plot_effective_Simpson_Fruit_long_diameter=ggboxplot(alpha_500_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Simpson_Effective", fill= "Fruit_long_diameter",  ylab = "Eff. Simpson div.", xlab = "")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "G")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_effective_Simpson_Fruit_long_diameter
#ggsave("plot_effective_Simpson.TIFF", plot = plot_effective_Simpson)

#EVENNESS
 
plot_evenness_Fruit_long_diameter=ggboxplot(alpha_500_PD_Fruit_long_diameter, x = "Fruit_long_diameter", y = "Evenness", fill= "Fruit_long_diameter", ylab = "Evenness", xlab = "Berry longitudinal diameter (cm)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ theme(legend.position="none") + 
  geom_jitter(width = 0.3,alpha=0.2)+theme(text = element_text(size = 8.5)) +scale_fill_manual(values=myco_longdiam)+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=19),
        axis.title.y = element_text(face="bold", colour="gray33", size=19),
        axis.text.x = element_text( colour="gray33", size=19),
        axis.text.y  = element_text( colour="gray33", size=19))+
  theme(text=element_text(size=19,  family="sans"))+ labs(tag = "H")+
  theme(plot.margin = margin(0, 0, 0, 0, "pt"));plot_evenness_Fruit_long_diameter
#ggsave("plot_evenness.TIFF", plot = plot_evenness)

----
#Fig.S12
----
tiff("richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff", units="in", width=13, height=12, res=300)
richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff=(plot_richness_Fruit_long_diameter+plot_effective_richness_Fruit_long_diameter+plot_Shannon_Fruit_long_diameter+plot_effective_Shannon_Fruit_long_diameter+plot_PD_Fruit_long_diameter+plot_Simpson_Fruit_long_diameter+plot_effective_Simpson_Fruit_long_diameter+plot_evenness_Fruit_long_diameter+ plot_layout(ncol = 2, byrow = FALSE));richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff
ggsave("richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff", plot = richness_shannon_pd_Simpson_eveness_Fruit_long_diameter.tiff)
dev.off()

```

